% Template for PLoS
%DIF LATEXDIFF DIFFERENCE FILE
%DIF DEL original.tex   Sun Oct 18 18:13:07 2020
%DIF ADD review_1.tex   Sun Oct 18 18:09:06 2020
% Version 3.5 March 2018
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended 
% to minimize problems and delays during our production 
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % % 
%
% Once your paper is accepted for publication, 
% PLEASE REMOVE ALL TRACKED CHANGES in this file 
% and leave only the final text of your manuscript. 
% PLOS recommends the use of latexdiff to track changes during review, as this will help to maintain a clean tex file.
% Visit https://www.ctan.org/pkg/latexdiff?lang=en for info or contact us at latex@plos.org.
%
%
% There are no restrictions on package use within the LaTeX files except that 
% no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% The manuscript LaTeX source should be contained within a single file (do not use \input, \externaldocument, or similar commands).
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file. 
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission. 
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig" instead of "Figure".
% See http://journals.plos.org/plosone/s/figures for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - spacing/line breaks within cells to alter layout or alignment
% - do not nest tabular environments (no tabular environments within tabular environments)
% - no graphics or colored text (cell background color/shading OK)
% See http://journals.plos.org/plosone/s/tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- eqnarrayS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your eqnarrays and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://journals.plos.org/plosone/s/latex
%
% For inline eqnarrays, please be sure to include all portions of an eqnarray in the math environment.  For example, x$^2$ is incorrect; this should be formatted as $x^2$ (or $\mathrm{x}^2$ if the romanized font is desired).
%
% Do not include text that is not math in the math environment. For example, CO2 should be written as CO\textsubscript{2} instead of CO$_2$.
%
% Please add line breaks to long display eqnarrays when possible in order to fit size of the column. 
%
% For inline eqnarrays, please do not include punctuation (commas, etc) within the math environment unless this is part of the eqnarray.
%
% When adding superscript or subscripts outside of brackets/braces, please group using {}.  For example, change "[U(D,E,\gamma)]^2" to "{[U(D,E,\gamma)]}^2". 
%
% Do not use \cal for caligraphic font.  Instead, use \mathcal{}
%
% % % % % % % % % % % % % % % % % % % % % % % % 
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8x]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% author's packages
\usepackage[shortcuts]{extdash}
\usepackage{lmodern} % Chloe: better font rendering 

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace} 
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in 
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother



% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}
\newcommand{\mean}[1]{$\overline{\mbox{#1}}$}
\newcommand{\median}[1]{$\hat{\mbox{#1}}$}

%% END MACROS SECTION


\usepackage[shortcuts]{extdash} 
 %DIF > 

 %DIF > 
%DIF PREAMBLE EXTENSION ADDED BY LATEXDIFF
%DIF UNDERLINE PREAMBLE %DIF PREAMBLE
\RequirePackage[normalem]{ulem} %DIF PREAMBLE
\RequirePackage{color}\definecolor{RED}{rgb}{1,0,0}\definecolor{BLUE}{rgb}{0,0,1} %DIF PREAMBLE
\providecommand{\DIFaddtex}[1]{{\protect\color{blue}\uwave{#1}}} %DIF PREAMBLE
\providecommand{\DIFdeltex}[1]{{\protect\color{red}\sout{#1}}}                      %DIF PREAMBLE
%DIF SAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddbegin}{} %DIF PREAMBLE
\providecommand{\DIFaddend}{} %DIF PREAMBLE
\providecommand{\DIFdelbegin}{} %DIF PREAMBLE
\providecommand{\DIFdelend}{} %DIF PREAMBLE
\providecommand{\DIFmodbegin}{} %DIF PREAMBLE
\providecommand{\DIFmodend}{} %DIF PREAMBLE
%DIF FLOATSAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddFL}[1]{\DIFadd{#1}} %DIF PREAMBLE
\providecommand{\DIFdelFL}[1]{\DIFdel{#1}} %DIF PREAMBLE
\providecommand{\DIFaddbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFaddendFL}{} %DIF PREAMBLE
\providecommand{\DIFdelbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFdelendFL}{} %DIF PREAMBLE
%DIF HYPERREF PREAMBLE %DIF PREAMBLE
\providecommand{\DIFadd}[1]{\texorpdfstring{\DIFaddtex{#1}}{#1}} %DIF PREAMBLE
\providecommand{\DIFdel}[1]{\texorpdfstring{\DIFdeltex{#1}}{}} %DIF PREAMBLE
\newcommand{\DIFscaledelfig}{0.5}
%DIF HIGHLIGHTGRAPHICS PREAMBLE %DIF PREAMBLE
\RequirePackage{settobox} %DIF PREAMBLE
\RequirePackage{letltxmacro} %DIF PREAMBLE
\newsavebox{\DIFdelgraphicsbox} %DIF PREAMBLE
\newlength{\DIFdelgraphicswidth} %DIF PREAMBLE
\newlength{\DIFdelgraphicsheight} %DIF PREAMBLE
% store original definition of \includegraphics %DIF PREAMBLE
\LetLtxMacro{\DIFOincludegraphics}{\includegraphics} %DIF PREAMBLE
\newcommand{\DIFaddincludegraphics}[2][]{{\color{blue}\fbox{\DIFOincludegraphics[#1]{#2}}}} %DIF PREAMBLE
\newcommand{\DIFdelincludegraphics}[2][]{% %DIF PREAMBLE
\sbox{\DIFdelgraphicsbox}{\DIFOincludegraphics[#1]{#2}}% %DIF PREAMBLE
\settoboxwidth{\DIFdelgraphicswidth}{\DIFdelgraphicsbox} %DIF PREAMBLE
\settoboxtotalheight{\DIFdelgraphicsheight}{\DIFdelgraphicsbox} %DIF PREAMBLE
\scalebox{\DIFscaledelfig}{% %DIF PREAMBLE
\parbox[b]{\DIFdelgraphicswidth}{\usebox{\DIFdelgraphicsbox}\\[-\baselineskip] \rule{\DIFdelgraphicswidth}{0em}}\llap{\resizebox{\DIFdelgraphicswidth}{\DIFdelgraphicsheight}{% %DIF PREAMBLE
\setlength{\unitlength}{\DIFdelgraphicswidth}% %DIF PREAMBLE
\begin{picture}(1,1)% %DIF PREAMBLE
\thicklines\linethickness{2pt} %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,0){\framebox(1,1){}}}% %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,0){\line( 1,1){1}}}% %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,1){\line(1,-1){1}}}% %DIF PREAMBLE
\end{picture}% %DIF PREAMBLE
}\hspace*{3pt}}} %DIF PREAMBLE
} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddbegin}{\DIFaddbegin} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddend}{\DIFaddend} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelbegin}{\DIFdelbegin} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelend}{\DIFdelend} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddbegin}{\DIFOaddbegin \let\includegraphics\DIFaddincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddend}{\DIFOaddend \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelbegin}{\DIFOdelbegin \let\includegraphics\DIFdelincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelend}{\DIFOaddend \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddbeginFL}{\DIFaddbeginFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddendFL}{\DIFaddendFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelbeginFL}{\DIFdelbeginFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelendFL}{\DIFdelendFL} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddbeginFL}{\DIFOaddbeginFL \let\includegraphics\DIFaddincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddendFL}{\DIFOaddendFL \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelbeginFL}{\DIFOdelbeginFL \let\includegraphics\DIFdelincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelendFL}{\DIFOaddendFL \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
%DIF END PREAMBLE EXTENSION ADDED BY LATEXDIFF

\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\DIFdelbegin %DIFDELCMD < \newline{Comparing and combining network-guided biomarker discovery to discover genetic susceptibility mechanisms to breast cancer on the GENESIS study} %%%
\DIFdelend \DIFaddbegin \newline{Biological networks and GWAS: comparing and combining network methods to understand the genetics of familial breast cancer susceptibility in the GENESIS study} \DIFaddend % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Héctor Climente-González\textsuperscript{1,2,3,4*}, 
Christine Lonjou\textsuperscript{1,2,3}, 
Fabienne Lesueur\textsuperscript{1,2,3},
Dominique Stoppa-Lyonnet\textsuperscript{5,6,7\textcurrency}, 
Nadine Andrieu\textsuperscript{1,2,3}, 
Chloé-Agathe Azencott\textsuperscript{3,1,2}\DIFaddbegin \DIFadd{,
GENESIS study group\textsuperscript{\textpilcrow}
}\DIFaddend \\
\bigskip
\textbf{1} Institut Curie, PSL Research University, F-75005 Paris, France;\\
\textbf{2} INSERM, U900, F-75005 Paris, France;\\
\textbf{3} MINES ParisTech, PSL Research University, CBIO-Centre for Computational Biology, F-75006 Paris, France;\\
\textbf{4} RIKEN Center for Advanced Intelligence Project (AIP), Tokyo, Japan; \\
\textbf{5} Service de Génétique, Institut Curie, F-75005 Paris, France;\\
\textbf{6} INSERM, U830, F-75005 Paris, France;\\
\textbf{7} Université Paris Descartes.\\
\bigskip

% Insert additional author notes using the symbols described below. Insert symbol callouts after author names as necessary.
% 
% Remove or comment out the author notes below if they aren't used.
%
% Primary Equal Contribution Note
% \Yinyang These authors contributed equally to this work.

% Additional Equal Contribution Note
% Also use this double-dagger symbol for special authorship notes, such as senior authorship.
% \ddag These authors also contributed equally to this work.

% Current address notes
\textcurrency For the GENESIS study group % change symbol to "\textcurrency a" if more than one current address note
% \textcurrency b Insert second current address 
% \textcurrency c Insert third current address

% Deceased author note
% \dag Deceased

% Group/Consortium Author Note
\textpilcrow Membership list can be found in the Acknowledgments section.

% Use the asterisk to denote corresponding authorship and provide email address in note below.
* hector.climente(at)riken.jp

\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}
\DIFdelbegin \DIFdel{Systems biology provides a comprehensive approach to biomarker discovery and biological hypothesis building. Indeed, it allows to jointly consider the statistical association between gene variation and a phenotype}\DIFdelend \DIFaddbegin \DIFadd{Network approaches to disease use biological networks, which model functional relationships between the molecules in a cell, to generate hypotheses about the genetics of complex diseases. Several among them jointly consider gene scores, representing the association between each gene and the disease}\DIFaddend , and the biological context of each gene, \DIFdelbegin \DIFdel{represented as }\DIFdelend \DIFaddbegin \DIFadd{modeled by }\DIFaddend a network. \DIFdelbegin \DIFdel{In this work}\DIFdelend \DIFaddbegin \DIFadd{Here}\DIFaddend , we study six \DIFdelbegin \DIFdel{network methods which identify subnetworks with high association scores to a phenotype. Specifically, we examine their utility to discover new biomarkers for breast cancer susceptibility by interrogating }\DIFdelend \DIFaddbegin \DIFadd{such network methods using gene scores from GENESIS, }\DIFaddend a genome-wide association study (GWAS) \DIFdelbegin \DIFdel{focused }\DIFdelend on French women with \DIFdelbegin \DIFdel{a family history of breast cancerand tested negative for pathogenic variants in }\emph{\DIFdel{BRCA1}} %DIFAUXCMD
\DIFdel{and }\emph{\DIFdel{BRCA2}}%DIFAUXCMD
\DIFdel{. We perform an in-depth benchmarking of the methodswith regards to size of the solution subnetwork, their utility as biomarkers, and the stability and the runtime of the methods. By trading statistical stringency for biological meaningfulness, most network methods give }\DIFdelend \DIFaddbegin \DIFadd{non-BRCA familial breast cancer. We provide a critical comparison of these six methods, discussing the impact of their mathematical formulation and parameters. Using a biological network yields }\DIFaddend more compelling results than standard \DIFdelbegin \DIFdel{SNP- and gene-level analyses, recovering causal subnetworks tightly related to }\DIFdelend \DIFaddbegin \DIFadd{GWAS analyses. Indeed, we find significant overlaps between our solutions and the genes identified in the largest GWAS on breast }\DIFaddend cancer susceptibility. \DIFdelbegin \DIFdel{For instance, we show a general alteration of the neighborhood of }\DIFdelend \DIFaddbegin \DIFadd{We further propose to combine these solutions into a consensus network, which brings further insights. The consensus network contains }\DIFaddend \emph{COPS5}, a gene related to multiple hallmarks of cancer\DIFdelbegin \DIFdel{. Importantly, we find a significantly large overlap between the genes in the solution networks and the genes significantly associated in the largest GWAS on susceptibility to breast cancer. Yet, network methods are notably unstable, producing different results when the input data changes slightly. To account for that, we produce }\DIFdelend \DIFaddbegin \DIFadd{, and 14 of its neighbors. The main drawback of network methods is that they are not robust to small perturbations in their inputs. Therefore, we propose }\DIFaddend a stable consensus \DIFdelbegin \DIFdel{subnetwork}\DIFdelend \DIFaddbegin \DIFadd{solution}\DIFaddend , formed by the most consistently selected genes \DIFdelbegin \DIFdel{. The stable consensus }\DIFdelend \DIFaddbegin \DIFadd{in multiple subsamples of the data. In GENESIS, it }\DIFaddend is composed of 68 genes, enriched in known breast cancer susceptibility genes (\emph{BLM}, \emph{CASP8}, \emph{CASP10}, \emph{DNAJC1}, \emph{FGFR2}, \emph{MRPS30}, and \emph{SLC4A7}, \DIFdelbegin \DIFdel{Fisher's exact test }\DIFdelend P\=/value~=~$3 \times 10^{-4}$) and occupying more central positions in the network than \DIFdelbegin \DIFdel{average}\DIFdelend \DIFaddbegin \DIFadd{most genes}\DIFaddend . The network \DIFdelbegin \DIFdel{seems }\DIFdelend \DIFaddbegin \DIFadd{is }\DIFaddend organized around \emph{CUL3}, \DIFdelbegin \DIFdel{encoding an ubiquitin ligase related protein that regulates the protein levels }\DIFdelend \DIFaddbegin \DIFadd{which is involved in the regulation }\DIFaddend of several genes \DIFdelbegin \DIFdel{involved in }\DIFdelend \DIFaddbegin \DIFadd{linked to }\DIFaddend cancer progression. In conclusion, \DIFdelbegin \DIFdel{this article shows the pertinence of network-based analyses to tackle known issues with GWAS, namely }\DIFdelend \DIFaddbegin \DIFadd{we showed how network methods help overcome the }\DIFaddend lack of statistical power \DIFdelbegin \DIFdel{and of interpretable solutions}\DIFdelend \DIFaddbegin \DIFadd{of GWAS and improve their interpretation}\DIFaddend . Project-agnostic implementations of \DIFdelbegin \DIFdel{each of the network }\DIFdelend \DIFaddbegin \DIFadd{all }\DIFaddend methods are available at \url{https://github.com/hclimente/gwas-tools}\DIFdelbegin \DIFdel{to facilitate their application to other GWAS datasets}\DIFdelend .

% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step. 
% Author Summary not valid for PLOS ONE submissions.   
\section*{Author summary}
\DIFdelbegin \DIFdel{In genome-wide }\DIFdelend \DIFaddbegin \DIFadd{Genome-wide }\DIFaddend association studies (GWAS) \DIFdelbegin \DIFdel{, }\DIFdelend \DIFaddbegin \DIFadd{scan }\DIFaddend thousands of genomes \DIFdelbegin \DIFdel{are scanned }\DIFdelend to identify variants associated with a complex trait. Over the last 15 years, GWAS have advanced our understanding of the genetics of complex diseases, and in particular of hereditary cancers. \DIFdelbegin \DIFdel{Yet}\DIFdelend \DIFaddbegin \DIFadd{However}\DIFaddend , they have led to an apparent paradox: the more we perform such studies, the more it seems that the entire genome is involved in every disease. \DIFdelbegin \DIFdel{An elegant explanation has been proposed with the omnigenic model }\DIFdelend \DIFaddbegin \DIFadd{The omnigenic model offers an appealing explanation}\DIFaddend : only a limited number of \DIFdelbegin \DIFdel{core }\DIFdelend \DIFaddbegin \emph{\DIFadd{core}} \DIFaddend genes are directly involved in the disease\DIFdelbegin \DIFdel{; }\DIFdelend \DIFaddbegin \DIFadd{, }\DIFaddend but gene functions are deeply interrelated, \DIFdelbegin \DIFdel{so that }\DIFdelend \DIFaddbegin \DIFadd{and so }\DIFaddend many other genes \DIFdelbegin \DIFdel{are able to }\DIFdelend \DIFaddbegin \DIFadd{can }\DIFaddend alter the function of the core genes. These interrelations are often modeled as networks, and multiple algorithms have been proposed to use these networks to identify the subset of core genes involved in a specific trait. \DIFdelbegin \DIFdel{In this study , we characterize }\DIFdelend \DIFaddbegin \DIFadd{This study applies and compares }\DIFaddend six such network methods on GENESIS, a GWAS dataset for familial breast cancer in the French population. Combining these approaches allows us to identify potentially novel breast cancer susceptibility genes \DIFdelbegin \DIFdel{, }\DIFdelend and provides a mechanistic explanation for their role in the development of the disease. \DIFdelbegin \DIFdel{Our pipeline can easily be applied to other diseases.
}%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend \DIFaddbegin \DIFadd{We provide ready-to-use implementations of all the examined methods.
}\DIFaddend \linenumbers

% Use "Eq" instead of "eqnarray" for eqnarray citations.
% For figure citations, please use "Fig" instead of "Figure".
% Place figure captions after the first paragraph in which they are cited.
\section{Introduction}

In human health, genome-wide association studies (GWAS) aim at quantifying how single-nucleotide polymorphisms (SNPs) predispose to complex diseases, like diabetes or some forms of cancer \cite{bush_chapter_2012}. To that end, in a typical GWAS\DIFaddbegin \DIFadd{, }\DIFaddend thousands of unrelated samples are genotyped: the cases, suffering \DIFaddbegin \DIFadd{from }\DIFaddend the disease of interest, and the controls\DIFaddbegin \DIFadd{, taken }\DIFaddend from the general population. Then, a \DIFdelbegin \DIFdel{per-SNP statistical association test is conducted }\DIFdelend \DIFaddbegin \DIFadd{statistical test of association }\DIFaddend (e.g.\DIFaddbegin \DIFadd{, }\DIFaddend based on logistic regression) \DIFaddbegin \DIFadd{is conducted between each SNP and the phenotype}\DIFaddend . Those SNPs with a P\=/value lower than a conservative Bonferroni threshold are candidates to further studies in an independent cohort. Once the risk SNPs have been discovered, they can be used for risk assessment \DIFdelbegin \DIFdel{, and to deepen }\DIFdelend \DIFaddbegin \DIFadd{and deepening }\DIFaddend our understanding of the disease.

GWAS have successfully identified thousands of variants underlying many common diseases \cite{buniello_nhgri-ebi_2019}. However, this experimental setting also presents \DIFdelbegin \DIFdel{intrinsic }\DIFdelend \DIFaddbegin \DIFadd{inherent }\DIFaddend challenges. Some of them stem from the high dimensionality of the problem, as every GWAS to date studies more variants than samples are genotyped. This limits the statistical power of the experiment, as \DIFdelbegin \DIFdel{only }\DIFdelend \DIFaddbegin \DIFadd{it can only detect }\DIFaddend variants with larger effects \DIFdelbegin \DIFdel{can be detected }\DIFdelend \cite{visscher_10_2017}. This is particularly problematic since the prevailing view is that most genetic architectures involve many variants with small effects \cite{visscher_10_2017}. Additionally, to avoid false positives, \DIFaddbegin \DIFadd{most GWAS apply }\DIFaddend a conservative multiple test correction\DIFdelbegin \DIFdel{is applied}\DIFdelend , typically the previously mentioned Bonferroni correction. However, Bonferroni correction is overly conservative when the statistical tests \DIFdelbegin \DIFdel{are correlated, as is the case }\DIFdelend \DIFaddbegin \DIFadd{correlate, as happens }\DIFaddend in GWAS \cite{wang_statistical_2018}. Another open issue is the interpretation of the results, as the functional consequences of most common variants are \DIFdelbegin \DIFdel{not well understood}\DIFdelend \DIFaddbegin \DIFadd{unknown}\DIFaddend . On top of that, recent large-sampled studies suggest that numerous loci spread all along the genome contribute to a degree to any complex trait, in accordance with the infinitesimal model \cite{barton_infinitesimal_2017}. The recently proposed omnigenic model \cite{boyle_expanded_2017} offers an explanation: genes are \DIFdelbegin \DIFdel{very functionally inter-related }\DIFdelend \DIFaddbegin \DIFadd{strongly interrelated }\DIFaddend and influence each other's \DIFdelbegin \DIFdel{behavior}\DIFdelend \DIFaddbegin \DIFadd{function}\DIFaddend , which allows alterations in most genes to impact the subset of ``core'' genes directly involved in the disease's mechanism. Hence, a comprehensive statistical framework \DIFdelbegin \DIFdel{which }\DIFdelend \DIFaddbegin \DIFadd{that }\DIFaddend includes the structure of biological data might \DIFdelbegin \DIFdel{address the aforementioned issues }\DIFdelend \DIFaddbegin \DIFadd{help alleviate the issues above}\DIFaddend .

For this reason, many authors turn to network biology to handle the complex interplay of biomolecules that lead to disease \DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{furlong_human_2013}}\hspace{0pt}%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{furlong_human_2013,leisersonNetworkAnalysisGWAS2013}}\hspace{0pt}%DIFAUXCMD
}\DIFaddend . As its name suggests, network biology models biology as a network, where the biomolecules under study, often genes, are nodes, and selected functional relationships are edges that link them. These relationships come from evidence that the genes jointly contribute to a biological function; for instance, their \DIFdelbegin \DIFdel{expression is }\DIFdelend \DIFaddbegin \DIFadd{expressions are }\DIFaddend correlated, or their products establish a protein-protein interaction. Under this view, complex diseases are not the consequence of a single altered gene, but of the interaction of multiple interdependent molecules \cite{barabasi_network_2011}. In fact, an examination of biological networks shows that disease genes have differential properties \DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{barabasi_network_2011,pinero_uncovering_2016}}\hspace{0pt}%DIFAUXCMD
. This is particularly true for cancer driver genes, which tend to be key players in connecting different, densely-connected communities of genes. Therefore, studying the neighborhood of disease-associated genes is effective at identifying new onesthat are involved in the same biological functions \mbox{%DIFAUXCMD
\cite{huang_systematic_2018}}\hspace{0pt}%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{barabasi_network_2011,cai_broker_2010}}\hspace{0pt}%DIFAUXCMD
: they tend to occupy central positions in the network (although not the most central ones); disease genes for the same pathology tend to cluster in modules; and often they are bottlenecks that interconnect modules}\DIFaddend .

Network-based \DIFdelbegin \DIFdel{biomarker }\DIFdelend discovery methods exploit \DIFdelbegin \DIFdel{this relatedness }\DIFdelend \DIFaddbegin \DIFadd{the differential properties described above }\DIFaddend to identify disease genes on GWAS data \DIFdelbegin \DIFdel{\mbox{%DIFAUXCMD
\cite{azencott_network-guided_2016}}\hspace{0pt}%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{azencott_network-guided_2016,huang_systematic_2018}}\hspace{0pt}%DIFAUXCMD
}\DIFaddend . In essence, each \DIFdelbegin \DIFdel{SNP has }\DIFdelend \DIFaddbegin \DIFadd{gene receives }\DIFaddend a score of association with the disease, \DIFdelbegin \DIFdel{given by the experiment, and functionally }\DIFdelend \DIFaddbegin \DIFadd{computed from the GWAS data, and a set of }\DIFaddend biological relationships, given by a network built on prior knowledge. Then, the problem becomes finding a functionally-related set of highly-scoring genes. \DIFdelbegin \DIFdel{Different }\DIFdelend \DIFaddbegin \DIFadd{Multiple }\DIFaddend solutions have been proposed to this problem, often stemming from \DIFdelbegin \DIFdel{divergent }\DIFdelend \DIFaddbegin \DIFadd{different }\DIFaddend mathematical frameworks and considerations of what the optimal solution looks like. \DIFdelbegin \DIFdel{Some methods strongly constrain }\DIFdelend \DIFaddbegin \DIFadd{For example, some methods restrict }\DIFaddend the problem to \DIFdelbegin \DIFdel{certain kinds }\DIFdelend \DIFaddbegin \DIFadd{specific types }\DIFaddend of subnetworks. Such is the \DIFdelbegin \DIFdel{extreme }\DIFdelend case of LEAN \cite{gwinner_network-based_2016}, which focuses on ``star'' subnetworks, i.e.\DIFdelbegin \DIFdel{instances were }\DIFdelend \DIFaddbegin \DIFadd{, instances where }\DIFaddend both a gene and its direct interactors are associated with the disease. Other algorithms, like dmGWAS \cite{jia_dmgwas:_2011} and heinz \cite{dittrich_identifying_2008}, \DIFdelbegin \DIFdel{focus on larger }\DIFdelend \DIFaddbegin \DIFadd{do not impose such strong constraints and search for }\DIFaddend subnetworks interconnecting genes with high association scores. However, they differ in their tolerance to the inclusion of low-scoring nodes \DIFdelbegin \DIFdel{, }\DIFdelend and the topology of the solution. Lastly, other methods also consider the topology of the network, favoring groups of nodes that are not only high-scoring \DIFdelbegin \DIFdel{, }\DIFdelend but also densely interconnected; such is the case of HotNet2 \cite{leiserson_pan-cancer_2015}, SConES \cite{azencott_efficient_2013}, and SigMod \cite{liu_sigmod:_2017}.

In this work, we \DIFdelbegin \DIFdel{analyze the effectiveness of }\DIFdelend \DIFaddbegin \DIFadd{studied the relevance of network-based approaches to genetics by applying }\DIFaddend these six network methods \DIFdelbegin \DIFdel{for biomarker discovery on }\DIFdelend \DIFaddbegin \DIFadd{to }\DIFaddend GWAS data. \DIFdelbegin \DIFdel{While all of them capture susceptibility mechanisms resembling that postulated by }\DIFdelend \DIFaddbegin \DIFadd{They use different interpretations of }\DIFaddend the omnigenic model \DIFdelbegin \DIFdel{, they do so in different ways, }\DIFdelend and provide a representative view of the field. We worked on the GENESIS dataset \cite{sinilnikova_genesis:_2016}, a study on familial breast cancer conducted in the French population. After a classical GWAS approach, we \DIFdelbegin \DIFdel{use }\DIFdelend \DIFaddbegin \DIFadd{used }\DIFaddend these network methods to \DIFdelbegin \DIFdel{recover }\DIFdelend \DIFaddbegin \DIFadd{identify }\DIFaddend additional breast cancer \DIFdelbegin \DIFdel{biomarkers}\DIFdelend \DIFaddbegin \DIFadd{susceptibility genes}\DIFaddend . Lastly, we \DIFdelbegin \DIFdel{carry out a comparison of }\DIFdelend \DIFaddbegin \DIFadd{compared }\DIFaddend the solutions obtained by the different methods \DIFdelbegin \DIFdel{, and aggregate }\DIFdelend \DIFaddbegin \DIFadd{and aggregated }\DIFaddend them to obtain \DIFdelbegin \DIFdel{a consensus network }\DIFdelend \DIFaddbegin \DIFadd{consensus solutions }\DIFaddend of predisposition to familial breast cancer \DIFaddbegin \DIFadd{that addressed their shortcomings}\DIFaddend . 

% Results and Discussion can be combined.
\section{Results}

\subsection{Conventional SNP- and gene-based analyses \DIFdelbegin \DIFdel{confirm that }\DIFdelend \DIFaddbegin \DIFadd{retrieve the }\DIFaddend \emph{FGFR2} locus \DIFdelbegin \DIFdel{is associated with familial breast cancer}\DIFdelend \DIFaddbegin \DIFadd{in the GENESIS dataset}\DIFaddend }
\label{results:conventional}

We conducted association analyses in the GENESIS dataset \DIFaddbegin \DIFadd{(Section~\ref{methods:data}) }\DIFaddend at both SNP and gene levels (Section~\DIFdelbegin \DIFdel{\ref{methods:node_score}). Two genomic regions have }\DIFdelend \DIFaddbegin \DIFadd{\ref{methods:conventional}). At the SNP level, two genomic regions had }\DIFaddend a P\=/value lower than the Bonferroni threshold on chromosomes 10 and 16 (\nameref{sfig:snp_gene_manhattan}A). The former overlaps with \DIFaddbegin \DIFadd{the }\DIFaddend gene \emph{FGFR2}\DIFdelbegin \DIFdel{; }\DIFdelend \DIFaddbegin \DIFadd{, }\DIFaddend the latter with \emph{CASC16} \DIFdelbegin \DIFdel{, and it is located near }\DIFdelend \DIFaddbegin \DIFadd{and }\DIFaddend the protein-coding gene \emph{TOX3}. Variants in both \emph{FGFR2} and \emph{TOX3} have been repeatedly associated with breast cancer susceptibility in other case-control studies \cite{Michailidou2017}, \emph{BRCA1} and \emph{BRCA2} carrier studies \cite{Mulligan2011}, and \DIFdelbegin \DIFdel{in }\DIFdelend hereditary breast and ovarian cancer families negative for mutations in \emph{BRCA1} and \emph{BRCA2} \cite{rinella_genetic_2013}. \DIFdelbegin \DIFdel{In our studied population}\DIFdelend \DIFaddbegin \DIFadd{At the gene level}\DIFaddend , only \emph{FGFR2} was significantly associated with breast cancer \DIFdelbegin \DIFdel{at the gene-level }\DIFdelend (\nameref{sfig:snp_gene_manhattan}B).

Closer examination \DIFdelbegin \DIFdel{reveals two }\DIFdelend \DIFaddbegin \DIFadd{revealed two other }\DIFaddend regions (3p24 and 8q24) having low, albeit not genome-wide significant, P\=/values. Both of them have been associated \DIFdelbegin \DIFdel{to }\DIFdelend \DIFaddbegin \DIFadd{with }\DIFaddend breast cancer susceptibility in the past \cite{brisbin_meta-analysis_2011,search_newly_2009}. We applied an L1-penalized logistic regression \DIFdelbegin \DIFdel{on the whole dataset (}\DIFdelend \DIFaddbegin \DIFadd{using all GENESIS genotypes as input and the phenotype (cancer/healthy) as the outcome (}\DIFaddend Section~\ref{methods:classifier}). \DIFdelbegin \DIFdel{It }\DIFdelend \DIFaddbegin \DIFadd{The algorithm }\DIFaddend selected 100 SNPs, both from all \DIFdelbegin \DIFdel{aforementioned regions }\DIFdelend \DIFaddbegin \DIFadd{regions mentioned above }\DIFaddend and new ones (\nameref{sfig:snp_gene_manhattan}C). \DIFdelbegin \DIFdel{Yet, it is }\DIFdelend \DIFaddbegin \DIFadd{However, it was }\DIFaddend unclear why those SNPs were selected, as emphasized by the high P\=/value of some of them, which further complicates the biological interpretation. Moreover, and in opposition to what would be expected under the omnigenic model, the genes to which these SNPs map \DIFdelbegin \DIFdel{to }\DIFdelend (Section~\DIFdelbegin \DIFdel{\ref{methods:node_score}) are }\DIFdelend \DIFaddbegin \DIFadd{\ref{methods:snp2gene}) were }\DIFaddend not interconnected in the \DIFdelbegin \DIFdel{PPIN (}\DIFdelend \DIFaddbegin \DIFadd{protein-protein interaction network (PPIN, }\DIFaddend Section~\ref{methods:networks}). \DIFdelbegin \DIFdel{In addition}\DIFdelend \DIFaddbegin \DIFadd{Moreover}\DIFaddend , the classification performance of the \DIFdelbegin \DIFdel{method is very low , and L1-penalized methods select only one of several correlated variables and are prone to instability, which further complicates interpretation. This motivates }\DIFdelend \DIFaddbegin \DIFadd{model was low (sensitivity = 55\%, specificity = 55\%, Section~\ref{methods:benchmark}). Together, these issues motivated }\DIFaddend exploring network methods, which \DIFdelbegin \DIFdel{trade statistical significance for biological relevance }\DIFdelend \DIFaddbegin \DIFadd{consider not only statistical association but also the location of each gene in a PPIN }\DIFaddend to find susceptibility \DIFdelbegin \DIFdel{subnetworks.

In fact, such methods provided comparably (poor) classification performance to L1-penalized logistic regression (Fig~\ref{fig:benchmark}B), while providing more interpretable solutions. 

}\DIFdelend \DIFaddbegin \DIFadd{genes.

}\DIFaddend 

\subsection{Network methods successfully identify genes associated with breast cancer}
\label{results:separate_networks}

We applied six network methods to the GENESIS dataset (Section~\ref{methods:methods}). As none of the networks examined by LEAN was significant (\DIFdelbegin \DIFdel{BH }\DIFdelend \DIFaddbegin \DIFadd{Benjamini-Hochberg }[\DIFadd{BH}] \DIFadd{correction }\DIFaddend adjusted P\=/value \DIFdelbegin \DIFdel{< }\DIFdelend \DIFaddbegin \DIFadd{$<$ }\DIFaddend 0.05), we obtained \DIFdelbegin \DIFdel{six }\DIFdelend \DIFaddbegin \DIFadd{five }\DIFaddend solutions (Fig~\ref{fig:solution_overview}): one for each of the remaining four gene-based methods, \DIFaddbegin \DIFadd{and }\DIFaddend one for SConES GI (which works at the SNP level)\DIFdelbegin \DIFdel{, and the consensus.

These solutions differ }\DIFdelend \DIFaddbegin \DIFadd{.

}

\begin{figure}[!ht]
  \centering
  \includegraphics[width=\textwidth]{./figures/figure_1.png}
  \caption{\textbf{\DIFaddFL{Overview of the solutions produced by the different network methods (Section~\ref{methods:methods}) on the GENESIS dataset.}} \DIFaddFL{As LEAN did not produce any significant solution (BH adjusted P\=/value~$<$~0.05), it is not shown. Unless indicated otherwise, results refer to SNPs for SConES GI, and to genes for the other methods. }\textbf{\DIFaddFL{(A)}}\DIFaddFL{~Overlap between the genes selected by each method, measured by Pearson correlation between indicator vectors (Sections~\ref{methods:algorithm_comparison} and \ref{methods:snp2gene}). }\textbf{\DIFaddFL{(B)}}\DIFaddFL{~Distribution of VEGAS2 P-values of the genes in the PPIN not selected by any network method (12\,213), and of those selected by 1 (575), 2 (73), or 3 (20) methods. }\textbf{\DIFaddFL{(C)}}\DIFaddFL{~Solution networks produced by the different methods. }\textbf{\DIFaddFL{(D)}}\DIFaddFL{~Manhattan plots of SNPs/genes; in black, the method's solution. The red line indicates the Bonferroni threshold (2.54~\texttimes{}~10\textsuperscript{-7} for SNPs, 1.53~\texttimes{}~10\textsuperscript{-6} for genes).}}
  \label{fig:solution_overview}
\end{figure}

\DIFadd{These solutions differed }\DIFaddend in many aspects, making it hard to draw joint conclusions. For starters, the overlap between the genes featured in each solution \DIFdelbegin \DIFdel{is quite }\DIFdelend \DIFaddbegin \DIFadd{was relatively }\DIFaddend small (Fig~\ref{fig:solution_overview}\DIFaddbegin \DIFadd{A). However, the methods tended to agree on the genes with the strongest signal: genes selected by more methods tended to have lower P-value of association (Fig~\ref{fig:solution_overview}}\DIFaddend B).

\DIFdelbegin \DIFdel{Another prominent difference is their }\DIFdelend \DIFaddbegin 

\DIFadd{Another major difference was the solution }\DIFaddend size: the largest solution, produced by HotNet2, \DIFdelbegin \DIFdel{contains }\DIFdelend \DIFaddbegin \DIFadd{contained }\DIFaddend 440 genes, while heinz's contained only 4 genes. While SConES GI \DIFdelbegin \DIFdel{failed to }\DIFdelend \DIFaddbegin \DIFadd{did not }\DIFaddend recover any protein coding gene, \DIFdelbegin \DIFdel{by dealing }\DIFdelend \DIFaddbegin \DIFadd{working }\DIFaddend with SNP networks \DIFdelbegin \DIFdel{it retrieved }\DIFdelend \DIFaddbegin \DIFadd{rather than gene networks allowed it to retrieve }\DIFaddend four subnetworks in intergenic regions \DIFdelbegin \DIFdel{, and another one }\DIFdelend \DIFaddbegin \DIFadd{and another subnetwork }\DIFaddend overlapping an RNA gene (\emph{RNU6-420P}).

\DIFdelbegin \DIFdel{Their topologies also differ}\DIFdelend \DIFaddbegin 

\DIFadd{The topologies of the five solutions differed as well (Fig~\ref{fig:solution_overview}C)}\DIFaddend , as measured by the median centrality \DIFdelbegin \DIFdel{(Table~\ref{tab:gene_solutions}) }\DIFdelend and the number of connected components (\DIFdelbegin \DIFdel{Fig~\ref{fig:solution_overview}D). Only two methods have }\DIFdelend \DIFaddbegin \DIFadd{Table~\ref{tab:gene_solutions}). Three methods yielded }\DIFaddend more than one connected component: SConES, as described above, \DIFaddbegin \DIFadd{SigMod, }\DIFaddend and HotNet2. HotNet2 produced 135 subnetworks, 115 of which have \DIFdelbegin \DIFdel{less }\DIFdelend \DIFaddbegin \DIFadd{fewer }\DIFaddend than five genes. The second largest subnetwork (13 nodes) \DIFdelbegin \DIFdel{contains }\DIFdelend \DIFaddbegin \DIFadd{contained }\DIFaddend the two breast cancer susceptibility genes \emph{CASP8} and \emph{BLM} \DIFdelbegin \DIFdel{. Lastly, a pathway enrichment analysis (Section\ref{methods:pathway_enrichment}) also showed similarities and differences in the underlying mechanisms. It linked different parts of SigMod's solution network to four processes: protein translation (including mitochondrial), mRNA splicing, protein misfolding, and keratinization (BH adjusted P\=/values~<}\DIFdelend \DIFaddbegin \DIFadd{(Section}\DIFaddend ~\DIFdelbegin \DIFdel{0.03).

Interestingly, dmGWAS solution is also related to protein misfolding (}\emph{\DIFdel{attenuation phase}}%DIFAUXCMD
\DIFdel{, BH adjusted P\=/value~=~0.01). But, additionally, it includes submodules of proteins related to mitosis, DNA damage, and regulation of TP53 (BH adjusted P\=/values~<~0.05), which match previously known mechanisms of breast cancer susceptibility \mbox{%DIFAUXCMD
\cite{nielsen_hereditary_2016}}\hspace{0pt}%DIFAUXCMD
. As with SigMod, the genes in HotNet2's solution are involved in mitochondrial translation (BH adjusted P\=/value~=~1.87 \texttimes{} 10\textsuperscript{-4}), but also in glycogen metabolism and transcription of nuclear receptors (BH adjusted P\=/value~<~0.04).

}\DIFdelend \DIFaddbegin \DIFadd{\ref{methods:bcac}).

}\DIFaddend 

\begin{table}[!ht]
  \begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
  \centering
  \caption{
  {\bf Summary statistics on the solutions of multiple network methods on the \DIFdelbeginFL \DIFdelFL{gene-gene interaction network}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{PPIN}\DIFaddendFL . The first row contains the summary statistics on the whole \DIFdelbeginFL \DIFdelFL{network}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{PPIN}\DIFaddendFL .}}
  \DIFdelbeginFL %DIFDELCMD < \begin{tabular}{lrrrrr}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \begin{tabular}{lrrrrrr}
  \DIFaddendFL {\bf Network } & {\bf \# genes } & {\bf \# edges } & {\bf \DIFaddbeginFL \DIFaddFL{\# components }} & {\bf \DIFaddendFL \mean{Betweenness} } & {\bf \median{P}\textsubscript{gene} } & {\bf \DIFdelbeginFL \DIFdelFL{\(\rho\)\textsubscript{consensus} }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{\# genes in consensus }\DIFaddendFL } \\
  \thickhline
  HINT HT           & 13\,619 & 142\,541  & \DIFaddbeginFL \DIFaddFL{15  }& \DIFaddendFL 16\,706   & 0.46  & \DIFdelbeginFL \DIFdelFL{0.066}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{93/93 }\DIFaddendFL \\
  \hline
  dmGWAS            & 194     & 450       & \DIFaddbeginFL \DIFaddFL{1   }& \DIFaddendFL 49\,115   & 0.19  & \DIFdelbeginFL \DIFdelFL{0.41}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{55/93 }\DIFaddendFL \\
  heinz             & 4       & 3         & \DIFaddbeginFL \DIFaddFL{1   }& \DIFaddendFL 113\,633  & 0.001 & \DIFdelbeginFL \DIFdelFL{0.21}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{4/93  }\DIFaddendFL \\
  HotNet2           & 440     & 374       & \DIFaddbeginFL \DIFaddFL{130 }& \DIFaddendFL 7\,739    & 0.048 & \DIFdelbeginFL \DIFdelFL{0.31}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{63/93 }\DIFaddendFL \\
  LEAN              & 0       & 0         & \DIFdelbeginFL \DIFdelFL{- }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{0   }\DIFaddendFL & -         & -     \DIFaddbeginFL & \DIFaddFL{0/93  }\DIFaddendFL \\
  SConES GI         & 0 (1)   & 0         & \DIFdelbeginFL \DIFdelFL{- }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{0   }\DIFaddendFL & -         & -     \DIFaddbeginFL & \DIFaddFL{0/93  }\DIFaddendFL \\
  SigMod            & 142     & 249       & \DIFaddbeginFL \DIFaddFL{11  }& \DIFaddendFL 92\,603   & 0.008 & \DIFdelbeginFL \DIFdelFL{0.73}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{84/93 }\DIFaddendFL \\
  \hline
  Consensus         & 93      & 186       & \DIFaddbeginFL \DIFaddFL{21  }& \DIFaddendFL 50\,737   & 0.006 & \DIFdelbeginFL \DIFdelFL{1}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{93/93 }\DIFaddendFL \\
  Stable consensus  & 68      & 49        & \DIFaddbeginFL \DIFaddFL{32  }& \DIFaddendFL 94\,854   & 0.005 & \DIFdelbeginFL \DIFdelFL{0.54 }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{43/93 }\DIFaddendFL \\
  \end{tabular}
  \begin{flushleft} \textbf{\DIFdelbeginFL \emph{\DIFdelFL{\# genes}}%DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{\# genes}\DIFaddendFL }:\DIFaddbeginFL \DIFaddFL{~}\DIFaddendFL number of genes selected out of those that are part of the PPIN; for SConES GI\DIFaddbeginFL \DIFaddFL{, }\DIFaddendFL the total number of genes, including RNA genes, was added in parentheses. \DIFaddbeginFL \textbf{\DIFaddFL{\# components}}\DIFaddFL{:~number of connected components. }\DIFaddendFL \textbf{\mean{Betweenness}}:\DIFdelbeginFL \DIFdelFL{mean }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{~median }\DIFaddendFL betweenness of the selected genes in the PPIN. \textbf{\median{P}\textsubscript{gene}}:\DIFdelbeginFL \DIFdelFL{median }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{~median VEGAS2 }\DIFaddendFL P\=/value of the selected genes. \textbf{\DIFdelbeginFL \DIFdelFL{\(\rho\)\textsubscript{consensus}}%DIFDELCMD < \MBLOCKRIGHTBRACE%%%
\DIFdelFL{:Pearson correlation between the subnetwork and the }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{\# genes in }\DIFaddendFL consensus\DIFdelbeginFL \DIFdelFL{network}\DIFdelendFL \DIFaddbeginFL }\DIFaddFL{:~number of genes in common between the method's solution and the 93 genes in the consensus solution}\DIFaddendFL .
  \end{flushleft}
  \label{tab:gene_solutions}
  \end{adjustwidth}
\end{table}

\DIFdelbegin %DIFDELCMD < \begin{figure}[!ht]
%DIFDELCMD <   \centering
%DIFDELCMD <   \includegraphics[height=\textwidth]{./figures/figure_1.png}
%DIFDELCMD <   %%%
%DIFDELCMD < \caption{%
{%DIFAUXCMD
%DIFDELCMD < {\bf %%%
\DIFdelFL{Overview of the solutions produced by the different network methods (Section~\ref{methods:methods}) on the GENESIS dataset. }%DIFDELCMD < } %%%
\DIFdelFL{As LEAN did not produce any significant gene (BH adjusted P\=/value~<~0.05), it was excluded. Unless indicated otherwise, results refer to genes, except for SConES GI which are at the SNP-level. }\textbf{\DIFdelFL{(A)}} %DIFAUXCMD
\DIFdelFL{Manhattan plots of SNPs/genes; in black, the method's solution. The Bonferroni threshold is indicated by a red line (2.54~\texttimes{}~10\textsuperscript{-7} for SNPs, 1.53~\texttimes{}~10\textsuperscript{-6} for genes). }\textbf{\DIFdelFL{(B)}} %DIFAUXCMD
\DIFdelFL{Overlap between the genes selected by each of the methods, measured by Pearson correlation between indicator vectors. }\textbf{\DIFdelFL{(C)}} %DIFAUXCMD
\DIFdelFL{Precision and recall of the evaluated methods with respect to Bonferroni-significant SNPs/genes in BCAC. For reference, we added a gray line with a slope of 1. }\textbf{\DIFdelFL{(D)}} %DIFAUXCMD
\DIFdelFL{Solution networks produced by the different methods.}}
  %DIFAUXCMD
%DIFDELCMD < \label{fig:solution_overview}
%DIFDELCMD < \end{figure}
%DIFDELCMD < 
%%%
\DIFdelend \DIFaddbegin \DIFadd{Lastly, a pathway enrichment analysis (Section~\ref{methods:pathway_enrichment}) also showed similarities and differences between the methods' solutions. It linked different parts of SigMod's solution to four processes (}\nameref{stab:sigmod_pwy}\DIFadd{): protein translation (including mitochondrial), mRNA splicing, protein misfolding, and keratinization (BH adjusted P\=/values~$<$~0.03). Interestingly, the dmGWAS solution (}\nameref{stab:dmgwas_pwy}\DIFadd{) was also related to protein misfolding (}\emph{\DIFadd{attenuation phase}}\DIFadd{, BH adjusted P\=/value~=~0.01). However, it additionally included submodules of proteins related to mitosis, DNA damage, and regulation of TP53 (BH adjusted P\=/values~$<$~0.05), which match previously known mechanisms of breast cancer susceptibility \mbox{%DIFAUXCMD
\cite{nielsen_hereditary_2016}}\hspace{0pt}%DIFAUXCMD
. As with SigMod, the genes in HotNet2's solution (}\nameref{stab:hotnet2_pwy}\DIFadd{) were involved in mitochondrial translation (BH adjusted P\=/value~=~1.87 \texttimes{} 10\textsuperscript{-4}), but also in glycogen metabolism and transcription of nuclear receptors (BH adjusted P\=/value~$<$~0.04).

}\DIFaddend 

Despite their differences, there \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend additional common themes. All obtained \DIFdelbegin \DIFdel{solution subnetworks have }\DIFdelend \DIFaddbegin \DIFadd{solutions had }\DIFaddend lower association P\=/values than the whole PPIN (median \DIFaddbegin \DIFadd{VEGAS2 }\DIFaddend P\=/value $\ll 0.46$, Table~\ref{tab:gene_solutions}), despite containing genes with higher P\=/values as well (Fig~\ref{fig:solution_overview}\DIFdelbegin \DIFdel{A}\DIFdelend \DIFaddbegin \DIFadd{D}\DIFaddend ). This \DIFdelbegin \DIFdel{exemplifies }\DIFdelend \DIFaddbegin \DIFadd{illustrates }\DIFaddend the trade-off between \DIFdelbegin \DIFdel{statistical significance }\DIFdelend \DIFaddbegin \DIFadd{controlling for type I error }\DIFaddend and biological relevance. However, there are nuances between solutions in this regard: heinz strongly favored genes with lower P\=/values, while dmGWAS was less conservative (median \DIFaddbegin \DIFadd{VEGAS2 }\DIFaddend P\=/values 0.0012 and 0.19, respectively); SConES tended to select whole LD-blocks; and HotNet2 and SigMod were less likely to select low scoring genes.

\DIFaddbegin 

\DIFaddend Additionally, the \DIFdelbegin \DIFdel{solution subnetworks }\DIFdelend \DIFaddbegin \DIFadd{solutions }\DIFaddend presented other desirable properties. First, \DIFdelbegin \DIFdel{five }\DIFdelend \DIFaddbegin \DIFadd{four }\DIFaddend of them were enriched in known breast cancer susceptibility genes (\DIFdelbegin \DIFdel{consensus, }\DIFdelend dmGWAS, heinz, HotNet2, and SigMod, Fisher's exact test one-sided P\=/value\DIFdelbegin \DIFdel{< }\DIFdelend \DIFaddbegin \DIFadd{~$<$~}\DIFaddend 0.03). Second, the genes in \DIFdelbegin \DIFdel{four solution subnetworks displayedon average a }\DIFdelend \DIFaddbegin \DIFadd{three solutions displayed, on average, a significantly }\DIFaddend higher betweenness centrality than the rest of the genes \DIFdelbegin \DIFdel{, a difference that is significant in four solutions (consensus, }\DIFdelend \DIFaddbegin \DIFadd{(}\DIFaddend dmGWAS, HotNet2, and SigMod, Wilcoxon rank-sum test P\=/value\DIFdelbegin \DIFdel{< }\DIFdelend \DIFaddbegin \DIFadd{~$<$~}\DIFaddend 1.4\DIFdelbegin \DIFdel{\texttimes{} }\DIFdelend \DIFaddbegin \DIFadd{~\texttimes{}~}\DIFaddend 10\textsuperscript{-21}). This agrees with the notion that disease genes are more central than other non-essential genes \cite{pinero_uncovering_2016}, an observation that holds in breast cancer (one-tailed Wilcoxon rank-sum test P\=/value~=~2.64 \texttimes{} 10\textsuperscript{-5} when comparing the betweenness of known susceptibility genes versus the rest). Interestingly, \DIFdelbegin \DIFdel{SConES selected SNPs that are }\DIFdelend \DIFaddbegin \DIFadd{the SNPs in SConES' solution were }\DIFaddend also more central than the average SNP (\DIFdelbegin \DIFdel{Supplementary table~}\DIFdelend \nameref{stab:snp_solutions}), suggesting that causal SNPs are also more central than \DIFdelbegin \DIFdel{non associated }\DIFdelend \DIFaddbegin \DIFadd{non-associated }\DIFaddend SNPs.

\subsection{A case study: the consensus \DIFdelbegin \DIFdel{network}\DIFdelend \DIFaddbegin \DIFadd{solution}\DIFaddend }
\label{results:consensus}

Despite \DIFdelbegin \DIFdel{the heterogeneity of the solutions, }\DIFdelend their shared properties\DIFdelbegin \DIFdel{suggest that each method captures }\DIFdelend \DIFaddbegin \DIFadd{, the differences between the solutions suggested that each of them captured }\DIFaddend different aspects of cancer susceptibility. Indeed, \DIFdelbegin \DIFdel{only 20 genes are common to more than twosolutions (}%DIFDELCMD < \nameref{sfig:consensus_stats}%%%
\DIFdel{A), but encouragingly, }\DIFdelend \DIFaddbegin \DIFadd{out of the 668 genes that were selected by at least one method, only 93 were selected by at least two, 20 by three, and none by four or more. Encouragingly, }\DIFaddend the more methods selected a gene, the higher its association score to the phenotype (\DIFdelbegin %DIFDELCMD < \nameref{sfig:consensus_stats}%%%
\DIFdel{B). To leverage on }\DIFdelend \DIFaddbegin \DIFadd{Fig~\ref{fig:solution_overview}B), a relationship that plateaued at 2. Hence, to leverage }\DIFaddend their strengths and compensate \DIFaddbegin \DIFadd{for }\DIFaddend their respective weaknesses, we built a consensus \DIFdelbegin \DIFdel{subnetwork that captures the mechanisms most shared among the solution subnetworks }\DIFdelend \DIFaddbegin \DIFadd{solution using the genes shared among at least two solutions }\DIFaddend (Section~\ref{methods:methods}). This \DIFdelbegin \DIFdel{subnetwork }\DIFdelend \DIFaddbegin \DIFadd{solution }\DIFaddend (Fig~\ref{fig:consensus}) \DIFdelbegin \DIFdel{contains }\DIFdelend \DIFaddbegin \DIFadd{contained }\DIFaddend 93 genes and \DIFdelbegin \DIFdel{exhibits }\DIFdelend \DIFaddbegin \DIFadd{exhibited }\DIFaddend the aforementioned properties of the individual solutions: enrichment in breast cancer susceptibility genes and higher betweenness centrality than the rest of the genes. 

\begin{figure}[!ht]
  \centering
  \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=.7\linewidth]{./figures/figure_3.pdf}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \includegraphics[width=.9\linewidth]{./figures/figure_3.pdf}
  \DIFaddendFL \caption{ \DIFdelbeginFL %DIFDELCMD < {\bf %%%
\DIFdelFL{Consensus subnetwork on GENESIS }\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{Consensus solution on GENESIS (Section~\ref{methods:methods}).}} \textbf{\DIFaddFL{(A)}}\DIFaddFL{~Manhattan plot of genes; in black, the ones in the consensus solution. The red line indicates the Bonferroni threshold }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{Section}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{1.53}\DIFaddendFL ~\DIFdelbeginFL \DIFdelFL{\ref{methods:methods}}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{\texttimes{}~10\textsuperscript{-6} for genes}\DIFaddendFL )\DIFdelbeginFL %DIFDELCMD < }%%%
\DIFdelendFL . \DIFaddbeginFL \textbf{\DIFaddFL{(B)}}\DIFaddFL{~Consensus network. }\DIFaddendFL Each \DIFdelbeginFL \DIFdelFL{node }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{gene }\DIFaddendFL is represented by a pie chart, which shows the methods that selected it. We \DIFdelbeginFL \DIFdelFL{labeled (and }\DIFdelendFL enlarged \DIFdelbeginFL \DIFdelFL{) }\DIFdelendFL the two most central genes (\emph{COPS5} and \emph{OFD1})\DIFdelbeginFL \DIFdelFL{and those genes that are }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, the }\DIFaddendFL known breast cancer susceptibility genes\DIFaddbeginFL \DIFaddFL{, }\DIFaddendFL and \DIFdelbeginFL \DIFdelFL{/or significantly associated with breast cancer susceptibility in }\DIFdelendFL the \DIFdelbeginFL \DIFdelFL{BCAC dataset}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{BCAC-significant genes (Section~\ref{methods:bcac})}\DIFaddendFL . \DIFaddbeginFL \textbf{\DIFaddFL{(C)}}\DIFaddFL{~}\DIFaddendFL The \DIFdelbeginFL \DIFdelFL{latter ones }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{nodes }\DIFaddendFL are \DIFdelbeginFL \DIFdelFL{also }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{in the same disposition as in panel B, but we indicated every gene name. We }\DIFaddendFL colored in pink \DIFdelbeginFL \DIFdelFL{. All gene }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the }\DIFaddendFL names \DIFdelbeginFL \DIFdelFL{are indicated in }%DIFDELCMD < \nameref{sfig:consensus-names}%%%
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{of known breast cancer susceptibility genes and BCAC-significant genes}\DIFaddendFL .}
  \label{fig:consensus}
\end{figure}

A pathway enrichment analysis of the genes in the consensus \DIFdelbegin \DIFdel{network also shows }\DIFdelend \DIFaddbegin \DIFadd{solution also showed }\DIFaddend similar pathways as the individual solutions \DIFaddbegin \DIFadd{(}\nameref{stab:consensus_pwy}\DIFadd{)}\DIFaddend . We found two involved mechanisms: \emph{mitochondrial translation} and \emph{attenuation phase}. The former is supported by genes like \emph{MRPS30} (VEGAS2 P\=/value~=~0.001), which encode a mitochondrial ribosomal protein and was also linked to breast cancer susceptibility \cite{quigley_5p12_2014}. Interestingly, increased mitochondrial translation has been found in cancer cells \cite{Yu2016Repositioning}, and its inhibition \DIFaddbegin \DIFadd{was }\DIFaddend proposed as a therapeutic target. With regards to \DIFaddbegin \DIFadd{the }\DIFaddend attenuation phase of heat shock response, it \DIFdelbegin \DIFdel{involves }\DIFdelend \DIFaddbegin \DIFadd{involved }\DIFaddend three Hsp70 chaperones: HSPA1A, HSPA1B, and HSPA1L. The genes encoding these proteins are all near each other at 6p21, in the region known as HLA. In fact, out of the 22 SNPs \DIFdelbegin \DIFdel{that map }\DIFdelend \DIFaddbegin \DIFadd{mapped }\DIFaddend to any of these three genes, 9 \DIFdelbegin \DIFdel{map to all of them}\DIFdelend \DIFaddbegin \DIFadd{mapped to all three}\DIFaddend , and 4 to two, \DIFdelbegin \DIFdel{making }\DIFdelend \DIFaddbegin \DIFadd{which made it }\DIFaddend hard to disentangle their effects. \emph{HSPA1A} was the most strongly associated gene (VEGAS2 P\=/value~=~8.37~\texttimes{}~10\textsuperscript{-4}).  

Topologically\DIFdelbegin \DIFdel{the consensus consists }\DIFdelend \DIFaddbegin \DIFadd{, the consensus consisted }\DIFaddend of a connected component composed of 49 genes \DIFdelbegin \DIFdel{, }\DIFdelend and multiple smaller subnetworks \DIFaddbegin \DIFadd{(Fig~\ref{fig:consensus}B and C)}\DIFaddend . Among the latter, 19 genes \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend in subnetworks containing a single gene or two connected nodes\DIFdelbegin \DIFdel{, implying that they do }\DIFdelend \DIFaddbegin \DIFadd{. This implied that they did }\DIFaddend not have a consistently altered neighborhood \DIFdelbegin \DIFdel{, but are }\DIFdelend \DIFaddbegin \DIFadd{but were }\DIFaddend strongly associated themselves and hence picked by at least two methods. The \DIFdelbegin \DIFdel{opposite would be the case of highly central genes }\DIFdelend \DIFaddbegin \DIFadd{large connected component contained genes that are highly central }\DIFaddend in the PPIN\DIFdelbegin \DIFdel{, a property which is weakly anti-correlated }\DIFdelend \DIFaddbegin \DIFadd{. This property weakly anticorrelated }\DIFaddend with the P\=/value of association to the disease (Pearson correlation coefficient~=~-0.26, \nameref{sfig:consensus_stats}\DIFdelbegin \DIFdel{D}\DIFdelend ). This \DIFdelbegin \DIFdel{suggests that they }\DIFdelend \DIFaddbegin \DIFadd{anticorrelation suggested that these genes }\DIFaddend were selected because they were on the shortest path between two \DIFdelbegin \DIFdel{highly associated genes. In view }\DIFdelend \DIFaddbegin \DIFadd{high scoring genes. Because }\DIFaddend of this, we hypothesize that highly central genes might contribute to the heritability through alterations of their neighborhood, \DIFdelbegin \DIFdel{consistently }\DIFdelend \DIFaddbegin \DIFadd{consistent }\DIFaddend with the omnigenic model of disease \cite{boyle_expanded_2017}. For instance, the most central node in the consensus \DIFdelbegin \DIFdel{network is COPS5}\DIFdelend \DIFaddbegin \DIFadd{solution was }\emph{\DIFadd{COPS5}}\DIFaddend , a component of the COP9 signalosome \DIFdelbegin \DIFdel{which regulates multiple signalling }\DIFdelend \DIFaddbegin \DIFadd{that regulates multiple signaling }\DIFaddend pathways. \emph{COPS5} is related to multiple hallmarks of cancer and is overexpressed in multiple tumors, including breast and ovarian cancer \cite{liu_jab1_cops5_2018}. Despite its lack of association in GENESIS or \DIFdelbegin \DIFdel{BCAC }\DIFdelend \DIFaddbegin \DIFadd{in studies conducted by the Breast Cancer Association Consortium (BCAC) \mbox{%DIFAUXCMD
\cite{Michailidou2017} }\hspace{0pt}%DIFAUXCMD
}\DIFaddend (VEGAS2 P\=/value of 0.22 and 0.14\DIFaddbegin \DIFadd{, }\DIFaddend respectively), its neighbors in the consensus \DIFdelbegin \DIFdel{subnetwork have }\DIFdelend \DIFaddbegin \DIFadd{solution had }\DIFaddend consistently low P\=/values (median VEGAS2 P\=/value~=~0.006).

\subsection{Network methods boost \DIFdelbegin \DIFdel{biomarker }\DIFdelend discovery}
\DIFaddbegin \label{results:boost}

\DIFaddend 

We compared the results \DIFdelbegin \DIFdel{of }\DIFdelend \DIFaddbegin \DIFadd{obtained with }\DIFaddend different network methods to the European sample of \DIFdelbegin \DIFdel{the Breast Cancer Association Consortium (BCAC) \mbox{%DIFAUXCMD
\cite{Michailidou2017}}\hspace{0pt}%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{BCAC}\DIFaddend , the largest GWAS to date on breast cancer (Section~\ref{methods:bcac}). Although BCAC case-control studies do not necessarily target cases with a \DIFdelbegin \DIFdel{family }\DIFdelend \DIFaddbegin \DIFadd{familial }\DIFaddend history of breast cancer \DIFaddbegin \DIFadd{like GENESIS does}\DIFaddend , this comparison is pertinent since we expect a shared genetic architecture at the gene level, at which most network \DIFdelbegin \DIFdel{metods operate. This shared genetic architecture, together }\DIFdelend \DIFaddbegin \DIFadd{methods operate. Together }\DIFaddend with BCAC's scale (90 times more samples than GENESIS)\DIFdelbegin \DIFdel{provides }\DIFdelend \DIFaddbegin \DIFadd{, this shared genetic architecture provided }\DIFaddend a reasonable counterfactual of what we would expect if GENESIS had a larger sample size. We computed a gene association score on BCAC \DIFdelbegin \DIFdel{, in an equivalent way to the one described in Section~\ref{methods:node_score}}\DIFdelend \DIFaddbegin \DIFadd{(Section~\ref{methods:bcac})}\DIFaddend . The solutions provided by the different  network \DIFdelbegin \DIFdel{approaches overlap }\DIFdelend \DIFaddbegin \DIFadd{methods overlapped }\DIFaddend significantly with BCAC \DIFdelbegin \DIFdel{findings }\DIFdelend \DIFaddbegin \DIFadd{hits }\DIFaddend (Fisher's exact test P\=/value \textless~0.019). The gene-based \DIFdelbegin \DIFdel{network methods achieve }\DIFdelend \DIFaddbegin \DIFadd{methods achieved }\DIFaddend comparable precision (2\%-25\%) and recall (1.3-12.1\%) at recovering BCAC-significant genes (\DIFdelbegin \DIFdel{Fig~\ref{fig:solution_overview}C}\DIFdelend \DIFaddbegin \nameref{sfig:additional-benchmarks}\DIFadd{A}\DIFaddend ). Interestingly, while SConES GI \DIFdelbegin \DIFdel{at the SNP-level achieves }\DIFdelend \DIFaddbegin \DIFadd{achieved }\DIFaddend a similar recall \DIFaddbegin \DIFadd{at the SNP-level }\DIFaddend (8.6\%), it \DIFdelbegin \DIFdel{shows }\DIFdelend \DIFaddbegin \DIFadd{showed }\DIFaddend a much higher precision (47.3\%).

\subsection{Network methods share limitations}
\DIFaddbegin \label{results:benchmark}

\DIFaddend 

We compared the six network methods in a 5-fold subsampling setting (Section\DIFdelbegin \DIFdel{\ref{methods:comparison}). Specifically, we measured five }\DIFdelend \DIFaddbegin \DIFadd{~\ref{methods:benchmark}). In this comparison we measured }\DIFaddend properties (Fig~\ref{fig:benchmark} \DIFdelbegin \DIFdel{): }\DIFdelend \DIFaddbegin \DIFadd{and }\nameref{sfig:additional-benchmarks}\DIFadd{): the }\DIFaddend size of the solution\DIFdelbegin \DIFdel{subnetwork; sensitivity and }\DIFdelend \DIFaddbegin \DIFadd{; the sensitivity and the }\DIFaddend specificity of an L1-penalized logistic regression classifier on the selected SNPs; \DIFdelbegin \DIFdel{stability; and }\DIFdelend \DIFaddbegin \DIFadd{the stability of the methods; and their }\DIFaddend computational runtime. The solution size varies greatly between the different methods (Fig~\ref{fig:benchmark}A). Heinz produced the smallest solutions, with an average of 182 selected SNPs \DIFdelbegin \DIFdel{. The largest solutions }\DIFdelend \DIFaddbegin \DIFadd{(Section~\ref{methods:snp2gene}) while the largest ones }\DIFaddend came from SConES GI (6\,256.6 SNPs) \DIFdelbegin \DIFdel{, }\DIFdelend and dmGWAS (4\,255.0 SNPs). \DIFaddbegin \DIFadd{LEAN did not produce any solution in any of the subsamples. 

}

\DIFaddend To determine whether the selected SNPs could \DIFdelbegin \DIFdel{be used for patient classification}\DIFdelend \DIFaddbegin \DIFadd{predict cancer susceptibility}\DIFaddend , we computed the \DIFdelbegin \DIFdel{performance of the classifier on the }\DIFdelend \DIFaddbegin \DIFadd{classifiers' performances on }\DIFaddend \emph{test \DIFdelbegin \DIFdel{dataset}\DIFdelend \DIFaddbegin \DIFadd{sets}\DIFaddend } (\DIFdelbegin \DIFdel{Fig~\ref{fig:benchmark}}\DIFdelend \DIFaddbegin \nameref{sfig:additional-benchmarks}\DIFaddend B). The different classifiers displayed similarly \DIFdelbegin \DIFdel{poor }\DIFdelend \DIFaddbegin \DIFadd{low }\DIFaddend sensitivities and specificities, all in the 0.52 -- 0.56 range. Interestingly, the classifier trained on all the SNPs had a similar performance, despite being the only method aiming \DIFdelbegin \DIFdel{only at minimizing }\DIFdelend \DIFaddbegin \DIFadd{to minimize }\DIFaddend prediction error. \DIFdelbegin \DIFdel{It should be considered that}\DIFdelend \DIFaddbegin \DIFadd{Of course}\DIFaddend , although these performances \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend low, we \DIFdelbegin \DIFdel{do }\DIFdelend \DIFaddbegin \DIFadd{did }\DIFaddend not expect to separate cases from controls well using exclusively genetic data \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{deloscamposComplexTraitPredictionEra2018}}\hspace{0pt}%DIFAUXCMD
}\DIFaddend .

\begin{figure}[!ht]
  \centering
  \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\linewidth]{./figures/figure_4.pdf}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \includegraphics[width=.8\linewidth]{./figures/figure_4.pdf}
  \DIFaddendFL \caption{\DIFdelbeginFL %DIFDELCMD < { \bf %%%
\DIFdelFL{Comparison of network-based GWAS methods on GENESIS. }%DIFDELCMD < } %%%
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{Comparison of network methods on GENESIS.}} \DIFaddendFL Each method was run 5 times on a random subset containing 80\% of the samples \DIFdelbeginFL \DIFdelFL{, }\DIFdelendFL and tested on the remaining samples (Section\DIFdelbeginFL \DIFdelFL{\ref{methods:comparison}}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{~\ref{methods:benchmark}}\DIFaddendFL ). As LEAN did not select any gene, \DIFdelbeginFL \DIFdelFL{it was }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{we }\DIFaddendFL excluded \DIFaddbeginFL \DIFaddFL{it }\DIFaddendFL from \DIFdelbeginFL \DIFdelFL{all }\DIFdelendFL panels \DIFdelbeginFL \DIFdelFL{except }\textbf{\DIFdelFL{D}}%DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{A and B}\DIFaddendFL . \textbf{(A)}\DIFaddbeginFL \DIFaddFL{~}\DIFaddendFL Number of SNPs selected by each method and number of SNPs in the active set \DIFdelbeginFL \DIFdelFL{found }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{(i.e., the number of SNPs selected }\DIFaddendFL by the classifier\DIFaddbeginFL \DIFaddFL{, Section~\ref{methods:classifier})}\DIFaddendFL . Points are the average over the 5 runs; \DIFdelbeginFL \DIFdelFL{lines }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the error bars }\DIFaddendFL represent the standard error of the mean. A grey diagonal line with slope 1 is added for comparison\DIFaddbeginFL \DIFaddFL{, indicating the upper bound of the active set}\DIFaddendFL . For reference, the active set of \DIFdelbeginFL \DIFdelFL{Lasso }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the classifier }\DIFaddendFL using all the SNPs \DIFaddbeginFL \DIFaddFL{as input }\DIFaddendFL included, on average, 154\,117.4 SNPs. \textbf{(B)}\DIFdelbeginFL \DIFdelFL{Sensitivity and specificity on test set of the L1-penalized logistic regression trained on the features selected by each of the methods. In addition, the performance of the classifier trained on all SNPs is displayed. Points are the average over the 5 runs; lines represent the standard error of the mean. }\textbf{\DIFdelFL{(C)}} %DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{~}\DIFaddendFL Pairwise Pearson correlations of the solutions \DIFdelbeginFL \DIFdelFL{used }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{produced }\DIFaddendFL by different methods. A Pearson correlation of 1 means the two solutions are the same. \DIFdelbeginFL \DIFdelFL{A Pearson correlation of 0 means that there is no SNP in common between the two solutions. }\textbf{\DIFdelFL{(D)}} %DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{(C)}}\DIFaddFL{~}\DIFaddendFL Runtime of the evaluated methods, by type of network used (\DIFdelbeginFL \DIFdelFL{gene }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{PPIN }\DIFaddendFL or SNP). For \DIFdelbeginFL \DIFdelFL{gene network-based }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{gene-based }\DIFaddendFL methods, inverted triangles represent the runtime of the algorithm \DIFdelbeginFL \DIFdelFL{itself}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{alone}\DIFaddendFL , and circles the total time, which includes the algorithm themselves and the additional 119\,980 seconds (1 day and 9.33 hours) that VEGAS2 took on average to compute the gene scores from SNP summary statistics. \DIFaddbeginFL \textbf{\DIFaddFL{(D)}}\DIFaddFL{~True positive rate and true negative rate of the methods, obtained using different parameter combinations (Section~\ref{results:parameters}). We used as true positives BCAC-significant SNPs (for SConES and $\chi^2$ + Bonferroni) and genes (for the remaining methods, Section~\ref{methods:bcac}). We used the whole dataset in this panel.}\DIFaddendFL }
  \label{fig:benchmark}
  \end{figure}

Another desirable quality of \DIFdelbegin \DIFdel{an }\DIFdelend \DIFaddbegin \DIFadd{a selection }\DIFaddend algorithm is the stability of the solution with \DIFdelbegin \DIFdel{regards }\DIFdelend \DIFaddbegin \DIFadd{respect }\DIFaddend to small changes in the input (Section~\ref{methods:algorithm_comparison}). \DIFdelbegin \DIFdel{Both heinz and LEAN displayed a high stability }\DIFdelend \DIFaddbegin \DIFadd{Heinz was highly stable }\DIFaddend in our benchmark, \DIFdelbegin \DIFdel{consistently selecting the same genes and no genes over the 5 subsamples, respectively (Fig~\ref{fig:benchmark}C). Conversely, the }\DIFdelend \DIFaddbegin \DIFadd{while the }\DIFaddend other methods displayed similarly low stabilities \DIFaddbegin \DIFadd{(Fig~\ref{fig:benchmark}B)}\DIFaddend .

In terms of computational runtime, the fastest method was heinz (Fig~\ref{fig:benchmark}\DIFdelbegin \DIFdel{D}\DIFdelend \DIFaddbegin \DIFadd{C}\DIFaddend ), which returned a solution in a few seconds. HotNet2 was the slowest (3 days and 14 hours on average). Including the time required to compute the gene scores, however, \DIFdelbegin \DIFdel{slows }\DIFdelend \DIFaddbegin \DIFadd{slowed }\DIFaddend down considerably gene-based methods; on this benchmark, that step took on average 1 day and 9.33 hours. \DIFdelbegin \DIFdel{Considering that}\DIFdelend \DIFaddbegin \DIFadd{Including this first step}\DIFaddend , it took 5 days on average for HotNet2 to produce a result.

\DIFaddbegin \DIFadd{Using different combinations of parameters (Section~\ref{methods:parameters}), we computed how good each of the methods was at recovering the results of a conventional GWAS on BCAC (Section~\ref{methods:bcac}, Fig~\ref{fig:benchmark}D). SConES exhibits the largest area under the curve since, when $\lambda = 0$ (i.e., network topology is disregarded), it is equivalent to a Bonferroni correction. The remaining network methods have similar areas under the curve, with heinz having the largest one.

}

\DIFaddend \subsection{Network topology \DIFdelbegin \DIFdel{matters, }\DIFdelend and \DIFaddbegin \DIFadd{association scores matter and }\DIFaddend might lead to ambiguous results}
\label{results:drawbacks}

As shown above, and despite their similarities, the \DIFdelbegin \DIFdel{network methods produced }\DIFdelend \DIFaddbegin \DIFadd{different ways of modeling the problem led to }\DIFaddend remarkably different solutions. \DIFdelbegin \DIFdel{This is due to the particularities of each methods , and directly or indirectly provide information about the dataset}\DIFdelend \DIFaddbegin \DIFadd{Importantly, understanding which assumptions the methods made allowed us to understand the results more in depth}\DIFaddend . For instance, the fact that LEAN did not return any \DIFdelbegin \DIFdel{biomarker implies that there is }\DIFdelend \DIFaddbegin \DIFadd{gene implied that there was }\DIFaddend no gene such that both itself and its environment \DIFdelbegin \DIFdel{are on average}\DIFdelend \DIFaddbegin \DIFadd{were, on average, }\DIFaddend strongly associated with the disease.

In \DIFdelbegin \DIFdel{this }\DIFdelend \DIFaddbegin \DIFadd{the GENESIS }\DIFaddend dataset, heinz's solution \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend very conservative, providing a small solution with the lowest median P\=/value \DIFdelbegin \DIFdel{for the subnetwork }\DIFdelend (Table~\ref{tab:gene_solutions}). \DIFdelbegin \DIFdel{Due to this parsimonious and highly associated solution, it was the best method to stably select a set of biomarkers }\DIFdelend \DIFaddbegin \DIFadd{By repeatedly selecting this compact solution, heinz was the most stable method }\DIFaddend (Fig~\ref{fig:benchmark}\DIFdelbegin \DIFdel{C}\DIFdelend \DIFaddbegin \DIFadd{B}\DIFaddend ). Its conservativeness \DIFdelbegin \DIFdel{stems }\DIFdelend \DIFaddbegin \DIFadd{stemmed }\DIFaddend from its preprocessing step, which \DIFdelbegin \DIFdel{models }\DIFdelend \DIFaddbegin \DIFadd{modeled }\DIFaddend the gene P\=/values as a mixture model of a beta distribution and a uniform distribution, controlled by an FDR parameter. Due to the limited signal at the gene level in this dataset (\nameref{sfig:snp_gene_manhattan}B), only 36 \DIFdelbegin \DIFdel{of all the genes retain }\DIFdelend \DIFaddbegin \DIFadd{genes retained }\DIFaddend a positive score after that transformation. \DIFdelbegin \DIFdel{Yet}\DIFdelend \DIFaddbegin \DIFadd{However}\DIFaddend , this small solution \DIFdelbegin \DIFdel{does }\DIFdelend \DIFaddbegin \DIFadd{did }\DIFaddend not provide much insight into the susceptibility mechanisms to cancer. Importantly, it \DIFdelbegin \DIFdel{ignores genes that are associated to }\DIFdelend \DIFaddbegin \DIFadd{ignored genes that were associated with }\DIFaddend cancer in this dataset\DIFaddbegin \DIFadd{, }\DIFaddend like \emph{FGFR2}. 

On the other end of the spectrum, dmGWAS, HotNet2, and SigMod produced large solutions. \DIFdelbegin \DIFdel{dmGWAS' subnetwork is the least associated subnetwork on average . This is due to the greedy frameworkit uses, which has a bias for }\DIFdelend \DIFaddbegin \DIFadd{DmGWAS' solution was the lowest scoring solution on average because of its greedy framework, which is biased towards }\DIFaddend larger solutions \cite{nikolayeva_network_2018}. It considered all nodes at distance 2 of the examined subnetwork \DIFdelbegin \DIFdel{, }\DIFdelend and accepted a weakly associated \DIFdelbegin \DIFdel{genes }\DIFdelend \DIFaddbegin \DIFadd{gene }\DIFaddend if it was linked to another, \DIFdelbegin \DIFdel{strongly associated one. This is exacerbated when }\DIFdelend \DIFaddbegin \DIFadd{high scoring one. Aggregating }\DIFaddend the results of successive greedy searches \DIFdelbegin \DIFdel{are aggregated}\DIFdelend \DIFaddbegin \DIFadd{exacerbates this bias}\DIFaddend , leading to a large, tightly connected cluster of unassociated genes (Fig~\ref{fig:issues}A). This relatively low signal-to-noise ratio combined with the large solution requires additional analyses to draw conclusions, such as enrichment analyses. In the same line, HotNet2's \DIFdelbegin \DIFdel{subnetwork is }\DIFdelend \DIFaddbegin \DIFadd{solution was }\DIFaddend even harder to interpret, being composed of 440 genes divided into 135 subnetworks. Lastly, SigMod \DIFdelbegin \DIFdel{misses }\DIFdelend \DIFaddbegin \DIFadd{missed }\DIFaddend some of the \DIFdelbegin \DIFdel{most strongly associated, }\DIFdelend \DIFaddbegin \DIFadd{highest scoring }\DIFaddend breast cancer susceptibility genes in the dataset, like \emph{FGFR2} and \emph{TOX3}.

\begin{figure}[!ht]
  \centering
  \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=\linewidth]{./figures/figure_2.pdf}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \includegraphics[width=.8\linewidth]{./figures/figure_2.pdf}
  \DIFaddendFL \caption{\DIFdelbeginFL %DIFDELCMD < {\bf %%%
\DIFdelFL{Drawbacks encountered when using network guided methods.}%DIFDELCMD < } %%%
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{Drawbacks encountered when using network methods.}} \DIFaddendFL \textbf{(A)}~DmGWAS solution\DIFdelbeginFL \DIFdelFL{subnetwork. Genes }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, }\DIFaddendFL with \DIFdelbeginFL \DIFdelFL{a }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the genes colored according to the -log$_{10}$ of their }\DIFaddendFL P\=/value\DIFdelbeginFL \DIFdelFL{< 0.1 are highlighted in red}\DIFdelendFL . \textbf{(B)}~\DIFaddbeginFL \DIFaddFL{Number of times a gene was selected by either dmGWAS, heinz, LEAN, or SigMod in 100 rewirings of the PPIN (Section~\ref{methods:pathway_enrichment}) and its centrality degree. }\textbf{\DIFaddFL{(C)}}\DIFaddFL{~}\DIFaddendFL Centrality degree and -log\textsubscript{10} of the VEGAS2 P\=/value in BCAC for each of the nodes in the PPIN. We highlighted the genes selected by each method \DIFdelbeginFL \DIFdelFL{, }\DIFdelendFL and the ones selected by more than one (``Consensus''). We labeled the three most central genes that were picked by any method. \DIFdelbeginFL \textbf{\DIFdelFL{(C)}}%DIFAUXCMD
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{(D)}}\DIFaddendFL ~Overlap between the solutions of SConES GS, GM\DIFaddbeginFL \DIFaddFL{, }\DIFaddendFL or GI\DIFdelbeginFL \DIFdelFL{in the different genomic regions}\DIFdelendFL . \DIFaddbeginFL \DIFaddFL{Barplots are colored based on whether the }\DIFaddendFL SNPs \DIFdelbeginFL \DIFdelFL{that were }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{map to a gene or }\DIFaddendFL not \DIFdelbeginFL \DIFdelFL{selected in the studied network, but were selected in another one, are displayed in background color}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{(Section~\ref{methods:snp2gene})}\DIFaddendFL .}
  \label{fig:issues}
  \end{figure}

Another peculiarity of network methods \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend their relationship to degree centrality. \DIFdelbegin \DIFdel{On the one hand we observed that }\DIFdelend \DIFaddbegin \DIFadd{We studied random rewirings of the PPIN that preserved node centrality (Section~\ref{methods:rewiring}). In this setting, network methods favored central genes (Fig~\ref{fig:issues}B) even though }\DIFaddend highly central genes often had no association to \DIFdelbegin \DIFdel{disease }\DIFdelend \DIFaddbegin \DIFadd{breast cancer susceptibility }\DIFaddend (Fig~\ref{fig:issues}\DIFdelbegin \DIFdel{B). On the other, network methods favor central genes, as they often connect high scoring nodes. This was specially the case of SigMod }\DIFdelend \DIFaddbegin \DIFadd{C). We found this bias especially in SigMod (}\nameref{sfig:rewirings}\DIFadd{)}\DIFaddend , which selected three highly central, unassociated genes \DIFaddbegin \DIFadd{in both the PPIN and in many of the random rewirings}\DIFaddend : \emph{COPS5}, \emph{CUL3}\DIFaddbegin \DIFadd{, }\DIFaddend and \emph{FN1}. \DIFdelbegin \DIFdel{As }\DIFdelend \DIFaddbegin \DIFadd{However, as }\DIFaddend we showed in Section\DIFdelbegin \DIFdel{\ref{results:consensus}, }\DIFdelend \DIFaddbegin \DIFadd{~\ref{results:consensus} }\DIFaddend and will show in \ref{results:stable-consensus}, there is evidence in the literature of the contribution of the first two to breast cancer susceptibility. With regards to \emph{FN1}, it encodes a fibronectin, a protein of the extracellular matrix involved in cell adhesion and migration. Overexpression of \emph{FN1} has been observed in breast cancer \cite{Ioachim2002}, and it \DIFdelbegin \DIFdel{is negatively correlated }\DIFdelend \DIFaddbegin \DIFadd{anticorrelates }\DIFaddend with poor prognosis in other cancer types \cite{Yi2016,Sponziello2016}.

By \DIFdelbegin \DIFdel{virtue of }\DIFdelend using a SNP subnetwork, SConES \DIFdelbegin \DIFdel{analyzes }\DIFdelend \DIFaddbegin \DIFadd{analyzed }\DIFaddend each SNP in \DIFdelbegin \DIFdel{their }\DIFdelend \DIFaddbegin \DIFadd{its }\DIFaddend functional context. \DIFdelbegin \DIFdel{It therefore can select SNPs in genes without any associated interactor, as well as SNPs in }\DIFdelend \DIFaddbegin \DIFadd{Therefore, it could select SNPs located in genes not included in the PPIN and in }\DIFaddend non-coding regions\DIFdelbegin \DIFdel{or in non-interacting genes. In fact, due to linkage disequilibrium, SConES favors such genes, as selecting SNPs in an LD-block which overlaps with a gene favors selecting the rest of the gene}\DIFdelend . \DIFdelbegin \DIFdel{This might explain why SConES produces }\DIFdelend \DIFaddbegin \DIFadd{We compared the solution of SConES in the GI network (using PPIN information), to the one using only positional information (GS network) and to the one using positional and gene annotations (GM network). Importantly, SConES produced }\DIFaddend similar results on the GS and GM networks \DIFdelbegin \DIFdel{, heavily affected by linkage disequilibrium }\DIFdelend (\nameref{sfig:pearson_methods}). \DIFdelbegin \DIFdel{On the other hand, SConES penalizes selecting SNPs and not their neighbors. This makes it conservative regarding SNPs with many interactions, like those mapped to hub genes in the PPIN. For this reason, SConES GIdid not select any protein coding gene, despite selecting similar regions as SConES GS }\DIFdelend \DIFaddbegin \DIFadd{While the solutions on those two considerably overlap with SConES GI's, they contained additional gene-coding segments }\DIFaddend (Fig~\ref{fig:issues}C). In fact\DIFaddbegin \DIFadd{, both }\DIFaddend SConES GS and \DIFdelbegin \DIFdel{SConES GM select }\DIFdelend \DIFaddbegin \DIFadd{GM selected chromosome }\DIFaddend regions related to breast cancer, like 3p24 (\emph{SLC4A7}/\emph{NEK10} \cite{ahmed_newly_2009}), 5p12 (\emph{FGF10}, \emph{MRPS30} \cite{quigley_5p12_2014}), 10q26 (\emph{FGFR2}), and 16q12 (\emph{TOX3}). \DIFdelbegin \DIFdel{On top of thoseSConES GS selects }\DIFdelend \DIFaddbegin \DIFadd{In addition to those, SConES GS selected }\DIFaddend region 8q24\DIFaddbegin \DIFadd{, also linked to breast cancer }\DIFaddend (\emph{POU5F1B} \cite{breyer_expressed_2014}). 

\DIFdelbegin \DIFdel{We hypothesize that the lack of results on the PPIN network of SConES GI and LEAN are due to the same cause: the absence of joint association of a geneand a majority of its neighbors. Although in the case of SConES other hyperparameters could lead to a more informative solution (e.

g. a lower \(\lambda\) in Equation~\ref{eq:scones}), it is unclear what the best strategy to find themis. In addition, due to the design of the iCOGS array, the genome of GENESIS participants has not been unbiasedly surveyed: some regions are fine-mapped --- which might distort gene structure in GM and GI networks --- while others are under studied --- hindering the accuracy with which the GS network captures the genome structure}\DIFdelend \DIFaddbegin 

\subsection{\DIFadd{Different parameters produce similarly-sized solutions}}
\label{results:parameters}

\DIFadd{We explored methods' parameter space by running them under different combinations of parameters (Section~\ref{methods:parameters}). In agreement with their formulations (Section~\ref{methods:methods}), larger values of specific parameters produced less stringent solutions (}\nameref{sfig:params_size}\DIFadd{A): for HotNet2 and heinz, this is the threshold above which genes receive a positive score; for dmGWAS, it is the }\texttt{\DIFadd{d}} \DIFadd{parameter, which controls how far neighbors could be added; for SigMod, it is }\texttt{\DIFadd{nmax}}\DIFadd{, which specifies the maximum size of the solution; and for LEAN, it is the P-value threshold to consider a solution significant. Two parameters had the opposite effect (the larger, the more stringent): SigMod's }\texttt{\DIFadd{maxjump}}\DIFadd{, which sets the threshold to consider an increment in $\lambda$ ``large enough''; and SConES' $\eta$, where higher values produce smaller solutions. However, two of the parameters did not have the expected effect: dmGWAS' }\texttt{\DIFadd{r}}\DIFadd{, which controls the minimum increment in the score required to add a gene; and SigMod's }\texttt{\DIFadd{maxjump}}\DIFadd{, which sets the threshold to consider an increment in $\lambda$ ``large enough''. In both cases, the size of the solution was very similar across the different values. Despite the differences in size, the solutions' size was relatively robust to the choice of parameters (}\nameref{sfig:params_size}\DIFadd{B).

}

\DIFadd{We computed the Pearson correlation between the different solutions as in Section~\ref{methods:algorithm_comparison} to study how the parameters affected which genes and SNPs were selected (}\nameref{sfig:params_pearson}\DIFadd{). This analysis showed that dmGWAS and SigMod were robust to two parameters: the parameter }\texttt{\DIFadd{d}} \DIFadd{determined dmGWAS' output more than }\texttt{\DIFadd{r}}\DIFadd{; for SigMod, it was $nmax$ rather than $maxjump$}\DIFaddend .

\DIFaddbegin \DIFadd{SConES presented an interesting case in terms of feature selection: most of the explored combinations of parameters led to trivial solutions (they included either all the SNPs or none of them) (}\nameref{sfig:params_pearson}\DIFadd{). To explore a more meaningful parameter space, we selected the parameters in two rounds in our experiments. First, we explored the whole sample space. Then, we focused on a range of $\eta$ and $\lambda$ 1.5 orders of magnitude above and below the best parameters, respectively. This second parameter space was more diverse, which allowed to find more interesting solutions.

}

\DIFaddend \subsection{\DIFdelbegin \DIFdel{Adjusting for instability }\DIFdelend \DIFaddbegin \DIFadd{Building a stable consensus network }\DIFaddend preserves global network properties}
\label{results:stable-consensus}

Most \DIFdelbegin \DIFdel{of the }\DIFdelend network methods, including the consensus, were highly unstable \DIFaddbegin \DIFadd{(Fig~\ref{fig:benchmark}B)}\DIFaddend , raising questions about the \DIFdelbegin \DIFdel{reliability of the results}\DIFdelend \DIFaddbegin \DIFadd{results' reliability}\DIFaddend . We built a new, \DIFdelbegin \DIFdel{stable consensus network }\DIFdelend \DIFaddbegin \emph{\DIFadd{stable consensus}} \DIFadd{solution }\DIFaddend using the genes selected most often across the 30 solutions obtained by running the 6 methods on 5 different splits of the data (Section\DIFdelbegin \DIFdel{\ref{methods:comparison}}\DIFdelend \DIFaddbegin \DIFadd{~\ref{methods:benchmark}}\DIFaddend ). Such a network \DIFdelbegin \DIFdel{is expected to }\DIFdelend \DIFaddbegin \DIFadd{should }\DIFaddend capture the subnetworks more often found altered, and hence should be more resistant to noise. We used only genes selected in at least 7 \DIFdelbegin \DIFdel{of the }\DIFdelend solutions, which corresponded to 1\% of all genes selected at least once. The resulting \DIFdelbegin \DIFdel{stablility-based }\DIFdelend \DIFaddbegin \DIFadd{stability-based }\DIFaddend consensus was composed of 68 genes (Fig\DIFaddbegin \DIFadd{~}\DIFaddend \ref{fig:stable-consensus}). This network \DIFdelbegin \DIFdel{shares }\DIFdelend \DIFaddbegin \DIFadd{shared }\DIFaddend most of the \DIFdelbegin \DIFdel{proterties }\DIFdelend \DIFaddbegin \DIFadd{properties }\DIFaddend of the consensus: breast cancer susceptibility genes \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend overrepresented (P\=/value = 3 \texttimes{} 10\textsuperscript{-4}), as well as genes involved in mitochondrial translation and the attenuation phase (adjusted P\=/values 0.001 and 3 \texttimes{} 10\textsuperscript{-5} respectively); the selected genes \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend more central than average (P\=/value = 1.1 \texttimes{} 10\textsuperscript{-14}); and a considerable number of nodes (19) \DIFdelbegin \DIFdel{are isolated }\DIFdelend \DIFaddbegin \DIFadd{were isolated (Fig~\ref{fig:stable-consensus}B and C)}\DIFaddend .

\begin{figure}[!ht]
  \centering
  \DIFdelbeginFL %DIFDELCMD < \includegraphics[width=.7\linewidth]{./figures/figure_5.pdf}
%DIFDELCMD <   %%%
\DIFdelendFL \DIFaddbeginFL \includegraphics[width=.8\linewidth]{./figures/figure_5.pdf}
  \DIFaddendFL \caption{\DIFdelbeginFL %DIFDELCMD < {\bf %%%
\DIFdelFL{Stable }\DIFdelendFL  \DIFaddbeginFL \textbf{\DIFaddFL{Stable consensus solution on GENESIS (Section~\ref{results:stable-consensus})}}\DIFaddFL{. }\textbf{\DIFaddFL{(A)}}\DIFaddFL{~Manhattan plot of genes; in black, the ones in the stable }\DIFaddendFL consensus \DIFdelbeginFL \DIFdelFL{subnetwork on GENESIS }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{solution. The red line indicates the Bonferroni threshold }\DIFaddendFL (\DIFdelbeginFL \DIFdelFL{Section}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{1.53}\DIFaddendFL ~\DIFdelbeginFL \DIFdelFL{\ref{results:stable-consensus}}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{\texttimes{}~10\textsuperscript{-6} for genes}\DIFaddendFL ). \DIFdelbeginFL %DIFDELCMD < } %%%
\DIFdelendFL \DIFaddbeginFL \textbf{\DIFaddFL{(B)}}\DIFaddFL{~Stable consensus network. }\DIFaddendFL Each \DIFdelbeginFL \DIFdelFL{node }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{gene }\DIFaddendFL is represented by a pie chart, which shows the methods that selected it. We \DIFdelbeginFL \DIFdelFL{labeled (and }\DIFdelendFL enlarged \DIFdelbeginFL \DIFdelFL{) }\DIFdelendFL the most central \DIFdelbeginFL \DIFdelFL{genes }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{gene }\DIFaddendFL (\emph{CUL3})\DIFdelbeginFL \DIFdelFL{and those genes that are }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, the }\DIFaddendFL known breast cancer susceptibility genes\DIFaddbeginFL \DIFaddFL{, }\DIFaddendFL and \DIFdelbeginFL \DIFdelFL{/or significantly associated with breast cancer susceptibility in }\DIFdelendFL the \DIFdelbeginFL \DIFdelFL{BCAC dataset}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{BCAC-significant genes (Section~\ref{methods:bcac})}\DIFaddendFL . \DIFaddbeginFL \textbf{\DIFaddFL{(C)}}\DIFaddFL{~}\DIFaddendFL The \DIFdelbeginFL \DIFdelFL{latter ones }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{nodes }\DIFaddendFL are \DIFdelbeginFL \DIFdelFL{also }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{in the same disposition as in panel B, but we indicated every gene name. We }\DIFaddendFL colored in pink \DIFdelbeginFL \DIFdelFL{. All gene }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{the }\DIFaddendFL names \DIFdelbeginFL \DIFdelFL{are indicated in }%DIFDELCMD < \nameref{sfig:stable-consensus-names}%%%
\DIFdelendFL \DIFaddbeginFL \DIFaddFL{of known breast cancer susceptibility genes and BCAC-significant genes}\DIFaddendFL .}
  \label{fig:stable-consensus}
  \end{figure}

\DIFdelbegin \DIFdel{However, although this new network exhibits similar global properties as the previous one, the lack of stability results in different genesbeing selected. In this case}\DIFdelend \DIFaddbegin \DIFadd{Despite these similarities, the consensus and the stable consensus included different genes. In the stable consensus network}\DIFaddend , the most central gene \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend \emph{CUL3}, which \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend absent from the previous consensus \DIFdelbegin \DIFdel{network and has }\DIFdelend \DIFaddbegin \DIFadd{solution and had }\DIFaddend a low association score in both GENESIS and BCAC (P\=/values of 0.04 and 0.26, respectively). This gene \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend a component of Cullin-RING ubiquitin ligases. Encouragingly, it impacts the protein levels of multiple genes relevant for cancer progression \cite{Chen2016}, and its overexpression was also linked to increased sensitivity to carcinogens \cite{Loignon2009}.

\section{Discussion}

In recent years, the \DIFdelbegin \DIFdel{ability of GWAS}\DIFdelend \DIFaddbegin \DIFadd{GWAS' ability }\DIFaddend to unravel the mechanisms leading to complex diseases has been called into question \cite{boyle_expanded_2017}. \DIFdelbegin \DIFdel{On the one hand}\DIFdelend \DIFaddbegin \DIFadd{First}\DIFaddend , the omnigenic model proposes that gene functions are interwoven \DIFdelbegin \DIFdel{with each other }\DIFdelend in a dense co-function network. The practical consequence is that larger and larger GWAS will lead to \DIFdelbegin \DIFdel{the discovery of }\DIFdelend \DIFaddbegin \DIFadd{discovering }\DIFaddend an uninformative wide-spread pleiotropy. \DIFdelbegin \DIFdel{On the other hand, discovery in GWAS is hindered by a }\DIFdelend \DIFaddbegin \DIFadd{Second, its }\DIFaddend conservative statistical framework \DIFaddbegin \DIFadd{hinders GWAS discovery}\DIFaddend . Network methods \DIFdelbegin \DIFdel{tackle }\DIFdelend \DIFaddbegin \DIFadd{elegantly address }\DIFaddend these two issues by using both \DIFdelbegin \DIFdel{the association score }\DIFdelend \DIFaddbegin \DIFadd{association scores }\DIFaddend and an interaction network to \DIFdelbegin \DIFdel{take into consideration }\DIFdelend \DIFaddbegin \DIFadd{consider }\DIFaddend the biological context of each of the genes and SNPs. Based on what could be considered diverse interpretations of the omnigenic model, several methods for network-guided \DIFdelbegin \DIFdel{biomarker }\DIFdelend discovery have been proposed in recent years. In this article we evaluated \DIFdelbegin \DIFdel{the relevance of six of them by examining }\DIFdelend \DIFaddbegin \DIFadd{six of these methods (Section~\ref{methods:methods}) by applying them to the GENESIS study, }\DIFaddend a GWAS dataset on familial breast cancer \DIFaddbegin \DIFadd{(Section~\ref{methods:data})}\DIFaddend .

\DIFdelbegin \DIFdel{Most of the network methods produced a relevant subset of biomarkers, recapitulating }\DIFdelend \DIFaddbegin \DIFadd{DmGWAS, Heinz, HotNet2, SConES, and SigMod all yielded compelling solutions, which include (but are not limited to) }\DIFaddend known breast cancer susceptibility genes \DIFaddbegin \DIFadd{(Section~\ref{results:separate_networks})}\DIFaddend . In general, the selected genes and SNPs were more central than \DIFdelbegin \DIFdel{average, in accordance }\DIFdelend \DIFaddbegin \DIFadd{most other genes and SNPs, agreeing }\DIFaddend with the observation that disease genes are more \DIFdelbegin \DIFdel{relatively }\DIFdelend central \cite{pinero_uncovering_2016}. However, very central nodes are also more likely to be connecting any given random pair of nodes, making them more likely to be selected by \DIFdelbegin \DIFdel{these network methods . Across this article we show that }\DIFdelend \DIFaddbegin \DIFadd{network methods (Section~\ref{results:drawbacks}). However, we found support in the literature for the involvement of the selected }\DIFaddend highly central genes \DIFdelbegin \DIFdel{that were selected }\DIFdelend (\emph{COPS5}, \DIFdelbegin \emph{\DIFdel{CUL3}} %DIFAUXCMD
\DIFdel{and }\DIFdelend \emph{FN1}\DIFdelbegin \DIFdel{) could plausibly be involved }\DIFdelend \DIFaddbegin \DIFadd{, and }\emph{\DIFadd{CUL3}}\DIFadd{) }\DIFaddend in breast cancer susceptibility \DIFdelbegin \DIFdel{. Yet, further work is needed to characterize the impact of centrality on network methods' outputs}\DIFdelend \DIFaddbegin \DIFadd{(Sections~\ref{results:consensus}, \ref{results:drawbacks}, and \ref{results:stable-consensus})}\DIFaddend . Despite these similarities, the \DIFaddbegin \DIFadd{methods' }\DIFaddend solutions were notably different. At one end of the spectrum, SConES and heinz preferred \DIFdelbegin \DIFdel{small, highly associated solutions, providing a conveniently short list of biomarkers, at the expense of not shedding }\DIFdelend \DIFaddbegin \DIFadd{high scoring solutions, which were also small and hence did not shed }\DIFaddend much light on the \DIFdelbegin \DIFdel{etiology of the disease}\DIFdelend \DIFaddbegin \DIFadd{disease's etiology}\DIFaddend . On the other end, \DIFdelbegin \DIFdel{SigMod and dmGWASgravitate towards larger, less associated solutionswhich provide }\DIFdelend \DIFaddbegin \DIFadd{dmGWAS, HotNet2 and SigMod gravitated towards lower scoring but larger solutions, which provided }\DIFaddend a wide overview of the biological context. While this \DIFdelbegin \DIFdel{deepens }\DIFdelend \DIFaddbegin \DIFadd{deepened }\DIFaddend our understanding of \DIFdelbegin \DIFdel{the disease and provide }\DIFdelend \DIFaddbegin \DIFadd{breast cancer susceptibility and provided }\DIFaddend biological hypotheses, they \DIFdelbegin \DIFdel{require further analyses, which might deter unexperienced practitioners. HotNet2 balances both approaches at the expense of producing the largest solution: a constellation of many, highly associated, small subnetworks. Additionally}\DIFdelend \DIFaddbegin \DIFadd{required further analyses. For instance, we examined the centrality of the selected genes to understand how much that property was driving their selection (Section~\ref{results:drawbacks}). However}\DIFaddend , all solutions \DIFdelbegin \DIFdel{share }\DIFdelend \DIFaddbegin \DIFadd{shared }\DIFaddend two drawbacks. First, they \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend all equally bad at discriminating cases from controls. \DIFdelbegin \DIFdel{Yet}\DIFdelend \DIFaddbegin \DIFadd{However}\DIFaddend , the classification accuracy of network methods \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend similar to that of a \DIFdelbegin \DIFdel{machine learning classifier }\DIFdelend \DIFaddbegin \DIFadd{classifier trained on the entire genome (Section~\ref{results:benchmark})}\DIFaddend , which suggests that cases and controls are difficult to separate in \DIFdelbegin \DIFdel{this }\DIFdelend \DIFaddbegin \DIFadd{the GENESIS }\DIFaddend dataset. This \DIFdelbegin \DIFdel{might }\DIFdelend \DIFaddbegin \DIFadd{may }\DIFaddend be due to \DIFdelbegin \DIFdel{unaccounted for environmental factors, and }\DIFdelend limited statistical power, which reduces the ability to identify relevant SNPs. \DIFaddbegin \DIFadd{However, in any event, we do not expect to separate people who have or will develop cancer from others on the sole basis of their genomes, ignoring all environmental factors and chance events. Hence, network methods were preferable to the logistic regression classifier since they did ``no worse'' at classification while providing an interpretable solution. }\DIFaddend Second, all methods \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend remarkably unstable, yielding different solutions for slightly different inputs. This might partly \DIFdelbegin \DIFdel{be }\DIFdelend \DIFaddbegin \DIFadd{have been }\DIFaddend caused by the instability of the P\=/values themselves in low statistical power settings \cite{halseyFickleValueGenerates2015}. Hence, heinz's conservative transformation of P\=/values, which \DIFdelbegin \DIFdel{favors }\DIFdelend \DIFaddbegin \DIFadd{favored }\DIFaddend only the most extreme ones, \DIFdelbegin \DIFdel{leads }\DIFdelend \DIFaddbegin \DIFadd{led }\DIFaddend to improved stability. Another source of instability might \DIFdelbegin \DIFdel{be }\DIFdelend \DIFaddbegin \DIFadd{have been }\DIFaddend the redundancy inherent to biological networks\DIFdelbegin \DIFdel{. 

}\DIFdelend \DIFaddbegin \DIFadd{, a consequence of an evolutionary pressure to avoid single points of failure \mbox{%DIFAUXCMD
\cite{wagnerAlternativeRoutesMutational2007}}\hspace{0pt}%DIFAUXCMD
. Hence, biological networks will often have multiple paths connecting two high-scoring nodes. 

}\DIFaddend 

To overcome \DIFdelbegin \DIFdel{the limitations of the individual methods while exploiting their }\DIFdelend \DIFaddbegin \DIFadd{these limitations while exploiting the each method's }\DIFaddend strengths, we proposed combining them into a consensus \DIFdelbegin \DIFdel{subnetwork. We use a }\DIFdelend \DIFaddbegin \DIFadd{solution. We used the }\DIFaddend straightforward strategy of including any node that was recovered by \DIFdelbegin \DIFdel{multiple }\DIFdelend \DIFaddbegin \DIFadd{at least two }\DIFaddend methods. We \DIFdelbegin \DIFdel{proposed two networks}\DIFdelend \DIFaddbegin \DIFadd{thus proposed two solutions (Sections~\ref{results:consensus} and \ref{results:stable-consensus})}\DIFaddend : a consensus \DIFdelbegin \DIFdel{network that tackled }\DIFdelend \DIFaddbegin \DIFadd{solution, which addressed }\DIFaddend the heterogeneity of the solutions\DIFdelbegin \DIFdel{in the full dataset}\DIFdelend , and a stable consensus \DIFdelbegin \DIFdel{network, that addressed }\DIFdelend \DIFaddbegin \DIFadd{solution, which addresseded }\DIFaddend the instability of the methods. They both \DIFdelbegin \DIFdel{synthesized the altered mechanisms: they both }\DIFdelend included the majority of the strongly associated smaller solutions and captured genes and broader mechanisms related to cancer\DIFaddbegin \DIFadd{, thus synthesizing the mechanisms altered in breast cancer cases}\DIFaddend . Thanks to their smaller size and \DIFdelbegin \DIFdel{their }\DIFdelend network structure, they provided compelling hypotheses on genes like \emph{COPS5} and \emph{CUL3}, which lack genome-wide association with the disease \DIFdelbegin \DIFdel{, but who }\DIFdelend \DIFaddbegin \DIFadd{but }\DIFaddend are related to cancer at the expression level and \DIFdelbegin \DIFdel{whose neighborhood has consistent high association scores. Crucially, the consensus }\DIFdelend \DIFaddbegin \DIFadd{consistently interact with high scoring genes. Notably, while the consensus approach }\DIFaddend was as unstable as the \DIFdelbegin \DIFdel{tested methods, while }\DIFdelend \DIFaddbegin \DIFadd{individual network-guided methods, }\DIFaddend the stable consensus \DIFdelbegin \DIFdel{shared these properties while accounting for instability. This supports that instability might be caused by }\DIFdelend \DIFaddbegin \DIFadd{network retained the ability to provide compelling hypotheses and had better stability. This supported that }\DIFaddend redundant but equivalent \DIFdelbegin \DIFdel{biological mechanisms , and hence validates }\DIFdelend \DIFaddbegin \DIFadd{mechanisms might cause instability and supported }\DIFaddend the conclusions obtained on the individual solutions\DIFdelbegin \DIFdel{and the consensus. }\DIFdelend \DIFaddbegin \DIFadd{.

}\DIFaddend 

\DIFaddbegin \DIFadd{In this work, we have compared our results to significant genes and SNPs in the BCAC study \mbox{%DIFAUXCMD
\cite{Michailidou2017}}\hspace{0pt}%DIFAUXCMD
. Network methods showed modest precision but much higher recall at recovering BCAC hits (Section~\ref{results:boost}). While precision might be desirable when a subset of useful markers is required (for instance, for diagnosis), higher recall is desirable in exploratory settings. Nonetheless, BCAC was not an ideal ground truth. First, the studied populations are not entirely overlapping: BCAC focused on a pan-European cohort, while GENESIS targeted the French population. Second, the study designs differed: a high proportion of breast cancer cases investigated in BCAC were sporadic (not selected according to family history), while GENESIS was a homogeneous dataset not included in BCAC focused on the French high-risk population attending the family cancer clinics. Finally, and this is indeed the motivation for this study, GWAS are unlikely to identify all genes relevant for the disease: some might only show up in rare-variant studies; others might have too small effect sizes. Network methods account for this by including genes with low association scores but with relevant topological properties. Hence, network methods and GWAS, even when well-powered, are unlikely to capture exactly the same sets of genes. This might partly excuse the low precisions displayed in Section~\ref{results:boost} and the low AUC displayed in Section~\ref{results:benchmark}.

}

\DIFadd{As not all PPIN databases compiled the same interactions, the choice of the PPIN determines the final output. In this work, we used only interactions from HINT from high-throughput experiments (Section~\ref{methods:networks}). This responded to concerns about adding interactions identified in targeted studies and falling into ``rich getting richer'' problems: since popular genes have a higher proportion of their interactions described \mbox{%DIFAUXCMD
\cite{cai_broker_2010,das_hint:_2012}}\hspace{0pt}%DIFAUXCMD
, they might bias discovery towards themselves by reducing the average shortest path length between two random nodes. On the other hand, Huang et al. \mbox{%DIFAUXCMD
\cite{huang_systematic_2018} }\hspace{0pt}%DIFAUXCMD
found that larger networks were more useful than smaller networks to identify disease genes. This would support using the largest networks in our experiments. However, when we compared the impact of using a larger PPIN containing interactions from both high-throughput experiments and the literature (Section~\ref{methods:networks}), for most of the methods it did not change much the size or the stability of the solution, the classification accuracy, or the runtime (}\nameref{sfig:lc_ht_comparison}\DIFadd{). This supports using only interactions from high-throughput experiments, which produced similar solutions and avoided falling into ``circular reasonings'', where the best-known genes were artificially pushed into the solutions, as we observed in Section~\ref{results:drawbacks}. 

}

\DIFaddend The strength of network-based analyses comes from leveraging prior knowledge to boost discovery. In consequence, they show their shortcomings \DIFdelbegin \DIFdel{with respect to }\DIFdelend \DIFaddbegin \DIFadd{on }\DIFaddend understudied genes, especially those not in the network. Out of the 32\,767 genes \DIFdelbegin \DIFdel{that we can map }\DIFdelend \DIFaddbegin \DIFadd{to which we mapped }\DIFaddend the genotyped SNPs\DIFdelbegin \DIFdel{to}\DIFdelend , 60.7\% (19\,887) \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend not in the \DIFdelbegin \DIFdel{protein-protein interaction network}\DIFdelend \DIFaddbegin \DIFadd{PPIN}\DIFaddend . The majority of those (14\,660) are non-coding genes, mainly lncRNA, miRNA, and snRNA (\nameref{sfig:biotypes_excluded}). \DIFdelbegin \DIFdel{Yet}\DIFdelend \DIFaddbegin \DIFadd{Nevertheless}\DIFaddend , RNA genes like \emph{CASC16} \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend associated to breast cancer (Section~\ref{results:conventional}), reminding us of the importance of using networks beyond coding genes. \DIFdelbegin \DIFdel{In addition}\DIFdelend \DIFaddbegin \DIFadd{Besides}\DIFaddend , even protein-coding genes linked to breast cancer susceptibility \cite{ahmed_newly_2009}, like \emph{NEK10} (P\=/value 1.6 \texttimes{} 10\textsuperscript{-5}, \DIFdelbegin \DIFdel{located near }\DIFdelend \DIFaddbegin \DIFadd{overlapping with }\DIFaddend \emph{SLC4A7}) or \emph{POU5F1B}, were absent from the \DIFdelbegin \DIFdel{network}\DIFdelend \DIFaddbegin \DIFadd{PPIN}\DIFaddend . However, on average protein-coding genes absent from the PPIN \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend less associated with \DIFdelbegin \DIFdel{this phenotype }\DIFdelend \DIFaddbegin \DIFadd{breast cancer susceptibility }\DIFaddend (Wilcoxon rank-sum P\=/value~=~2.79 \texttimes{} 10\textsuperscript{-8}, median P\=/values of 0.43 and 0.47). \DIFdelbegin \DIFdel{As we are using interactions from high-throughput experiments, such difference cannot }\DIFdelend \DIFaddbegin \DIFadd{This could not }\DIFaddend be due to well-known genes having more known interactions \DIFaddbegin \DIFadd{because we only used interactions from high-throughput experiments}\DIFaddend . As disease genes tend to be more central \cite{pinero_uncovering_2016}, we hypothesize that it \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend due to interactions between central genes being more likely. It is worth noting that network \DIFdelbegin \DIFdel{approaches }\DIFdelend \DIFaddbegin \DIFadd{methods }\DIFaddend that do not use PPIs, like SConES GS and GM, \DIFdelbegin \DIFdel{did recover }\DIFdelend \DIFaddbegin \DIFadd{recovered }\DIFaddend SNPs in \emph{NEK10} and \emph{CASC16}. Moreover, both SConES GM and GI recovered intergenic regions, which might contain key regulatory elements \cite{gallagherPostGWASEraAssociation2018}\DIFdelbegin \DIFdel{and, yet, }\DIFdelend \DIFaddbegin \DIFadd{, but }\DIFaddend are excluded from gene-centric approaches. This shows the potential of SNP networks, in which SNPs are linked when there is evidence of co-function, to perform network-guided GWAS even in the absence of gene-level interactions. Lastly, all the methods are heavily affected by how SNPs are mapped to genes\DIFdelbegin \DIFdel{. In Section~\ref{results:consensus} we highlight ambiguities that appear when genes overlap or are in linkage disequilibrium. In fact, the presented case is paradigmatic, since the genes are in the most gene-dense region of the genome \mbox{%DIFAUXCMD
\cite{Xie2003}}\hspace{0pt}%DIFAUXCMD
. Network methods are prone to selecting such genes when they are functionally related, and hence interconnected in the network, but might be more resilient to them when the overlapping genes are unrelated. Making use of more targeted mappings of SNPs to genes }\DIFdelend \DIFaddbegin \DIFadd{, and other strategies }\DIFaddend (e.g.\DIFaddbegin \DIFadd{, }\DIFaddend eQTLs, SNPs associated to \DIFdelbegin \DIFdel{the expression of a gene ), altogether with a stringent LD pruning, might address such problems.

}\DIFdelend \DIFaddbegin \DIFadd{gene expression) might lead to different results. 

}\DIFaddend 

\DIFdelbegin \DIFdel{As not all databases compile the same interactions, the choice of the PPIN determines the final output. In this work we used exclusively interactions from HINT from high-throughput experiments. This responds to concerns about adding interactions identified in targeted studies and prone to a ``rich getting richer'' phenomenon: popular genes have a higher proportion of their interactions described \mbox{%DIFAUXCMD
\cite{cai_broker_2010,das_hint:_2012}}\hspace{0pt}%DIFAUXCMD
, and they might bias discovery by reducing the average shortest path length between two random nodes. 

On the other hand, Huang et al. \mbox{%DIFAUXCMD
\cite{huang_systematic_2018} }\hspace{0pt}%DIFAUXCMD
found that the best predictor of the performance of a network for disease gene discovery is the size of the network, which supports using the largest amount of interactions. When we compared the impact of using a larger network containing interactions from both high-throughput experiments and the literature (Section~\ref{methods:networks}), we found that for most of the methods it did not greatly change the size or the stability of the solution, the classification accuracy, or the runtime (}%DIFDELCMD < \nameref{sfig:lc_ht_comparison}%%%
\DIFdel{). This supports using only interactions from high-throughput experiments, which produces apparently similar solutions and avoids falling into ``circular reasonings'', where the best known genes are artificially pushed into the solutions. 

}%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend A crucial step for the \DIFdelbegin \DIFdel{gene based }\DIFdelend \DIFaddbegin \DIFadd{gene-based }\DIFaddend methods is the computation of \DIFdelbegin \DIFdel{the gene score}\DIFdelend \DIFaddbegin \DIFadd{gene scores}\DIFaddend . In this work\DIFaddbegin \DIFadd{, }\DIFaddend we used VEGAS2 \cite{mishra_vegas2:_2015} due to the flexibility it offers to use user-specified gene annotations. However, it presents known problems\DIFdelbegin \DIFdel{(}\DIFdelend \DIFaddbegin \DIFadd{: }\DIFaddend selection of an appropriate percentage of top SNPs, long runtimes and P\=/value precision limited to the number of permutations \cite{nakka_gene_2016})\DIFdelbegin \DIFdel{, and other algorithms \mbox{%DIFAUXCMD
\cite{nakka_gene_2016,ionita-laza_sequence_2013,wang_combat:_2017} }\hspace{0pt}%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{. Additionally, other algorithms like PEGASUS \mbox{%DIFAUXCMD
\cite{nakka_gene_2016}}\hspace{0pt}%DIFAUXCMD
, SKAT \mbox{%DIFAUXCMD
\cite{ionita-laza_sequence_2013} }\hspace{0pt}%DIFAUXCMD
or COMBAT \mbox{%DIFAUXCMD
\cite{wang_combat:_2017} }\hspace{0pt}%DIFAUXCMD
}\DIFaddend might have more statistical power.

\DIFdelbegin \DIFdel{Another important decision is how to handle LD in a GWAS. }\DIFdelend \DIFaddbegin 

\DIFadd{How to handle linkage disequilibrium (LD) is often a concern among GWAS practitioners. Often, the question is whether an LD-based pruning of the genotypes will improve the results. }\DIFaddend VEGAS2 accounts for LD patterns, and hence an LD pruning step would not impact gene-based network methods, although it would speed up VEGAS2's computation time. \DIFaddbegin \DIFadd{In Section~\ref{results:consensus} we highlighted ambiguities that appear when genes overlap or are in LD. The presented case is paradigmatic since all three genes are in the HLA region, the most gene-dense region of the genome \mbox{%DIFAUXCMD
\cite{Xie2003}}\hspace{0pt}%DIFAUXCMD
. Network methods are prone to selecting such genes when they are functionally related, and hence interconnected in the PPIN. But the opposite case is also true: when genes are not functionally related (and hence disconnected in the PPIN), network methods might disregard them even if they have high association scores. }\DIFaddend With regards to SConES, fewer SNPs would lead to simpler SNP networks and, possibly, shorter runtimes. However, \DIFdelbegin \DIFdel{as mentioned in }\DIFdelend \DIFaddbegin \DIFadd{LD patterns also affect SConES' in other ways, since its formulation penalizes selecting a SNP and not its neighbors, via a nonzero parameter $\eta$ in Eq~\ref{eq:scones}. Due to LD, nearby SNPs' P-values correlate; since positional information determines SNP networks, nearby SNPs are likely to be connected. Hence, SConES tends to select LD-blocks formed by low P-value SNPs. This might explain why SConES produced similar results on the GS and GM networks, heavily affected by LD (}\DIFaddend Section~\ref{results:drawbacks}\DIFdelbegin \DIFdel{, LD patterns seem paramount to SConES' solutions, and an LD pruning step could potentially alter them}\DIFdelend \DIFaddbegin \DIFadd{). However, this same behavior raises the burden of proof required to select SNPs with many interactions, like those mapped to hub genes in the PPIN. For this reason, SConES GI did not select any protein coding gene. This could be caused by the absence of joint association of a gene and most of its neighbors, a hypothesis supported by LEAN's lack of results. Yet, a different combination of parameters could lead to a more informative SConES' solution (e.g., a lower \(\lambda\) in Eq~\ref{eq:scones}), although it is unclear how to find it. In addition, due to the design of the iCOGS array (Section~\ref{methods:data}), the genome of GENESIS participants has not been unbiasedly surveyed: some regions are fine-mapped --- which might distort gene structure in GM and GI networks --- while others are understudied --- hindering the accuracy with which the GS network captures the genome structure. A strong LD pruning might address such problems}\DIFaddend . 

\DIFdelbegin \DIFdel{In order to produce the consensus networks}\DIFdelend \DIFaddbegin \DIFadd{To produce the two consensus solutions}\DIFaddend , we faced \DIFdelbegin \DIFdel{the different }\DIFdelend \DIFaddbegin \DIFadd{practical challenges due to the differences in }\DIFaddend interfaces, preprocessing steps, and unexpected behaviors of the various methods. To \DIFdelbegin \DIFdel{facilitate that other authors apply them }\DIFdelend \DIFaddbegin \DIFadd{make it easier for others to apply these methods }\DIFaddend to new datasets and aggregate their solutions, we built six \DIFdelbegin \DIFdel{nextflow }\DIFdelend \DIFaddbegin \texttt{\DIFadd{nextflow}} \DIFaddend pipelines \cite{di_tommaso_nextflow_2017} with a consistent interface and, whenever possible, parallelized computation. They are available on GitHub: \DIFdelbegin %DIFDELCMD < \url{https://github.com/hclimente/gwas-tools}%%%
\DIFdelend \DIFaddbegin \href{https://github.com/hclimente/gwas-tools}{\DIFadd{hclimente/gwas-tools}} \DIFadd{(Section~\ref{methods:code})}\DIFaddend . Importantly, \DIFdelbegin \DIFdel{those methods that had }\DIFdelend \DIFaddbegin \DIFadd{we compiled those methods with }\DIFaddend a permissive license \DIFdelbegin \DIFdel{were compiled }\DIFdelend into a Docker image for easier use, \DIFdelbegin \DIFdel{which is }\DIFdelend available on Docker Hub \href{https://hub.docker.com/r/hclimente/gwas-tools}{hclimente/gwas-tools}.

\section{Materials and methods}

\subsection{GENESIS \DIFaddbegin \DIFadd{dataset, preprocessing, and quality control}\DIFaddend }
\DIFaddbegin \label{methods:data}

\DIFaddend 

The GENE Sisters (GENESIS) study \DIFdelbegin \DIFdel{was designed to investigate }\DIFdelend \DIFaddbegin \DIFadd{investigated }\DIFaddend risk factors for familial breast cancer in the French population \cite{sinilnikova_genesis:_2016}. Index cases \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend patients with infiltrating mammary or ductal adenocarcinoma, who had a sister with breast cancer, and \DIFdelbegin \DIFdel{who have been }\DIFdelend tested negative for \emph{BRCA1} and \emph{BRCA2} pathogenic variants. Controls \DIFdelbegin \DIFdel{are unaffected colleagues and/}\DIFdelend \DIFaddbegin \DIFadd{were unaffected colleagues }\DIFaddend or friends of the cases \DIFdelbegin \DIFdel{, }\DIFdelend born around the year of birth of their corresponding case (\textpm{} 3 years). We focused on the 2\,577 samples of European ancestry, of which 1\,279 \DIFdelbegin \DIFdel{are controls}\DIFdelend \DIFaddbegin \DIFadd{were controls, }\DIFaddend and 1\,298 \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend cases. The genotyping \DIFdelbegin \DIFdel{was performed using }\DIFdelend \DIFaddbegin \DIFadd{platform was }\DIFaddend the iCOGS array, a custom Illumina array designed to study \DIFdelbegin \DIFdel{genetic susceptibility of }\DIFdelend \DIFaddbegin \DIFadd{the genetic susceptibility to }\DIFaddend hormone-related cancers \cite{sakoda_turning_2013}. It \DIFdelbegin \DIFdel{contains }\DIFdelend \DIFaddbegin \DIFadd{contained }\DIFaddend 211\,155 SNPs, including SNPs putatively associated with breast, ovarian, and prostate cancers, SNPs associated with survival after diagnosis, and SNPs associated to other cancer-related traits, as well as candidate functional variants in selected genes and pathways.

\DIFdelbegin \subsection{\DIFdel{Preprocessing and quality control}}

%DIFAUXCMD
\addtocounter{subsection}{-1}%DIFAUXCMD
%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend We discarded SNPs with a minor allele frequency lower than 0.1\%, those not in Hardy--Weinberg equilibrium in controls (P\=/value \textless~0.001), and those with genotyping data missing on more than 10\% of the samples. \DIFdelbegin \DIFdel{A }\DIFdelend \DIFaddbegin \DIFadd{We also removed a }\DIFaddend subset of 20 duplicated SNPs in \emph{FGFR2}\DIFdelbegin \DIFdel{were also removed. In addition, we removed }\DIFdelend \DIFaddbegin \DIFadd{. We excluded }\DIFaddend the samples with more than 10\% missing genotypes. After controlling for relatedness, \DIFaddbegin \DIFadd{we excluded }\DIFaddend 17 additional samples \DIFdelbegin \DIFdel{were removed }\DIFdelend (6 for sample identity error, 6 controls related to other samples, 2 cases \DIFdelbegin \DIFdel{being }\DIFdelend related to an index case, and 3 additional controls having a high relatedness score). Lastly, based on study selection criteria, 11 other samples were removed (1 control having cancer, 4 index cases with no affected sister, 3 half-sisters, 1 sister with lobular carcinoma \emph{in situ}, 1 with a \emph{BRCA1} or \emph{BRCA2} pathogenic variant detected in the family, 1 with unknown molecular diagnosis). The final dataset included 1\,271 controls and 1\,280 cases, genotyped over 197\,083 SNPs. 

We looked for population structure that could produce spurious associations. A principal component analysis revealed no visual differential population structure between cases and controls (\nameref{sfig:pcs}). Independently, we did not find evidence of genomic inflation (\(\lambda\)~=~1.05) either, further confirming the absence of confounding population structure.

\subsection{\DIFdelbegin \DIFdel{High-score subnetwork search algorithms}\DIFdelend \DIFaddbegin \DIFadd{SNP- and gene-based GWAS}\DIFaddend }
\DIFdelbegin \subsubsection{\DIFdel{SNP and gene association}}
%DIFAUXCMD
\addtocounter{subsubsection}{-1}%DIFAUXCMD
%DIFDELCMD < \label{methods:node_score}
%DIFDELCMD < %%%
\DIFdel{To measure association between a genotype and the phenotype}\DIFdelend \DIFaddbegin \label{methods:conventional}

\DIFadd{To measure the association between genotype and susceptibility to breast cancer}\DIFaddend , we performed a per-SNP 1~d.f.~\DIFdelbegin \DIFdel{\(\chi\)\textsuperscript{2} }\DIFdelend \DIFaddbegin \DIFadd{$\chi^2$ }\DIFaddend allelic test using PLINK v1.90 \cite{chang_second-generation_2015}. \DIFaddbegin \DIFadd{To obtain significant SNPs, we performed a Bonferroni correction to keep the family-wise error rate below 5\%. The threshold used was $\frac{0.05}{197083} = 2.54 \times 10^{-7}$.

}

\DIFaddend Then, we used VEGAS2 \cite{mishra_vegas2:_2015} to compute the gene-level association score from the P\=/values of the SNPs mapped to them. \DIFdelbegin \DIFdel{Specifically, for each gene we only used the 10\% of SNPs mapped to it with lowest P\=/values. We mapped }\DIFdelend \DIFaddbegin \DIFadd{More specifically, we mapped }\DIFaddend SNPs to genes through their genomic coordinates: all SNPs located within the boundaries of a gene, \textpm 50 kb, were mapped to that gene. We \DIFaddbegin \DIFadd{computed VEGAS2 scores for each gene using only the 10\% of SNPs with the lowest P\=/values among all those mapped to it. We }\DIFaddend used the 62\,193 genes described in GENCODE 31 \cite{frankish_gencode_2019}, although only 54\,612 \DIFdelbegin \DIFdel{could be }\DIFdelend mapped to at least one SNP. Out of those, we focused exclusively on the 32\,767 that had a gene symbol. Out of the 197\,083 SNPs remaining after quality control, 164\,037 \DIFdelbegin \DIFdel{were }\DIFdelend mapped to at least one of these genes. \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdel{We used such mapping to compare the outputs of methods that produce SNP-lists to those that produce gene-lists, and vice versa. For the former, we considered any gene that can be mapped to any of the selected SNPs as selected as well. 

For the latter, we considered all the SNPs that can be mapped to that gene as selected by the method.

}\DIFdelend \DIFaddbegin \DIFadd{We also performed a Bonferroni correction to obtain significant genes; in this case, the threshold of significance was $\frac{0.05}{32767} = 1.53 \times 10^{-6}$. 

}\DIFaddend 

\DIFaddbegin \subsection{\DIFadd{Network methods}}

\DIFaddend \subsubsection{Mathematical notations}
\label{methods:notation}
In this article, we \DIFdelbegin \DIFdel{use }\DIFdelend \DIFaddbegin \DIFadd{used }\DIFaddend undirected, vertex-weighted networks, or graphs, $G = (V,E,w)$. $V = \{v_{1}, \dots{}, v_{n}\}$ refers to the vertices, with weights $w: V \rightarrow \mathbb{R}$. Equivalently, $E \subseteq \{\{x,y\} | x,y \in V \wedge x \neq y\}$ refers to the edges. When referring to a subnetwork $S$, $V_{S}$ is the set of nodes in $S$ and $E_{S}$ is the set of edges in $S$. A special case of subgraphs are \emph{connected} subgraphs, which occur when every node in the subgraph can be reached from any other node.

\DIFdelbegin \DIFdel{In addition to a weight, nodes have other properties , }\DIFdelend \DIFaddbegin \DIFadd{Nodes can be described by properties }\DIFaddend provided by the topology of the graph. \DIFdelbegin \DIFdel{In this article we focus }\DIFdelend \DIFaddbegin \DIFadd{We focused }\DIFaddend on two of those: degree centrality \DIFdelbegin \DIFdel{, }\DIFdelend and betweenness centrality. The degree centrality, or \DIFdelbegin \DIFdel{degree}\DIFdelend \DIFaddbegin \DIFadd{simply emph}{\DIFadd{degree}}\DIFaddend , is the number of edges that a node has. The betweenness centrality, or \DIFdelbegin \DIFdel{betweenness}\DIFdelend \DIFaddbegin \emph{\DIFadd{betweenness}}\DIFaddend , is the number of times a node participates in \DIFdelbegin \DIFdel{the }\DIFdelend shortest paths between two other nodes.

\DIFdelbegin \DIFdel{In addition, we use }\DIFdelend \DIFaddbegin \DIFadd{We also used }\DIFaddend two matrices that describe \DIFaddbegin \DIFadd{two }\DIFaddend different properties of a graph. \DIFdelbegin \DIFdel{The described }\DIFdelend \DIFaddbegin \DIFadd{Both }\DIFaddend matrices are square \DIFdelbegin \DIFdel{, }\DIFdelend and have as many rows and columns as nodes are in the network. The element $(i,j)$ \DIFdelbegin \DIFdel{represents a  selected }\DIFdelend \DIFaddbegin \DIFadd{hence represents a }\DIFaddend relationship between $v_i$ and $v_j$. The \emph{adjacency matrix} $W_G$ contains a 1 when the corresponding nodes are connected, and 0 otherwise; \DIFdelbegin \DIFdel{the }\DIFdelend \DIFaddbegin \DIFadd{its }\DIFaddend diagonal is zero. The \emph{degree matrix} $D_G$ is a diagonal matrix \DIFdelbegin \DIFdel{which }\DIFdelend \DIFaddbegin \DIFadd{that }\DIFaddend contains the degree of the different nodes.

\subsubsection{Networks}
\label{methods:networks}

\paragraph{Gene network}
The \DIFdelbegin \DIFdel{statistical frameworks }\DIFdelend \DIFaddbegin \DIFadd{mathematical formulations }\DIFaddend of the different network methods are compatible with any type of \DIFdelbegin \DIFdel{network (protein interactions, gene coexpression, regulatory, etc.). Yet}\DIFdelend \DIFaddbegin \DIFadd{biological network (e.g., from protein interactions or gene co-expression). Here}\DIFaddend , we used protein-protein interaction networks (PPIN) for all \DIFdelbegin \DIFdel{of them }\DIFdelend \DIFaddbegin \DIFadd{network methods }\DIFaddend except SConES, as \DIFdelbegin \DIFdel{they }\DIFdelend \DIFaddbegin \DIFadd{PPINs }\DIFaddend are interpretable, \DIFdelbegin \DIFdel{well characterized, and they }\DIFdelend \DIFaddbegin \DIFadd{well-characterized, and the methods }\DIFaddend were designed to run efficiently on \DIFdelbegin \DIFdel{networks of their size}\DIFdelend \DIFaddbegin \DIFadd{them}\DIFaddend . We built our PPIN from both binary and co-complex interactions stored in the HINT database (release April 2019) \cite{das_hint:_2012}. Unless otherwise specified, we used only interactions coming from high-throughput experiments, leaving out targeted studies that might bias the topology of the \DIFdelbegin \DIFdel{network}\DIFdelend \DIFaddbegin \DIFadd{PPIN}\DIFaddend . Out of the 146\,722 interactions from high-throughput experiments that HINT stores, we \DIFdelbegin \DIFdel{were able to }\DIFdelend \DIFaddbegin \DIFadd{could }\DIFaddend map 142\,541 to a pair of gene symbols, involving 13\,619 genes. 12\,880 of those mapped to a genotyped SNP after quality control, involving 127\,604 interactions. The scoring function for the nodes changed from method to method (Section~\ref{methods:methods}).

Additionally, we compared the results \DIFdelbegin \DIFdel{of the aforementioned }\DIFdelend \DIFaddbegin \DIFadd{obtained on this }\DIFaddend PPIN with those obtained on \DIFdelbegin \DIFdel{another }\DIFdelend \DIFaddbegin \DIFadd{a }\DIFaddend PPIN built using interactions coming from both high-throughput and targeted studies. In that case, out of the 179\,332 interactions in HINT, 173\,797 mapped to a pair of gene symbols. Out of those, 13\,735 mapped to a genotyped SNP after quality control, involving 156\,190 interactions.

\paragraph{SNP networks}
SConES \cite{azencott_efficient_2013} \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend the only network method designed to handle SNP networks. As in gene networks, two SNPs \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{were }\DIFaddend connected in a SNP network when there \DIFdelbegin \DIFdel{is }\DIFdelend \DIFaddbegin \DIFadd{was }\DIFaddend evidence of shared functionality between \DIFdelbegin \DIFdel{two SNPs}\DIFdelend \DIFaddbegin \DIFadd{them}\DIFaddend . Azencott et al. \cite{azencott_efficient_2013} proposed three ways of building \DIFdelbegin \DIFdel{these }\DIFdelend \DIFaddbegin \DIFadd{such }\DIFaddend networks: connecting the SNPs consecutive in the genomic sequence (``GS network''); interconnecting all the SNPs mapped to the same gene, on top of GS (``GM network''); and interconnecting all SNPs mapped to two genes for which a protein-protein interaction exists, on top of GM (``GI network''). We focused on the GI network \DIFaddbegin \DIFadd{using the PPIN described above}\DIFaddend , as it \DIFdelbegin \DIFdel{fits }\DIFdelend \DIFaddbegin \DIFadd{fitted }\DIFaddend the scope of this work better\DIFdelbegin \DIFdel{, using the PPIN described above}\DIFdelend . However, at different stages\DIFdelbegin \DIFdel{of this work we also used }\DIFdelend \DIFaddbegin \DIFadd{, we also compared GI to }\DIFaddend GS and GM \DIFdelbegin \DIFdel{for comparison}\DIFdelend \DIFaddbegin \DIFadd{to understand how including the PPIN affects SConES' output}\DIFaddend . For the GM network, we used the mapping described in Section~\DIFdelbegin \DIFdel{\ref{methods:node_score}}\DIFdelend \DIFaddbegin \DIFadd{\ref{methods:snp2gene}}\DIFaddend . In all three\DIFdelbegin \DIFdel{the node scores are the association scores of the individual SNPs with the phenotype (}\DIFdelend \DIFaddbegin \DIFadd{, we scored the nodes using the }\DIFaddend 1~d.f.~\(\chi\)\textsuperscript{2} \DIFdelbegin \DIFdel{)}\DIFdelend \DIFaddbegin \DIFadd{statistic of association}\DIFaddend . The properties of these three subnetworks are available in \nameref{stab:snp_solutions}.

\subsubsection{\DIFdelbegin \DIFdel{Network methods}\DIFdelend \DIFaddbegin \DIFadd{High-score subnetwork search algorithms}\DIFaddend }
\label{methods:methods}

Genes that contribute to the same function are nearby in the PPIN \DIFdelbegin \DIFdel{, }\DIFdelend and can be topologically related to each other in diverse ways (densely interconnected modules, nodes around a hub, a path, etc.). \DIFdelbegin \DIFdel{But this is not the only aspect to model }\DIFdelend \DIFaddbegin \DIFadd{Several aspects have to be considered }\DIFaddend when developing a network method: how to score the nodes, whether the affected mechanisms form a single connected component or several, how to frame the problem in a computationally efficient fashion, which network to use, etc. Unsurprisingly, multiple solutions have been proposed. We examined six of them: five that explore the PPIN, and one which explores SNP networks. We selected \DIFdelbegin \DIFdel{methods that were open source, }\DIFdelend \DIFaddbegin \DIFadd{open-source methods that }\DIFaddend had an implementation available \DIFdelbegin \DIFdel{, and an }\DIFdelend \DIFaddbegin \DIFadd{and }\DIFaddend accessible documentation. \DIFdelbegin \DIFdel{Their main differences are summarized }\DIFdelend \DIFaddbegin \DIFadd{We summarize their main differences }\DIFaddend in Table~\ref{tab:method_comparison}. \DIFaddbegin \DIFadd{We scored both SNPs and genes with the P-values (or transformations) computed in Section~\ref{methods:conventional}.

}\DIFaddend 

\begin{table}[!ht]
  \begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
  \centering
  \caption{
  {\bf Summary of the differences between the network methods.}}
  \begin{tabular}{l|llllllll}
  {\bf Method} & {\bf Field } & {\bf Nodes} & {\bf Exhaustive} & {\bf Solution} & {\bf Comp.} & {\bf Input} & {\bf Scoring} & {\bf Ref.}\\ \thickhline
  dmGWAS & GWAS & Genes & No & - & 1 & Summary & -log\textsubscript{10}(P) & \cite{jia_dmgwas:_2011}\\
  heinz & Omics & Genes & Yes & - & 1 & Summary & BUM & \cite{dittrich_identifying_2008}\\
  HotNet2 & Omics & Genes & Yes & Module & \(\ge\) 1 & Summary & Local FDR & \cite{leiserson_pan-cancer_2015}\\
  LEAN & Omics & Genes & Yes & Star & \(\ge\) 1 & Summary & -log\textsubscript{10}(P) & \cite{gwinner_network-based_2016}\\
  SConES & GWAS & SNPs & Yes & Module & \(\ge\) 1 & Genotypes & 1 d.f. \(\chi\)\textsuperscript{2} & \cite{azencott_efficient_2013}\\
  SigMod & GWAS & Genes & Yes & Module & \(\ge\) 1 & Summary & $\Phi$\textsuperscript{-1}(1 - P) & \cite{liu_sigmod:_2017}\\
  \end{tabular}
  \begin{flushleft} \textbf{\emph{Field}}: field in which the algorithm was developed. \textbf{\emph{Nodes}}: the type of nodes in the network, either genes (PPIN) or SNPs. \textbf{\emph{Exhaustive}}: whether \DIFaddbeginFL \DIFaddFL{the method explores }\DIFaddendFL all the possible solutions given the selected \DIFdelbeginFL \DIFdelFL{hyperparameters are explored}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{parameters}\DIFaddendFL . \textbf{\emph{Solution}}: additional properties \DIFdelbeginFL \DIFdelFL{are }\DIFdelendFL enforced on the solution\DIFdelbeginFL \DIFdelFL{subnetwork}\DIFdelendFL , other than containing high scoring, connected nodes. \textbf{\emph{Comp.}}: number of connected components in the solution. \textbf{\emph{Input}}: genotype data or GWAS summary statistics. \textbf{\emph{Scoring}}: how SNP/gene P\=/values \DIFdelbeginFL \DIFdelFL{are }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{were }\DIFaddendFL transformed into node scores. In the case of heinz, BUM stands for beta-uniform model\DIFdelbeginFL \DIFdelFL{, used to transform the P-values}\DIFdelendFL ; for SigMod, $\Phi$\textsuperscript{-1} represents the inverse of the cumulative distribution function of the standard Normal distribution. \textbf{\emph{Ref.}}: original publication featuring the algorithm.
  \end{flushleft}
  \label{tab:method_comparison}
  \end{adjustwidth}
\end{table}

\begin{description}
\item[{dmGWAS}] dmGWAS seeks the subgraph with the highest local density in low P\=/values \cite{jia_dmgwas:_2011}. To that end\DIFaddbegin \DIFadd{, }\DIFaddend it searches candidate \DIFdelbegin \DIFdel{subnetwork }\DIFdelend solutions using a greedy, ``seed and extend'', heuristic:

\begin{enumerate}
\item Select a seed node $i$ and form the subnetwork $S_i=\{i\}$.
\item Compute Stouffer's Z-score Z\textsubscript{m} for $S_i$ as

\begin{eqnarray} 
Z_m = \frac{1}{\sqrt{k}} \sum_{j \in S_i} z_j,
\end{eqnarray}

where \emph{k} is the number of genes in $S_i$\DIFdelbegin \DIFdel{; }\DIFdelend \DIFaddbegin \DIFadd{, }\DIFaddend z\textsubscript{j} is the Z score of gene $j$, computed as \(\phi\)\textsuperscript{-1}(1 - P\=/value\textsubscript{j})\DIFdelbegin \DIFdel{; }\DIFdelend \DIFaddbegin \DIFadd{, }\DIFaddend and \(\phi\)\textsuperscript{-1} is the inverse normal distribution function.
\item Identify neighboring nodes of $S_i$, i.e.\DIFaddbegin \DIFadd{, }\DIFaddend nodes at distance \(\le\) \DIFdelbegin \DIFdel{d}\DIFdelend \DIFaddbegin \texttt{\DIFadd{d}}\DIFaddend .
\item Add the neighboring nodes whose inclusion increases \DIFdelbegin \DIFdel{the Z\textsubscript{m+1} }\DIFdelend \DIFaddbegin \DIFadd{Z\textsubscript{m+1} by }\DIFaddend more than a threshold Z\textsubscript{m}~\texttimes{}~(1 + \DIFdelbegin \DIFdel{r}\DIFdelend \DIFaddbegin \texttt{\DIFadd{r}}\DIFaddend ).
\item Repeat 2-4 until no further enlargement is possible.
\item Add $S_i$ to the list of subnetworks to return. \DIFdelbegin \DIFdel{Its }\DIFdelend \DIFaddbegin \DIFadd{Normalize its }\DIFaddend Z-score \DIFdelbegin \DIFdel{is normalized }\DIFdelend as

\begin{eqnarray}
Z_{N}=\frac{Z_{m}-\operatorname{mean}\left(Z_{m}(\pi)\right)}{\operatorname{SD}\left(Z_{m}(\pi)\right)},
\end{eqnarray} 

where Z\textsubscript{m}(\(\pi\)) represents a vector containing 100\,000 random subsets of the same number of genes.
\end{enumerate}

DmGWAS carries out this process on every gene in the \DIFdelbegin \DIFdel{network}\DIFdelend \DIFaddbegin \DIFadd{PPIN}\DIFaddend . We used the implementation of dmGWAS in the dmGWAS 3.0 R package \cite{dmgwas}. \DIFdelbegin \DIFdel{We }\DIFdelend \DIFaddbegin \DIFadd{Unless otherwise specified, we }\DIFaddend used the suggested \DIFdelbegin \DIFdel{hyperparameters $d = 2$ and $r = 0.1$}\DIFdelend \DIFaddbegin \DIFadd{parameters }\texttt{\DIFadd{d}} \DIFadd{$ = 2$ and }\texttt{\DIFadd{r}} \DIFadd{$ = 0.1$}\DIFaddend . We used the function \DIFdelbegin \emph{\DIFdel{simpleChoose}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \texttt{\DIFadd{simpleChoose}} \DIFaddend to select the solution\DIFdelbegin \DIFdel{subnetwork}\DIFdelend , which aggregates the top 1\% subnetworks.

\item[{heinz}] The goal of heinz is to identify the highest-scored connected subnetwork \cite{dittrich_identifying_2008}. The authors \DIFdelbegin \DIFdel{propose }\DIFdelend \DIFaddbegin \DIFadd{proposed }\DIFaddend a transformation of the genes' P\=/value into a score that is negative under \DIFdelbegin \DIFdel{no }\DIFdelend \DIFaddbegin \DIFadd{weak }\DIFaddend association with the phenotype, and positive \DIFdelbegin \DIFdel{when there is}\DIFdelend \DIFaddbegin \DIFadd{under a strong one}\DIFaddend . This transformation is achieved by \DIFdelbegin \DIFdel{modelling }\DIFdelend \DIFaddbegin \DIFadd{modeling }\DIFaddend the distribution of P\=/values by a beta-uniform model (BUM) parameterized by the desired false discovery rate (FDR). Thus formulated, the problem is NP-complete, and hence solving it would require a prohibitively long computational time. To solve it efficiently\DIFaddbegin \DIFadd{, }\DIFaddend it is re-cast as the Prize-Collecting Steiner Tree Problem\DIFdelbegin \DIFdel{(PCST)}\DIFdelend , which seeks to select the connected subnetwork S that maximizes the \emph{profit} p(S), defined as:

\begin{eqnarray}
p(S) = \sum_{v \in V_S} p(v) - \sum_{e \in E_S} c(e). 
\end{eqnarray}

were $p(v)~=~w(v) - w'$ is the \emph{profit} of adding a node, $c(e)~=~w'$ is the \emph{cost} of adding an edge, and $w' = min_{v \in V_{G}} w(v)$ is the smallest node weight of $G$. All three are positive quantities. Heinz implements the algorithm from Ljubić et al. \cite{ljubic_algorithmic_2006} which, in practice\DIFaddbegin \DIFadd{, }\DIFaddend is often fast and optimal, although neither is guaranteed. We used BioNet's implementation of heinz \cite{beisser_bionet:_2010,heinz}.

\item[{HotNet2}] HotNet2 was developed to find connected subgraphs of genes frequently mutated in cancer \cite{leiserson_pan-cancer_2015}. To that end, it considers both the local topology of the \DIFdelbegin \DIFdel{network and the scores of the nodes. The former is captured by an }\DIFdelend \DIFaddbegin \DIFadd{PPIN and the nodes' scores. An }\DIFaddend insulated heat diffusion process \DIFaddbegin \DIFadd{captures the former}\DIFaddend : at initialization, the score of the node determines its initial heat; iteratively each node yields heat to its ``colder'' neighbors \DIFdelbegin \DIFdel{, }\DIFdelend and receives heat from its ``hotter'' neighbors \DIFdelbegin \DIFdel{, }\DIFdelend while retaining part of its own (hence, \emph{insulated}). This process continues until \DIFdelbegin \DIFdel{equilibrium }\DIFdelend \DIFaddbegin \DIFadd{a stationary state }\DIFaddend is reached, \DIFaddbegin \DIFadd{in which the temperature of the nodes does not change anymore, }\DIFaddend and results in a diffusion matrix $F$. $F$ is used to compute the similarity matrix $E$ that models exchanged heat as
\begin{eqnarray} 
E = F \operatorname{diag}(w(V)), 
\end{eqnarray}
where $\operatorname{diag}(w(V))$ is a diagonal matrix with the node scores in its diagonal. For any two nodes $i$ and $j$, $E_{ij}$ models the amount of heat that diffuses from node $j$ to node $i$\DIFdelbegin \DIFdel{, which }\DIFdelend \DIFaddbegin \DIFadd{. Hence, $E_{ij}$ }\DIFaddend can be interpreted as a (non-symmetric) similarity between those two nodes. To obtain densely connected \DIFdelbegin \DIFdel{subnetworks}\DIFdelend \DIFaddbegin \DIFadd{solutions}\DIFaddend , HotNet2 prunes $E$, only preserving edges such that $w(E) > \delta$. Lastly, HotNet2 evaluates the statistical significance of the \DIFdelbegin \DIFdel{subnetworks }\DIFdelend \DIFaddbegin \DIFadd{solutions }\DIFaddend by comparing their size to the size of \DIFdelbegin \DIFdel{networks }\DIFdelend \DIFaddbegin \DIFadd{PPINs }\DIFaddend obtained by permuting the node scores. We assigned \DIFaddbegin \DIFadd{the }\DIFaddend initial node scores as in Nakka et al. \cite{nakka_gene_2016}, \DIFdelbegin \DIFdel{assigning a score of }\DIFdelend \DIFaddbegin \DIFadd{giving a }\DIFaddend 0 \DIFdelbegin \DIFdel{for }\DIFdelend \DIFaddbegin \DIFadd{to }\DIFaddend the genes with \DIFdelbegin \DIFdel{low probability of being associated to }\DIFdelend \DIFaddbegin \DIFadd{a VEGAS2 P-values of association with }\DIFaddend the disease, and -log\textsubscript{10}(P\=/value) to those likely to be. In \DIFdelbegin \DIFdel{this }\DIFdelend \DIFaddbegin \DIFadd{the GENESIS }\DIFaddend dataset, the threshold separating both was a P\=/value of 0.125, which \DIFdelbegin \DIFdel{was }\DIFdelend \DIFaddbegin \DIFadd{we }\DIFaddend obtained using a local FDR approach \cite{scheid_twilight;_2005}. HotNet2 has two parameters: the restart probability \(\beta\), and the threshold heat \(\delta\). Both parameters are set automatically by the algorithm, which is robust to their values \cite{leiserson_pan-cancer_2015}. HotNet2 is implemented in Python \cite{hotnet2}.

\item[{LEAN}] LEAN searches altered ``star'' subnetworks, that is, subnetworks composed \DIFdelbegin \DIFdel{by }\DIFdelend \DIFaddbegin \DIFadd{of }\DIFaddend one central node and all its interactors \cite{gwinner_network-based_2016}. By imposing this restriction, LEAN \DIFdelbegin \DIFdel{is able to }\DIFdelend \DIFaddbegin \DIFadd{can }\DIFaddend exhaustively test all such subnetworks (one per node). For a particular \DIFaddbegin \DIFadd{star }\DIFaddend subnetwork of size \emph{m} \DIFdelbegin \DIFdel{, }\DIFdelend \DIFaddbegin \DIFadd{LEAN performs three steps:

}

\begin{enumerate}
\item \DIFadd{Rank }\DIFaddend the P\=/values \DIFdelbegin \DIFdel{corresponding to }\DIFdelend \DIFaddbegin \DIFadd{of }\DIFaddend the involved nodes \DIFdelbegin \DIFdel{are ranked }\DIFdelend as p\textsubscript{1} \(\le\) \dots{} \(\le\) p\textsubscript{m}. 
\DIFdelbegin \DIFdel{Then, }\DIFdelend \DIFaddbegin \item \DIFadd{Conduct }\DIFaddend \emph{k} binomial tests \DIFdelbegin \DIFdel{are conducted, }\DIFdelend to compute the probability of having \emph{k} out of \emph{m} P\=/values lower or equal to p\textsubscript{k} under the null hypothesis. The minimum of these \emph{k} P\=/values is the score of the subnetwork.
\DIFdelbegin \DIFdel{This score is transformed }\DIFdelend \DIFaddbegin \item \DIFadd{Transform this score }\DIFaddend into a P\=/value through an empirical distribution obtained via a subsampling scheme, where gene sets of the same size are selected randomly, and their score computed. 
\DIFdelbegin \DIFdel{Lastly, }\DIFdelend \DIFaddbegin \end{enumerate}

\DIFadd{We adjust these }\DIFaddend P\=/values \DIFdelbegin \DIFdel{are corrected }\DIFdelend for multiple testing through a Benjamini-Hochberg correction. We used the implementation of LEAN from the \DIFdelbegin \DIFdel{LEANR }\DIFdelend \DIFaddbegin \texttt{\DIFadd{LEANR}} \DIFaddend R package \cite{leanr}.

\DIFaddbegin 

\DIFaddend \item[{SConES}] SConES searches the minimal, modular, and maximally associated subnetwork in a SNP graph \cite{azencott_efficient_2013}. Specifically, it solves the problem

\begin{eqnarray}
\label{eq:scones}
\underset{S \subseteq G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} - \underbrace{\lambda \sum_{v \in V_S} \sum_{u \not\in V_S} {W}_{vu} }_{\text { connectivity }}-\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}\DIFaddbegin \DIFadd{,
}\DIFaddend \end{eqnarray}

where \(\lambda\) and \(\eta\) are \DIFdelbegin \DIFdel{hyperparameters }\DIFdelend \DIFaddbegin \DIFadd{parameters }\DIFaddend that control the sparsity and the connectivity of the model. The connectivity term penalizes disconnected solutions, with many edges between \DIFdelbegin \DIFdel{nodes that are selected and nodes that are not}\DIFdelend \DIFaddbegin \DIFadd{selected and unselected nodes}\DIFaddend . Given a $\lambda$ and an $\eta$, \DIFdelbegin \DIFdel{the aforementioned problem }\DIFdelend \DIFaddbegin \DIFadd{Eq~\ref{eq:scones} }\DIFaddend has a unique solution \DIFdelbegin \DIFdel{, }\DIFdelend that SConES finds using a graph min-cut procedure. As in Azencott et al. \cite{azencott_efficient_2013}, we selected \(\lambda\) and \(\eta\) by cross-validation, choosing the values that produce the most stable solution across folds. In this case, the selected \DIFdelbegin \DIFdel{hyperparameters }\DIFdelend \DIFaddbegin \DIFadd{parameters }\DIFaddend were \(\eta\)~=~3.51, \(\lambda\)~=~210.29 for SConES GS; \(\eta\)~=~3.51, \(\lambda\)~=~97.61 for SConES GM; and \(\eta\)~=~3.51, \(\lambda\)~=~45.31 for SConES GI. We used the version on SConES implemented in the R package \DIFdelbegin \DIFdel{martini }\DIFdelend \DIFaddbegin \texttt{\DIFadd{martini}} \DIFaddend \cite{martini}.

\item[{SigMod}] SigMod \DIFdelbegin \DIFdel{aims at identifying }\DIFdelend \DIFaddbegin \DIFadd{searches }\DIFaddend the highest-scoring, most densely connected \DIFdelbegin \DIFdel{gene }\DIFdelend subnetwork \cite{liu_sigmod:_2017}. It addresses an optimization problem similar to that of SConES (\DIFdelbegin \DIFdel{Equation }\DIFdelend \DIFaddbegin \DIFadd{Eq~}\DIFaddend \ref{eq:scones}), but \DIFdelbegin \DIFdel{the connectivity term encourages connected solutions by favoring solutions where many edgesconnect two selected nodes, rather than penalizing disconnected ones.  

}\DIFdelend \DIFaddbegin \DIFadd{with a different connectivity term that favors solutions containing many edges:  

}\DIFaddend 

\begin{eqnarray}
\underset{S \in G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} + \underbrace{\lambda \sum_{v \in V_S} \sum_{u \in V_S} W_{vu} }_{\text { connectivity }} -\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}.
\end{eqnarray}

As \DIFaddbegin \DIFadd{for }\DIFaddend SConES, this optimization problem can also be solved by a graph min-cut approach. 

SigMod presents three important differences with SConES. First\DIFdelbegin \DIFdel{it is designed for gene-gene networks}\DIFdelend \DIFaddbegin \DIFadd{, it was designed for PPINs}\DIFaddend . Second, it favors \DIFdelbegin \DIFdel{subnetworks }\DIFdelend \DIFaddbegin \DIFadd{solutions }\DIFaddend containing many edges \DIFaddbegin \DIFadd{between the selected nodes}\DIFaddend . SConES, instead, penalizes connections between \DIFdelbegin \DIFdel{the }\DIFdelend selected and unselected nodes. Third, it explores the grid of \DIFdelbegin \DIFdel{hyperparameters }\DIFdelend \DIFaddbegin \DIFadd{parameters }\DIFaddend differently, and processes their respective solutions. Specifically, for the range of \(\lambda\)~=~\(\lambda\)\textsubscript{min}, \dots{}, \(\lambda\)\textsubscript{max} for the same \(\eta\), it prioritizes the solution with the largest change in size from \(\lambda\)\textsubscript{n} to \(\lambda\)\textsubscript{n+1}. \DIFaddbegin \DIFadd{Additionally, that change needs to be larger than a user-specified threshold }\texttt{\DIFadd{maxjump}}\DIFadd{. }\DIFaddend Such a large change implies that the network is densely interconnected. This results in one candidate solution for each \(\eta\), which \DIFdelbegin \DIFdel{are }\DIFdelend \DIFaddbegin \DIFadd{is }\DIFaddend processed by removing any node not connected to any other. A score is assigned to each candidate solution by summing their node scores and normalizing by size. \DIFdelbegin \DIFdel{The }\DIFdelend \DIFaddbegin \DIFadd{Finally, SigMod chooses the }\DIFaddend candidate solution with the highest standardized score\DIFdelbegin \DIFdel{is the chosen solution}\DIFdelend \DIFaddbegin \DIFadd{, and that is not larger than a user-specified threshold (}\texttt{\DIFadd{nmax}}\DIFadd{). We used the default parameters }\texttt{\DIFadd{maxjump}} \DIFadd{$ = 10$ and }\texttt{\DIFadd{nmax}} \DIFadd{$= 300$}\DIFaddend . SigMod is implemented in an R package \cite{sigmod}.

\item[{Consensus}] We built a consensus \DIFdelbegin \DIFdel{network }\DIFdelend \DIFaddbegin \DIFadd{solution }\DIFaddend by retaining the \DIFdelbegin \DIFdel{nodes that were }\DIFdelend \DIFaddbegin \DIFadd{genes }\DIFaddend selected by at least two of the six methods (using SConES GI for SConES). \DIFaddbegin \DIFadd{It includes any edge between the selected genes in the PPIN.
}\DIFaddend \end{description}

\DIFaddbegin \DIFadd{We performed all the computations in the cluster described in Section~\ref{methods:computation}. 

}

\subsubsection{\DIFadd{Parameter space}}
\label{methods:parameters}

\DIFadd{We used the network methods with the parameters recommended by their authors, or with the default parameters in their absence. Additionally, we explored the parameter space of the different methods to study how they alter the output.

}

\begin{description}
\item[{\DIFadd{dmGWAS}}] \DIFadd{We tested multiple values for }\texttt{\DIFadd{r}} \DIFadd{(0.0001, 0.001, 0.01, 0.05, 0.1, 0.25, 0.5, and 1) and }\texttt{\DIFadd{d}} \DIFadd{(1, 2, and 3). 
}\item[{\DIFadd{heinz}}] \DIFadd{We tested multiple FDR thresholds (0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95, 1).
}\item[{\DIFadd{HotNet2}}] \DIFadd{We tested different thresholds to decide which genes would receive a score of 0 and which ones a score of $-log_{10}$(P-value): 0.001, 0.01, 0.05, 0.125, 0.25, and 0.5.
}\item[{\DIFadd{LEAN}}] \DIFadd{We used the following significance cutoffs for LEAN's P-values (0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95, and 1).
}\item[{\DIFadd{SConES}}] \DIFadd{We used the values of $\lambda$ and $\eta$ that }\texttt{\DIFadd{martini}} \DIFadd{explores by default (35.54, 5.40, 0.82, 0.12, 0.02, 0.01, 4.39e-4, 6.68e-5, 1.02e-5, and 1.55e-6 in both cases)
}\item[{\DIFadd{SigMod}}] \DIFadd{We tested multiple values for the parameters }\texttt{\DIFadd{nmax}} \DIFadd{(10, 50, 100, 300, 700, 1000, and 10\,000) and }\texttt{\DIFadd{maxjump}} \DIFadd{(5, 10, 20, 30, and 50).
}\end{description}

\subsubsection{\DIFadd{Comparing SNP-methods to gene-methods and vice versa}}
\label{methods:snp2gene}

\DIFadd{In multiple steps of this article, we compared the outcome of a method that works on genes with the outcome of one that works on SNPs. For this purpose, we used the SNP-gene correspondence described in Section~\ref{methods:conventional}. To convert a list of SNPs into a list of genes, we included all the genes mapped to any of those SNPs. Conversely, to convert a list of genes into a list of SNPs, we included all the SNPs mapped to any of those genes.

}

\DIFaddend \subsection{Pathway enrichment analysis}
\label{methods:pathway_enrichment}

We searched for pathways enriched in the gene \DIFdelbegin \DIFdel{subnetworks }\DIFdelend \DIFaddbegin \DIFadd{solutions }\DIFaddend produced by the above methods. We conducted \DIFdelbegin \DIFdel{an }\DIFdelend \DIFaddbegin \DIFadd{a }\DIFaddend hypergeometric test on pathways from Reactome \cite{Jassal2019} using the \DIFdelbegin \DIFdel{R package ReactomePA }\DIFdelend \DIFaddbegin \DIFadd{function }\texttt{\DIFadd{enrichPathway}} \DIFadd{from the }\texttt{\DIFadd{ReactomePA}} \DIFadd{R package }\DIFaddend \cite{Yu2016}. The universe of genes included any gene that we could map to a SNP in the iCOGS array (Section\DIFdelbegin \DIFdel{\ref{methods:node_score}}\DIFdelend \DIFaddbegin \DIFadd{~\ref{methods:conventional}}\DIFaddend ). We adjusted the P-values for multiple testing as in Benjamini and Hochberg \cite{Benjamini1995} (BH)\DIFdelbegin \DIFdel{. Pathways with an }\DIFdelend \DIFaddbegin \DIFadd{: pathways with a }\DIFaddend BH adjusted P\=/value \textless~0.05 were deemed significant.

\DIFdelbegin \DIFdel{As the significant pathways are often overlapping and redundant, the results were manually curated afterwards.

}\DIFdelend 

\subsection{\DIFdelbegin \DIFdel{Evaluation }\DIFdelend \DIFaddbegin \DIFadd{Benchmark }\DIFaddend of methods}
\DIFdelbegin %DIFDELCMD < \label{methods:comparison}
%DIFDELCMD < 
%%%
\DIFdelend \DIFaddbegin \label{methods:benchmark}

\DIFaddend 

We evaluated multiple properties \DIFdelbegin \DIFdel{(described below) }\DIFdelend of the different methods \DIFaddbegin \DIFadd{(described in Sections~\ref{methods:algorithm_comparison} and \ref{methods:classifier}) }\DIFaddend through a 5-fold subsampling setting. We applied each method to 5 random subsets of the original \DIFaddbegin \DIFadd{GENESIS }\DIFaddend dataset containing 80\% of the samples (\emph{train set}). When pertinent, we evaluated the solution on the remaining 20\% (\emph{test set}). We used the 5 repetitions to estimate the average and the standard deviation of the different measures. \DIFaddbegin \DIFadd{Every method and repetition ran in the same computational settings (Section~\ref{methods:computation}).

}\DIFaddend 

\subsubsection{Properties \DIFdelbegin \DIFdel{ot }\DIFdelend \DIFaddbegin \DIFadd{of }\DIFaddend the solution}
\label{methods:algorithm_comparison}

We compared the runtime, the number of selected \DIFdelbegin \DIFdel{genes/SNPs}\DIFdelend \DIFaddbegin \DIFadd{features (genes or SNPs)}\DIFaddend , and the stability (sensitivity \DIFdelbegin \DIFdel{of the result to small changes in the input, here, using different }\emph{\DIFdel{train}} %DIFAUXCMD
\DIFdel{sets}\DIFdelend \DIFaddbegin \DIFadd{to the choice of train set}\DIFaddend ) of the different network methods. \DIFdelbegin \DIFdel{The stability was quantified }\DIFdelend \DIFaddbegin \DIFadd{Nogueira and Brown \mbox{%DIFAUXCMD
\cite{nogueira_measuring_2016} }\hspace{0pt}%DIFAUXCMD
proposed quantifying a method's stability }\DIFaddend using the Pearson correlation \DIFdelbegin \DIFdel{to measure the overlap between different runs as suggested by Nogueira and Brown \mbox{%DIFAUXCMD
\cite{nogueira_measuring_2016}}\hspace{0pt}%DIFAUXCMD
}\DIFdelend \DIFaddbegin \DIFadd{between the genes selected on different subsamples. This correlation was calculated between vectors with the length of the total number of features, containing a 0 at position $i$ if feature $i$ was not selected and a 1 if it was}\DIFaddend .

\subsubsection{Classification accuracy of selected SNPs}
\label{methods:classifier}

A desirable solution offers good predictive power on \DIFdelbegin \DIFdel{unseen }\emph{\DIFdel{test}} %DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{the unseen test }\DIFaddend samples. We evaluated the predicting power of the SNPs selected by the different methods through the performance of an L1-penalized logistic regression classifier, \DIFdelbegin \DIFdel{a machine learning algorithm that searches }\DIFdelend \DIFaddbegin \DIFadd{which searches for }\DIFaddend a small subset of SNPs \DIFdelbegin \DIFdel{which provide }\DIFdelend \DIFaddbegin \DIFadd{that provides }\DIFaddend good classification accuracy \DIFdelbegin \DIFdel{. We trained the classifier exclusively on those selected SNPs to predict the }\DIFdelend \DIFaddbegin \DIFadd{at predicting the }\DIFaddend outcome (case/control). The L1 penalty helps to account for linkage disequilibrium by reducing the number of SNPs included in the model (\emph{active set})\DIFdelbegin \DIFdel{, while improving the generalization of the classifier. This }\DIFdelend \DIFaddbegin \DIFadd{. The active set was a plausible, more sparse solution with comparable predictive power to the original solution. The L1 }\DIFaddend penalty was set by cross-validation, choosing the value that minimized misclassification error.

\DIFaddbegin 

\DIFaddend We applied each network method to each \DIFdelbegin \emph{\DIFdel{train}} %DIFAUXCMD
\DIFdel{set, and }\DIFdelend \DIFaddbegin \DIFadd{train set. Then, we }\DIFaddend trained the classifier on \DIFdelbegin \DIFdel{it as well using only on }\DIFdelend \DIFaddbegin \DIFadd{the same train set using only }\DIFaddend the selected SNPs. When the method retrieved a list of genes\DIFdelbegin \DIFdel{(all of them except SConES), we considered as selected all the SNPs mapped to any of those genes. Then }\DIFdelend \DIFaddbegin \DIFadd{, }\DIFaddend we \DIFaddbegin \DIFadd{proceeded as explained in Section~\ref{methods:snp2gene}. Lastly we }\DIFaddend evaluated the sensitivity and the specificity \DIFdelbegin \DIFdel{on the }\emph{\DIFdel{test set}}%DIFAUXCMD
\DIFdel{. The active setgave an estimate of a plausible, more sparse solution with a comparable predictive power to the original solution}\DIFdelend \DIFaddbegin \DIFadd{of the classifier on the test set}\DIFaddend . To obtain a baseline, we also trained the classifier on all the SNPs \DIFdelbegin \DIFdel{. We do }\DIFdelend \DIFaddbegin \DIFadd{of the train set.

}

\DIFadd{We did }\DIFaddend not expect a linear model on selected SNPs to \DIFdelbegin \DIFdel{be able to }\DIFdelend separate cases from controls well. Indeed, the lifetime \DIFaddbegin \DIFadd{cumulative }\DIFaddend incidence of breast cancer among women with a family history of breast or ovarian cancer, and no \emph{BRCA1/2} mutations, is only 3.9 times more than in the general population \cite{Metcalfe2008}. However, classification accuracy may be one additional informative criterion on which to evaluate solutions.

\DIFdelbegin \subsubsection{\DIFdel{Comparison to state-of-the-art}}
%DIFAUXCMD
\addtocounter{subsubsection}{-1}%DIFAUXCMD
\DIFdelend \DIFaddbegin \subsection{\DIFadd{Comparison to state-of-the-art}}
\DIFaddend \label{methods:bcac}

An alternative way to evaluate the \DIFdelbegin \DIFdel{results is comparing our results }\DIFdelend \DIFaddbegin \DIFadd{methods is by comparing their solutions }\DIFaddend to an external dataset. For that purpose, we \DIFdelbegin \DIFdel{recovered a list of }\DIFdelend \DIFaddbegin \DIFadd{used the }\DIFaddend 153 genes associated to familial breast cancer \DIFdelbegin \DIFdel{from }\DIFdelend \DIFaddbegin \DIFadd{on }\DIFaddend DisGeNET \cite{pinero_disgenet:_2017}. Across this article\DIFaddbegin \DIFadd{, }\DIFaddend we refer to these genes as \emph{breast cancer susceptibility genes}.

Additionally, we used the summary statistics from the Breast Cancer Association Consortium (BCAC), a meta-analysis of case-control studies conducted in multiple countries\DIFdelbegin \DIFdel{which }\DIFdelend \DIFaddbegin \DIFadd{. BCAC }\DIFaddend included 13\,250\,\DIFdelbegin \DIFdel{642 }\DIFdelend \DIFaddbegin \DIFadd{641 }\DIFaddend SNPs genotyped or imputed on 228\,951 women of European ancestry\DIFaddbegin \DIFadd{, }\DIFaddend mostly from the general population \cite{Michailidou2017}. \DIFdelbegin \DIFdel{Hence, a high proportion of breast cancer cases investigated in BCAC are sporadic (not selected according to family history), while GENESIS is a homogeneous dataset not included in BCAC and which focus on the French high-risk population attending the family cancer clinics. Despite these differences, we expect some degree of shared genetic architecture, especially at the gene level. For that purpose, we searched associated genes }\DIFdelend \DIFaddbegin \DIFadd{Through imputation, BCAC includes more SNPs than the iCOGS array used for GENESIS (Section~\ref{methods:data}). However, in all the comparisons in this paper we focused on the SNPs that passed quality control in GENESIS. Hence, we used the same Bonferroni threshold }\DIFaddend as in Section~\DIFdelbegin \DIFdel{\ref{methods:node_score}. We provided }\DIFdelend \DIFaddbegin \DIFadd{\ref{methods:conventional} to determine the significant SNPs in BCAC. We also computed gene-scores in the BCAC data using }\DIFaddend VEGAS2\DIFdelbegin \DIFdel{with both }\DIFdelend \DIFaddbegin \DIFadd{, as in Section~\ref{methods:data}. In this case, we did use }\DIFaddend the summary statistics of all \DIFdelbegin \DIFdel{available SNPs, }\DIFdelend \DIFaddbegin \DIFadd{13\,250\,641 available SNPs }\DIFaddend and the genotypes from European samples from the 1000 Genomes Project \cite{the_1000_genomes_project_consortium_global_2015} to compute the LD patterns. \DIFaddbegin \DIFadd{Since these genotypes did not include chromosome X, we excluded it from this analysis. All comparisons included only the genes in common between GENESIS and BCAC, so we used a different Bonferroni threshold ($1.66 \times 10^{-6}$) to call gene significance. 

}\DIFaddend 

\subsection{\DIFdelbegin \DIFdel{Code availability}\DIFdelend \DIFaddbegin \DIFadd{Network rewirings}\DIFaddend }
\DIFaddbegin \label{methods:rewiring}

\DIFaddend 

\DIFaddbegin \DIFadd{Rewiring the PPIN while preserving the number of edges of each gene allowed to study the impact of the topology on the output of network methods. Indeed, the edges lose their biological meaning while the topology of the network is conserved. We produced 100 such rewirings by randomly swapping edges in the PPIN. We still scored the genes as described in Section~\ref{methods:methods}. We only applied only four methods on the rewirings: heinz, dmGWAS, LEAN, and SigMod. We excluded HotNet2 and SConES since they took notably longer to run. 

}

\subsection{\DIFadd{Computational resources}}
\label{methods:computation}

\DIFadd{We ran all the computations on a Slurm cluster, running Ubuntu 16.04.2 on the nodes. The CPU models on the nodes were Intel Xeon CPU E5-2450 v2 at 2.50GHz and Intel Xeon E5-2440 at 2.40GHz. The nodes running heinz and HotNet2 had 20GB of memory; the ones running dmGWAS, LEAN, SConES, and SigMod, 60GB. For the benchmark (Section~\ref{methods:benchmark}), we ran each of the methods on the same Ubuntu 16.04.2 node, with a CPU Intel Xeon E5-2450 v2 at 2.50GHz, and 60GB of memory.

}

\subsection{\DIFadd{Code and data availability}}
\label{methods:code}

\DIFaddend We developed computational pipelines for several steps of GWAS analyses, such as physically mapping SNPs to genes, computing gene scores, and \DIFdelbegin \DIFdel{performing }\DIFdelend \DIFaddbegin \DIFadd{running }\DIFaddend six different network \DIFdelbegin \DIFdel{analyses. For each of those processes, we }\DIFdelend \DIFaddbegin \DIFadd{methods. We }\DIFaddend created a pipeline with a clear interface that should work on any GWAS dataset \DIFaddbegin \DIFadd{for each of those processes}\DIFaddend . They are compiled in \url{https://github.com/hclimente/gwas-tools}. \DIFdelbegin \DIFdel{Although the GENESIS data is not public, the code to apply the pipelines to this data}\DIFdelend \DIFaddbegin \DIFadd{The code that applies them to GENESIS}\DIFaddend , as well as the code that reproduces all the analyses in this article are available at \url{https://github.com/hclimente/genewa}. We deposited all the produced gene \DIFdelbegin \DIFdel{subnetworks }\DIFdelend \DIFaddbegin \DIFadd{solutions }\DIFaddend on NDEx (\url{http://www.ndexbio.org}), under the UUID e9b0e22a-e9b0-11e9-bb65-0ac135e8bacf.

\DIFaddbegin \DIFadd{Summary statistics for SNPs and genes are available at }\url{https://github.com/hclimente/genewa}\DIFadd{. We cannot share genotype data publicly for confidentiality reasons, but are available from GENESIS. Interested researchers can contact }\href{mailto:nadine.andrieu@curie.fr}{\DIFadd{nadine.andrieu(at)curie.fr}}\DIFadd{.

}

\DIFaddend \section{Supporting information}

% Include only the SI item label in the paragraph heading. Use the \nameref{label} command to cite SI items in the text.
\paragraph*{S1 Table.}
\label{stab:snp_solutions}
\DIFdelbegin %DIFDELCMD < {\bf %%%
\DIFdel{Summary statistics on the results of SConES on the three SNP-SNP interaction networks.}%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \textbf{\DIFadd{Summary statistics on the results of SConES on the three SNP networks (Section~\ref{methods:networks}).}} \DIFaddend The first row within each block contains the summary statistics on the whole network.

\paragraph*{S2 Table.}
\DIFdelbegin %DIFDELCMD < \label{stab:scones_gene_solutions}
%DIFDELCMD < {\bf %%%
\DIFdel{Summary statistics on the results of multiple network methods on the gene-gene interaction network.}%DIFDELCMD < } %%%
\DIFdel{The first row contains the summary statistics on the whole network.

}\DIFdelend \DIFaddbegin \label{stab:sigmod_pwy}
\textbf{\DIFadd{Pathway enrichment analyses of the genes in SigMod's solution.}}

\DIFaddend 

\DIFaddbegin \paragraph*{\DIFadd{S3 Table.}}
\label{stab:dmgwas_pwy}
\textbf{\DIFadd{Pathway enrichment analyses of the genes in dmGWAS' solution.}}

\paragraph*{\DIFadd{S4 Table.}}
\label{stab:hotnet2_pwy}
\textbf{\DIFadd{Pathway enrichment analyses of the genes in HotNet2's solution.}}

\paragraph*{\DIFadd{S5 Table.}}
\label{stab:consensus_pwy}
\textbf{\DIFadd{Pathway enrichment analyses of the genes in the consensus' solution.}}

\DIFaddend \paragraph*{S1 Fig.}
\label{sfig:pcs}
\DIFdelbegin %DIFDELCMD < {\bf %%%
\DIFdel{GENESIS shows no differential population structure between cases and controls.}%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \textbf{\DIFadd{GENESIS shows no differential population structure between cases and controls.}} \DIFaddend \textbf{(A,B,C,D)}\DIFaddbegin \DIFadd{~}\DIFaddend Eight main principal components\DIFaddbegin \DIFadd{, }\DIFaddend computed on the genotypes of GENESIS. Cases are colored in green, controls in orange.
% \includegraphics[width=.9\linewidth]{./figures/sfigure_1.pdf}

\paragraph*{S2 Fig.}
\label{sfig:snp_gene_manhattan}
\DIFdelbegin %DIFDELCMD < {\bf %%%
\DIFdel{Association in GENESIS. The red line represents the Bonferroni threshold.}%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \textbf{\DIFadd{Association in GENESIS. The red lines represent the Bonferroni thresholds.}} \DIFaddend \textbf{(A)}\DIFaddbegin \DIFadd{~}\DIFaddend SNP association, measured from the outcome of a 1 d.f. $\chi^2$ allelic test \DIFaddbegin \DIFadd{(Section~\ref{methods:conventional})}\DIFaddend . Significant SNPs \DIFdelbegin \DIFdel{that are }\DIFdelend within a coding gene, or within 50 kilobases of its boundaries, are annotated. The Bonferroni threshold is 2.54 \texttimes{} 10\textsuperscript{-7}. \textbf{(B)}\DIFaddbegin \DIFadd{~}\DIFaddend Gene association, measured by P\=/value of VEGAS2 \cite{mishra_vegas2:_2015} using the 10\% of SNPs with the lowest P\=/values \DIFaddbegin \DIFadd{(Section~\ref{methods:conventional})}\DIFaddend . The Bonferroni threshold is 1.53 \texttimes{} 10\textsuperscript{-6}. \textbf{(C)}\DIFaddbegin \DIFadd{~}\DIFaddend SNP association as in panel (A). The SNPs in black \DIFdelbegin \DIFdel{are selected by a }\DIFdelend \DIFaddbegin \DIFadd{were selected by an }\DIFaddend L1-penalized logistic regression (Section~\ref{methods:classifier}, $\lambda = 0.03$).
%\includegraphics[width=.7\linewidth]{./figures/sfigure_2.png}

\paragraph*{S3 Fig.}
\DIFaddbegin \label{sfig:consensus_stats}
\textbf{\DIFadd{Relationship between the log\textsubscript{10} of the betweenness centrality and the -log\textsubscript{10} of the VEGAS2 P\=/value of the genes in the consensus solution.}} \DIFadd{The blue line represents a fitted generalized linear model.
%DIF >  \includegraphics[width=.65\textwidth]{./figures/sfigure_3.pdf}
  
}

\paragraph*{\DIFadd{S4 Fig.}}
\label{sfig:additional-benchmarks}
\textbf{\DIFadd{Additional benchmarks of the network methods.}} \textbf{\DIFadd{(A)}}\DIFadd{~Precision and recall of the evaluated methods with respect to Bonferroni-significant SNPs/genes in BCAC. For reference, we added a gray line with a slope of 1. This panel is identical to Fig~\ref{fig:consensus}. }\textbf{\DIFadd{(B)}}\DIFadd{~Sensitivity and specificity on the test set of the L1-penalized logistic regression trained on the features selected by each of the methods. The performance of the classifier trained on all SNPs is also displayed. Points are the average over the 5 runs; the error bars represent the standard error of the mean.
%DIF >  \includegraphics[width=.675\linewidth]{./figures/sfigure_4.pdf

}

\paragraph*{\DIFadd{S5 Fig.}}
\DIFaddend \label{sfig:pearson_methods}
\DIFdelbegin %DIFDELCMD < {\bf %%%
\DIFdel{Pearson correlation between the different solution subnetworks. }%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \textbf{\DIFadd{Pearson correlation between the different solutions.}} \DIFaddend \textbf{(A)}\DIFaddbegin \DIFadd{~}\DIFaddend Correlation between selected SNPs. \textbf{(B)}\DIFaddbegin \DIFadd{~}\DIFaddend Correlation between selected genes. In general, the solutions display a very low overlap.
%DIF < \includegraphics[width=.9\linewidth]{./figures/sfigure_3.pdf}
  
%DIF > \includegraphics[width=.9\linewidth]{./figures/sfigure_6.pdf}

\paragraph*{\DIFdelbegin \DIFdel{S4 }\DIFdelend \DIFaddbegin \DIFadd{S6 }\DIFaddend Fig.}
\DIFdelbegin %DIFDELCMD < \label{sfig:consensus-names}
%DIFDELCMD < {\bf %%%
\DIFdel{Consensus subnetwork on GENESIS (Section~\ref{methods:methods}). }%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \label{sfig:rewirings}
\DIFaddend \textbf{\DIFaddbegin \DIFadd{Number of times a gene was selected by either dmGWAS, heinz, LEAN, or SigMod in 100 rewirings of the PPIN }\DIFaddend (\DIFdelbegin \DIFdel{A}\DIFdelend \DIFaddbegin \DIFadd{Section~\ref{methods:rewiring}}\DIFaddend ) \DIFdelbegin %DIFDELCMD < \MBLOCKRIGHTBRACE %%%
\DIFdel{Each node is represented by a pie chart, which shows the methods that selected it. We labeled (and enlarged) the two most central genes (}\emph{\DIFdel{COPS5}} %DIFAUXCMD
\DIFdelend and \DIFdelbegin \emph{\DIFdel{OFD1}}%DIFAUXCMD
\DIFdel{) and those genes that are known breast cancer susceptibility genes and/or significantly associated with breast cancer susceptibility in the BCAC dataset. The latter ones are also colored in pink. This panel is identical }\DIFdelend \DIFaddbegin \DIFadd{its centrality degree.}} \DIFadd{This figure is equivalent }\DIFaddend to Fig~\DIFdelbegin \DIFdel{\ref{fig:consensus}. }\textbf{\DIFdel{(B)}} %DIFAUXCMD
\DIFdel{Same network, but every gene name is indicated. 
%DIF <  \includegraphics[width=.675\linewidth]{./figures/sfigure_4.pdf}
  
}\DIFdelend \DIFaddbegin \DIFadd{\ref{fig:issues}B, split by method. 
%DIF >  \includegraphics[width=.9\linewidth]{./figures/sfigure_5.pdf

}\DIFaddend 

\paragraph*{\DIFdelbegin \DIFdel{S5 }\DIFdelend \DIFaddbegin \DIFadd{S7 }\DIFaddend Fig.}
\DIFdelbegin %DIFDELCMD < \label{sfig:consensus_stats}
%DIFDELCMD < {\bf %%%
\DIFdel{Genes on the consensus network. Breast cancer susceptibility genes are colored in pink; the rest are colored in grey. }%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \label{sfig:params_size}
\textbf{\DIFadd{Size of the solutions obtained under different parameters.}} \DIFaddend \textbf{(A)}\DIFdelbegin \DIFdel{Number of methods selecting every gene in the subnetwork. }\textbf{\DIFdel{(B)}} %DIFAUXCMD
\DIFdel{VEGAS2 P\=/valuesof association of the genes, with regards to the number of methods that selected them. }\textbf{\DIFdel{(C)}} %DIFAUXCMD
\DIFdel{Comparison of betweenness centrality of the genes in the consensus network and the other genes in the PPIN and not in the consensus network. To improve visualization, we removed outliers. }\DIFdelend \DIFaddbegin \DIFadd{~Size of the solution produced by different parameter values, expressed as a percentage of the maximum solution size for the method, or the highest tested value for the parameter, respectively. The size of the solution is the median among all the solution sizes for the same parameter. }\DIFaddend \textbf{(\DIFdelbegin \DIFdel{D}\DIFdelend \DIFaddbegin \DIFadd{B}\DIFaddend )}\DIFdelbegin \DIFdel{Relationship between the log\textsubscript{10} of the betweenness centrality and the -log\textsubscript{10} of the VEGAS2 P\=/value of the genes in the consensus network. The blue line represents a fitted generalized linear model.
%DIF <  \includegraphics[width=.65\textwidth]{./figures/sfigure_5.pdf}
  
}%DIFDELCMD < 

%DIFDELCMD < %%%
\paragraph*{\DIFdel{S6 Fig.}}
%DIFAUXCMD
%DIFDELCMD < \label{sfig:stable-consensus-names}
%DIFDELCMD < {\bf %%%
\DIFdel{Stable consensus subnetwork on GENESIS }\DIFdelend \DIFaddbegin \DIFadd{~Boxplot of the solution sizes of the methods under the explored parameters }\DIFaddend (Section~\DIFdelbegin \DIFdel{\ref{results:stable-consensus}).
}%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \DIFadd{\ref{methods:parameters}).
%DIF >  \includegraphics[width=.9\linewidth]{./figures/sfigure_7.pdf

}

\paragraph*{\DIFadd{S8 Fig.}}
\label{sfig:params_pearson}
\DIFaddend \textbf{\DIFdelbegin \DIFdel{(A)}\DIFdelend \DIFaddbegin \DIFadd{Pearson correlation between the solutions obtained under different parameters, computed as in Section~\ref{methods:algorithm_comparison}.}\DIFaddend } \DIFdelbegin \DIFdel{Each node is represented by a pie chart, which shows the methods that selected it. We labeled (and enlarged) the two most central genes (}\emph{\DIFdel{CUL3}}%DIFAUXCMD
\DIFdel{) and those genesthat are known breast cancer susceptibility genes and}\DIFdelend \DIFaddbegin \DIFadd{Grey tiles represent the cases where we could not compute the Pearson correlation because the two vectors were either all ones (all genes}\DIFaddend /\DIFdelbegin \DIFdel{or significantly associated with breast cancer susceptibility in the BCAC dataset.
The latter ones are also colored in pink. This panel is identical to Fig~\ref{fig:stable-consensus}. }\textbf{\DIFdel{(B)}} %DIFAUXCMD
\DIFdel{Same network, but every gene name is indicated.
%DIF <  \includegraphics[width=.675\linewidth]{./figures/sfigure_6.pdf}

}\DIFdelend \DIFaddbegin \DIFadd{SNPs were selected) or zeros (no genes/SNPs were selected).
%DIF >  \includegraphics[width=.9\linewidth]{./figures/sfigure_8.pdf

}\DIFaddend 

\paragraph*{\DIFdelbegin \DIFdel{S7 }\DIFdelend \DIFaddbegin \DIFadd{S9 }\DIFaddend Fig.}
\label{sfig:biotypes_excluded}
\DIFdelbegin %DIFDELCMD < {\bf %%%
\DIFdel{Biotypes of genes from the annotation that are not present in the HINT protein-protein interaction network.}%DIFDELCMD < }
%DIFDELCMD < %%%
%DIF <  \includegraphics[width=.9\linewidth]{./figures/sfigure_7.pdf}
  
\DIFdelend \DIFaddbegin \textbf{\DIFadd{Biotypes of genes from the annotation that are not present in the HINT PPIN.}}
%DIF >  \includegraphics[width=.9\linewidth]{./figures/sfigure_8.pdf}
  
\DIFaddend 

\paragraph*{\DIFdelbegin \DIFdel{S8 }\DIFdelend \DIFaddbegin \DIFadd{S10 }\DIFaddend Fig.}
\label{sfig:lc_ht_comparison}
\DIFdelbegin %DIFDELCMD < {\bf %%%
\DIFdel{Comparison of benchmark on high-throughput (HT) interactions to benchmark on both high-throughput and literature curated interactions (HT+LC). }%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \textbf{\DIFadd{Comparison of the benchmark on high-throughput (HT) interactions to the benchmark on both high-throughput and literature curated interactions (HT+LC).}} \DIFaddend Grey lines represent no change in the statistic between the benchmarks (1 for ratios mean(HT) / mean(HT + LC), 0 for differences mean(HT) - mean(HT + LC)). \textbf{(A)}\DIFaddbegin \DIFadd{~}\DIFaddend Ratios of the selected features between both benchmarks and of the active set \DIFaddbegin \DIFadd{(Section~\ref{methods:classifier})}\DIFaddend . \textbf{(B)}\DIFaddbegin \DIFadd{~}\DIFaddend Shifts in sensitivity and specificity. \textbf{(C)}\DIFaddbegin \DIFadd{~}\DIFaddend Shift in Pearson correlation between benchmarks. \textbf{(D)}\DIFaddbegin \DIFadd{~}\DIFaddend Ratio between the runtimes of the benchmarks. For \DIFdelbegin \DIFdel{gene network-based }\DIFdelend \DIFaddbegin \DIFadd{gene-based }\DIFaddend methods, inverted triangles represent the ratio of runtimes of the algorithms themselves, and circles the total time, which includes the algorithm themselves and the additional 119\,980 seconds (1 day and 9.33 hours) that VEGAS2 took on average to compute the gene scores from SNP summary statistics. In general, adding additional interactions slightly \DIFdelbegin \DIFdel{improves }\DIFdelend \DIFaddbegin \DIFadd{improved }\DIFaddend the stability of the solution\DIFdelbegin \DIFdel{, but increases }\DIFdelend \DIFaddbegin \DIFadd{. However, it increased }\DIFaddend the solution size \DIFdelbegin \DIFdel{, has }\DIFdelend \DIFaddbegin \DIFadd{and the required runtime, and had }\DIFaddend mixed effects on the sensitivity and specificity\DIFdelbegin \DIFdel{, and impacts negatively the required runtime of the algorithms.
%DIF <  \includegraphics[width=.85\linewidth]{./figures/sfigure_8.pdf}
  
}\DIFdelend \DIFaddbegin \DIFadd{.
%DIF >  \includegraphics[width=.85\linewidth]{./figures/sfigure_9.pdf}
  
}\DIFaddend 

\paragraph*{\DIFdelbegin \DIFdel{S9 }\DIFdelend \DIFaddbegin \DIFadd{S11 }\DIFaddend Fig.}
\label{sfig:scones_gsm}
\DIFdelbegin %DIFDELCMD < {\bf %%%
\DIFdel{Overview of the solutions produced by the SConES on the GS and GM networks (Section \ref{methods:networks}) on the GENESIS dataset.}%DIFDELCMD < } %%%
\DIFdelend \DIFaddbegin \textbf{\DIFadd{Overview of the solutions produced by the SConES on the GS and GM networks (Section~\ref{methods:networks}) on the GENESIS dataset.}} \DIFaddend \textbf{(A)}\DIFaddbegin \DIFadd{~}\DIFaddend Manhattan plots of SNPs \DIFaddbegin \DIFadd{(Section~\ref{methods:conventional})}\DIFaddend ; in black, the method’s solution. The \DIFaddbegin \DIFadd{red line indicates the }\DIFaddend Bonferroni threshold (2.54 \texttimes{} 10\textsuperscript{-7})\DIFdelbegin \DIFdel{is indicated by a red line}\DIFdelend . \textbf{(B)}\DIFaddbegin \DIFadd{~}\DIFaddend Precision and recall of the evaluated methods with respect to Bonferroni-significant SNPs (SConES) or genes (other methods) in BCAC. For reference, we added a gray line with a slope of 1. \textbf{(C)}\DIFaddbegin \DIFadd{~}\DIFaddend Solution networks.
%DIF <  \includegraphics[width=\linewidth]{./figures/sfigure_9.png}

%DIF >  \includegraphics[width=\linewidth]{./figures/sfigure_10.png}

\section*{Acknowledgments}

\DIFaddbegin 

\DIFaddend We wish to thank Om Kulkarni for helpful discussion on gene-based GWAS and PPIN databases, and the genetic epidemiology platform (the PIGE, Plateforme d'Investigation en Génétique et Epidemiologie\DIFaddbegin \DIFadd{: S. Eon-Marchais, M. Marcou, D. Le Gal, L. Toulemonde, J. Beauvallet, N. Mebirouk, E. Cavaciuti}\DIFaddend ), the biological resource \DIFdelbegin \DIFdel{centre and }\DIFdelend \DIFaddbegin \DIFadd{center (S. Mazoyer, F. Damiola, L. Barjhoux, C. Verny-Pierre, V. Sornin). We wish to pay tribute to Olga M. Sinilnikova, one of the initiators and principal investigators of GENESIS, and who died prematurely on June 30, 2014.

}

\DIFadd{We thank }\DIFaddend all the GENESIS collaborating cancer clinics clinics \DIFaddbegin \DIFadd{(Clinique Sainte Catherine, Avignon: H. Dreyfus; Hôpital Saint Jacques, Besançon: M-A}\DIFaddend . \DIFaddbegin \DIFadd{Collonge-Rame; Institut Bergonié, Bordeaux: M.Longy, A. Floquet, E. Barouk-Simonet; CHU, Brest: S. Audebert; Centre François Baclesse, Caen: P. Berthet; Hôpital Dieu, Chambéry: S. Fert-Ferrer; Centre Jean Perrin, Clermont-Ferrand: Y-J. Bignon; Hôpital Pasteur, Colmar: J-M. Limacher; Hôpital d’Enfants CHU – Centre Georges François Leclerc, Dijon: L. Faivre-Olivier; CHU, Fort de France: O. Bera; CHU Albert Michallon, Grenoble: D. Leroux; Hôpital Flaubert, Le Havre: V. Layet; Centre Oscar Lambret, Lille: P. Vennin, C. Adenis; Hôpital Jeanne de Flandre, Lille: S. Lejeune-Dumoulin, S. Manouvier-Hanu; CHRU Dupuytren, Limoges: L. Venat-Bouvet; Centre Léon Bérard, Lyon: C. Lasset, V. Bonadona; Hôpital Edouard Herriot, Lyon: S. Giraud; Institut Paoli-Calmettes, Marseille: F. Eisinger, L. Huiart; Centre Val d’Aurelle – Paul Lamarque, Montpellier: I. Coupier; CHU Arnaud de Villeneuve, Montpellier: I. Coupier, P. Pujol; Centre René Gauducheau, Nantes: C. Delnatte; Centre Catherine de Sienne, Nantes: A. Lortholary; Centre Antoine Lacassagne, Nice: M. Frénay, V. Mari; Hôpital Caremeau, Nîmes: J. Chiesa; Réseau Oncogénétique Poitou Charente, Niort: P. Gesta; Institut Curie, Paris: D. Stoppa-Lyonnet, M. Gauthier-Villars, B. Buecher, A. de Pauw, C. Abadie, M. Belotti; Hôpital Saint-Louis, Paris: O. Cohen-Haguenauer; Centre Viggo-Petersen, Paris: F. Cornélis; Hôpital Tenon, Paris: A. Fajac; GH Pitié Salpétrière et Hôpital Beaujon, Paris: C. Colas, F. Soubrier, P. Hammel, A. Fajac; Institut Jean Godinot, Reims: C. Penet, T. D. Nguyen; Polyclinique Courlancy, Reims: L. Demange*, C. Penet; Centre Eugène Marquis, Rennes: C. Dugast*; Centre Henri Becquerel, Rouen: A. Chevrier, T. Frebourg, J. Tinat, I. Tennevet, A. Rossi; Hôpital René Huguenin/Institut Curie, Saint Cloud: C. Noguès, L. Demange*, E. Mouret-Fourme; CHU, Saint-Etienne: F. Prieur; Centre Paul Strauss, Strasbourg: J-P. Fricker, H. Schuster; Hôpital Civil, Strasbourg: O. Caron, C. Maugard; Institut Claudius Regaud, Toulouse: L. Gladieff, V. Feillel; Hôpital Bretonneau, Tours: I. Mortemousque; Centre Alexis Vautrin, Vandoeuvre-les-Nancy: E. Luporsi; Hôpital de Bravois, Vandoeuvre-les-Nancy: P. Jonveaux; Gustave Roussy, Villejuif: A. Chompret*, O. Caron).
*Deceased prematurely

}\DIFaddend 

\section*{Author contributions}

\begin{description}
  \item[Conceptualization] Héctor Climente-González, Christine Lonjou, Chloé-Agathe Azencott.
  \item[Data curation] Christine Lonjou, GENESIS Study collaborators.
  \item[Formal Analysis] Héctor Climente-González, Christine Lonjou.
  \item[Funding acquisition] Dominique Stoppa-Lyonnet, Nadine Andrieu, Chloé-Agathe Azencott.
  \item[Investigation] Héctor Climente-González, Christine Lonjou.
  \item[Methodology] Héctor Climente-González, Christine Lonjou, Chloé-Agathe Azencott.
  \item[Project administration] Chloé-Agathe Azencott.
  \item[Resources] GENESIS Study collaborators, Dominique Stoppa-Lyonnet, Nadine Andrieu.
  \item[Software] Héctor Climente-González, Christine Lonjou.
  \item[Supervision] Christine Lonjou, Fabienne Lesueur, Nadine Andrieu, Chloé-Agathe Azencott.
  \item[Validation] Christine Lonjou, Fabienne Lesueur.
  \item[Visualization] Héctor Climente-González.
  \item[Writing – original draft] Héctor Climente-González.
  \item[Writing – review \& editing] Héctor Climente-González, Christine Lonjou, Fabienne Lesueur, Nadine Andrieu, Chloé-Agathe Azencott.
\end{description}

\nolinenumbers

\bibliography{bibliography.bib}

% Either type in your references using
% \begin{thebibliography}{}
% \bibitem{}
% Text
% \end{thebibliography}
%
% or
%
% Compile your BiBTeX database using our plos2015.bst
% style file and paste the contents of your .bbl file
% here. See http://journals.plos.org/plosone/s/latex for 
% step-by-step instructions.
% 

\end{document}
