#+TITLE: Combining network-guided GWAS to discover susceptibility mechanisms for breast cancer
#+AUTHOR: Héctor Climente-González, Christine Lonjou, Fabienne Lesueur, GENESIS investigators, Nadine Andrieu, Chloé-Agathe Azencott
#+OPTIONS: toc:nil num:nil

#+LATEX_HEADER: \usepackage{threeparttable}
#+LATEX_HEADER: \newcommand{\mean}[1]{$\overline{\mbox{#1}}$}
#+LATEX_HEADER: \newcommand{\median}[1]{$\hat{\mbox{#1}}$}

\begin{abstract}

Systems biology provides a comprehensive approach to biomarker discovery and biological hypothesis building. It does so by jointly considering the statistical association between a gene and a phenotype, obtained experimentally, and the biological context of each gene, represented as a network. In this work we study the utility of six network methods to discover new biomarkers in GWAS data, which search subnetworks highly associated to a phenotype. We interrogate a familial breast cancer GWAS focused on BRCA1/2 negative French women. By trading statistical astringency for biological meaningfulness, most network methods get more compelling results than standard SNP- and gene-level analyses, recovering causal subnetworks tightly related to cancer susceptibility. We perform an in-depth benchmarking of the methods with regards to size of the solution subnetwork, their utility as biomarkers, the stability of the solutions and their runtime. Interestingly, a combination solution subnetworks provided a concise subnetwork of 93 genes, enriched in known familial breast cancer susceptibility genes (BABAM1, BLM, CASP8, FGFR2, and TOX3, Fisher's exact test P-value = 7.8 \times 10^{-5}) and more central than average. Additionally, it includes subnetworks of mechanisms related to cancer, like protein folding (HSPA1A, HSPA1B, and HSPA1L) or mitocondrial ribosomes (MRPS30, MRPS31, MRPS18B). We also observed a general disregulation in the neighborhood of COPS5, a gene related to multiple hallmarks of cancer.

\end{abstract}

* Introduction

In human health, genome-wide association studies (GWAS) aim at quantifying how single-nucleotide polymorphisms (SNPs) predispose to complex diseases, like diabetes or some forms of cancer cite:bush_chapter_2012. To that end, in a typical GWAS thousands of unrelated samples are genotyped: the cases, suffering the disease of interest, and the controls from the general population. Then, a per-SNP statistical association test is conducted (e.g. \chi^2). Those SNPs with a P-value lower than a conservative Bonferroni threshold are candidates to further studies in an independent cohort. Once the risk SNPs have been discovered, they can be used for diagnosis, guide the treatment, and deepen our understanding of the disease.

GWAS have successfully identified thousands of variants underlying many common diseases cite:buniello_nhgri-ebi_2019. However, the experimental setting also presents intrinsic challenges. Some of them stem from the high-dimensionality of the problem, as every GWAS to date study more variants than samples are genotyped. This limits the statistical power of the experiment, as only variants with large and moderate effects can be detected. However, the prevailing view is that most genetic architectures involve many variants with small effects cite:visscher_10_2017. Additionally, to avoid false positives, a conservative multiple test correction is applied, typically Bonferroni. However, Bonferroni is known to be overly conservative when the statistical tests are correlated, as is the case in GWAS cite:wang_statistical_2018. Another open issue is the interpretability of the results, as the functional consequences of most common variants are not well understood. The fact that SNPs in non-coding regions often top the results of GWAS illustrates well this challenge. On top of that, recent large-sampled studies suggest that most of the genome contributes to a degree to any complex trait, in accordance with the infinitesimal model cite:barton_infinitesimal_2017. The omnigenic model cite:boyle_expanded_2017 explains this by the dense interconnection between genes, influencing each others' behavior, which allows alterations in most genes to impact the "core" genes involved in a disease. A comprehensive statistical framework which includes the structure of biological data might address the aforementioned issues.

# TODO Add citations Barrenas 2012, and Cowen 2017.
In this regard many authors turn to network biology to handle the complex interplay of biomolecules that lead to disease cite:furlong_human_2013. As its name suggests, network biology models biology as a network, where the biomolecules under study, often genes, are nodes, and selected functional relationships between them are the edges that link them. These functional relationships come from evidence that the genes jointly contribute to a biological function; for instance, their expression is correlated, or a protein-protein interaction between their gene products have been discovered. Under this view, complex diseases are not the consequence of a single altered gene, but of the interaction of multiple interdependent biomolecules cite:barabasi_network_2011. In fact, an examination of biological networks shows that disease genes have differential properties cite:barabasi_network_2011 cite:pinero_uncovering_2016. This is particularly true for cancer driver genes, which tend to be key players in connecting different densely-connected communities of genes. Additionally, as genes that contribute to a disease tend to participate in similar biological functions, guilt-by-association strategies have proved effective at identifying disease genes cite:huang_systematic_2018. 

Network-based, biomarker discovery methods exploit the guilt-by-association strategy to identify disease genes on GWAS data cite:azencott_network-guided_2016. In essence, each SNP has a measure of association with the disease, given by the experiment, and functionally biological relationships, given by a network built on prior knowledge. Then, the problem becomes finding a functionally-related set of genes that is highly associated with the disease. Different solutions have been proposed to this problem, often stemming from divergent different mathematical frameworks and considerations of what the optimal solution looks like. Some methods strongly constrain the problem to certain kinds of subnetworks. Such is the extreme case of LEAN cite:gwinner_network-based_2016, which focuses on star subnetworks, i.e. instances were a gene and its direct interactors are associated with the disease. Other algorithms, like dmGWAS cite:jia_dmgwas:_2011 and heinz cite:dittrich_identifying_2008, focus on interconnected genes with high association with the disease. However, they differ in their tolerance to the inclusion of lowly associated nodes, and the possible number of disconnected subnetworks in the solution. Lastly, other methods put further requirements on the solution subnetwork, favoring those that are densely interconnected; such is the case of HotNet2 cite:leiserson_pan-cancer_2015, SConES cite:azencott_efficient_2013, and SigMod cite:liu_sigmod:_2017.

In this work, we analyze the effectiveness of these six methods to discover new biomarkers on GWAS data. We focus on the GENESIS dataset cite:sinilnikova_genesis:_2016, a homogeneous dataset of familial breast cancer in French population of European ancestry. After following a classical GWAS approach, we use these network-based methods to recover additional familial breast cancer biomarkers. Some of them are known, while others are specific to this dataset. Lastly, we carry out a comparison of the solutions obtained by the different methods, and aggregate them to obtain a consensus network of predisposition to familial breast cancer. 

* Methods
** GENESIS

GENESIS is a GWAS study of familial breast cancer on the French population cite:sinilnikova_genesis:_2016. The index cases are patients with infiltrating mammary or ductal adenocarcinoma, who had sister with breast cancer, and negative for BRCA1 and BRCA2 mutations. The controls are unaffected colleagues and/or friends of the cases, born around the year of birth of the corresponding case (\pm 3 years). In total, the GENESIS dataset consists of 2577 samples, of which 1279 are controls and 1298 are cases. The genotyping platform was iCOGS, a custom Illumina array designed to study genetic susceptibility of hormone-related cancers cite:sakoda_turning_2013. It contains 210918 SNPs previously associated with cancer susceptibility, survival, or other cancer-related traits; and in selected genes and pathways.

# Molecular subtypes? TNBC, HER2+...

** Preprocessing and quality control

We discarded SNPs with a minor allele frequency lower than 0.1%, not in Hardy - Weinberg equilibrium (P-value \textless 0.001), and those with missing values on more than 10% of the samples. A subset of 20 duplicated SNPs in FGFR2 were also removed. In addition, we removed the samples with more than 10% missing genotypes, and an additional 28 samples with TODO. The final dataset included 1271 controls and 1280 cases, genotyped over 197083 SNPs. 

We looked for population structure that could create confounding associations. A PCA revealed no differential population structure between cases and controls (Supplementary Figure [[sfig:pcs]]). Independently, we did not find evidence of genomic inflation (\lambda = 1.05) either, further confirming the absence of confounding population structure.

** High-score subnetwork search algorithms
*** SNP and gene association 
    :PROPERTIES:
    :CUSTOM_ID: methods:node_score
    :END:
To measure association between a genotype and the phenotype, we performed a per-SNP 1df \chi^2 allelic test using PLINK v1.90 cite:chang_second-generation_2015. Then, we used VEGAS2v2 to compute the gene-level association score cite:mishra_vegas2:_2015 from the SNP P-values. In order to map SNPs to genes we used their overlap on the genome: all SNPs located within the boundaries of a gene, \pm 50 kb, were mapped to that gene. To compute the gene association we used the 10% of SNPs with lowest P-values. We used the 62193 genes described in GENCODE 31 cite:frankish_gencode_2019, although only 54612 could be mapped to at least one SNP. Out of those, we focused exclusively on the 32767 that could be mapped to an HGNC symbol. Out of the SNPs 197083 in iCOGS after quality control, 164037 mapped to at least one of these genes. 

We use such mapping when to compare the outputs of methods who produce SNP- and gene-lists. For the former, we consider any gene that can be mapped to any of the selected SNPs as selected as well. For the latter, we consider all the SNPs that can be mapped to that gene as selected by the method.  

*** Gene-gene network
    :PROPERTIES:
    :CUSTOM_ID: methods:gene_network
    :END:

Out of the six methods tested, five use a gene-gene interaction network (Section [[#methods:methods]]), and their respective statistical frameworks are compatible with any type of network (protein interactions, gene coexpression, regulatory, etc.). However, in order to make the results comparable, we needed to apply all the methods to the same network. Hence, for practical reasons, we focused on a protein-protein interaction network (PPIN), as they are interpretable and well characterized, and most of the methods were designed to scale appropriately to it. We built our PPIN from both binary and co-complex interactions stored in the HINT database (release April 2019) cite:das_hint:_2012. Unless specified otherwise, we used only interactions coming from high-throughput experiments to avoid biasing the topology of the network by well-studied genes with more known interactions on average. Out of the 146722 interactions from high-throughput experiments that HINT stores, we were able to map 142541 to a pair of HGNC symbols. The scoring function for the nodes changed from method to method (Section [[#methods:methods]]). 

Additionally, we compared the results of the aforementioned network with those obtained on a network built using interactions coming from both high-throughput and targeted studies. In that case, out of the 179332 interactions in HINT, we mapped 173797 to a pair of HGNC symbols. 

*** SNP networks
    :PROPERTIES:
    :CUSTOM_ID: methods:snp_network
    :END:

SConES cite:azencott_efficient_2013 is the only of the studied methods designed to handle SNP networks. As in gene networks, two SNPs are linked in a SNP network when there is evidence of shared functionality between two SNPs. The authors suggested three ways of building these networks: connecting the genotyped SNPs consecutive in the genomic sequence ("GS network"); interconnecting all the SNPs mapped to the same gene, on top of GS ("GM network"); and interconnecting all SNPs mapped to two genes for which a protein-protein interaction exists ("GI network"). Although we focused on the GI network, as it is the network that fits better the scope of this article, at different stages of the manuscript we also used GS and GM. For the GM network, we used the mapping described in Section [[#methods:node_score]]. For the GI network, we used the PPI as described in Section [[#methods:gene_network]]. For all three networks the node score used is the association of the individual SNPs with the phenotype; specifically, we used the 1 d.f. \chi^2.

*** Mathematical notation
    :PROPERTIES:
    :CUSTOM_ID: methods:notation
    :END:

In this article, we refer to undirected, vertex-weighted networks, or graphs, G = (V,E,w). V = {v_1, \dots, v_n} refers to the vertices, with weights w: V \rightarrow \mathbb{R}. Equivalently, E \subseteq {{x,y} | x,y \in V \wedge x \neq y} refers to the edges. When referring to a subnetwork S, V_S is the set of nodes in S and E_S is the set of edges in S. A special case of subgraphs are /connected/ subgraphs, which occur when every node in the subgraph can be reached from any other node.

In addition, we use several matrices that describe different properties of a graph. The described matrices are square, and have as many rows and columns as nodes are in the network. The element (i,j) represents a  selected relationship between v_i and v_j. The /adjacency matrix/ W_G contains a 1 when the corresponding nodes are connected through an edge, and 0 otherwise; the diagonal is zero. The /degree matrix/ D_G is a diagonal matrix which contains the degree of the different nodes. Lastly, the /Laplacian matrix/ L_G is defined as L_G = D_G - W_G.

*** Methods used
    :PROPERTIES:
    :CUSTOM_ID: methods:methods
    :END:

#+CAPTION:Summary of the differences between the studied algorithms.  
#+NAME:   tab:method_comparison
| Method  | Field | Nodes | Exhaustive | Solution | Components | Input     | Scoring      |
|---------+-------+-------+------------+----------+------------+-----------+--------------|
| /       | <     |       |            |          |            |           |              |
| dmGWAS  | GWAS  | Gene  | No         | -        | 1          | Summary   | -log_{10}(P) |
| heinz   | Omics | Gene  | Yes        | -        | 1          | Summary   | BUM          |
| HotNet2 | Omics | Gene  | Yes        | Modular  | \geq 1     | Summary   | Local FDR    |
| LEAN    | Omics | Gene  | Yes        | Star     | \geq 1     | Summary   | -log_{10}(P) |
| SConES  | GWAS  | SNP   | Yes        | Modular  | \geq 1     | Genotypes | \chi^2       |
| SigMod  | GWAS  | Gene  | Yes        | Modular  | 1          | Summary   | -log_{10}(P) |
# Add as footnote: The columns are: /Field/: field in which the algorithm was developed; /Nodes/, the type of network, either gene (protein-protein interaction network usually) or a SNP network; /Exhaustive/: whether all the possible solutions given the selected hyperparameters are explored; /Solution/: additional properties are enforced on the solution subnetwork, other than being dense in high scores and connected; /Components/: number of connected subnetworks in the solution; /Input/: genotype data or GWAS summary statistics; and /Scoring/: how SNP/gene P-values are transformed into node scores.

Beyond the assumption that genes that contribute to the same function will be nearby in the PPIN, they might be topologically related to each other in diverse ways (densely interconnected modules, nodes around a hub, a path, etc.). That is not the only choice to make: how to score the nodes, whether the affected mechanisms form a single connected component or several, how to frame the problem in a computationally efficient fashion, what is the best network to use, etc. In consequence, multiple solutions have been proposed. In this article, we examine six of them: five that explore the protein-protein interaction network, and one which explores SNP networks. We selected methods that were open source, had an implementation available, and an accessible documentation. Their main differences are summarized in Table [[tab:method_comparison]].

# TODO Reformulate heinz to show similarities to SConES.

- dmGWAS :: dmGWAS searches the subgraph with the highest local density in low P-values cite:jia_dmgwas:_2011. To that end it searches candidate subnetwork solutions using a greedy procedure involving the following steps:

  1. Select a seed node.
  2. Compute Stouffer's Z-score Z_m for the current subgraph as
    
    \begin{equation*} 
    Z_m = \frac{\sum z_i}{\sqrt{k}}
    \end{equation*}

    where /k/ is the number of genes in the subgraph, z_i = \phi^{-1}(1 - P-value_i), and \phi^{-1} is the inverse normal distribution function.
  3. Identify neighboring nodes i.e. nodes at distance \le /d/. We set d = 2.
  4. Add the neighboring nodes whose inclusion increases the Z_{m+1} more than Z_m \times (1 + r). In our experiments, we set r = 0.1.
  5. Repeat 2-4 until no increment Z_m \times (1 + r) is possible.
  
  Lastly, the module's Z-score is normalized as

  \begin{equation*}
  Z_{N}=\frac{Z_{m}-\operatorname{mean}\left(Z_{m}(\pi)\right)}{\operatorname{SD}\left(Z_{m}(\pi)\right)}
  \end{equation*} 

  where Z_{m}(\pi) represent a vector containing 100000 random subsets of the same number of genes.

  We used the implementation of dmGWAS in the dmGWAS 3.0 R package cite:dmgwas. We used the function /simpleChoose/ to select the solution subnetwork, which aggregates the top 1% modules into the solution subnetwork.

# TODO Re-read heinz paper. It's the solution heuristic? If so, how good is it? Efficient enough to be used in SNP network?
- heinz :: The goal of heinz is to identify the highest-scored connected subgraph on the network cite:dittrich_identifying_2008. The authors propose a transformation of the genes' P-value into a score which is negative when no association with the phenotype is detected, and positive value when it is. To that end, the distribution of P-values is modeled by a beta-uniform model (BUM) parameterized by the desired FDR. Thus formulated, the problem is NP-complete. To solve it efficiently it is re-casted as the Prize-Collecting Steiner Tree Problem (PCST), which seeks to select the connected subnetwork S that maximizes the /profit/ p(S):

    \begin{equation*}
    p(S) = \sum_{v \in V_S} p(v) - \sum_{e \in E_S} c(e). 
    \end{equation*}

    were p(v) = w(v) - w' is the /profit/ of adding a node, c(e) = w' is the /cost/ of adding an edge, and w' = min_{v \in V_G} w(v). All three are positive quantities. heinz implements the algorithm from cite:ljubic_algorithmic_2006, which in practice is often fast and optimal, neither is guaranteed. We used BioNet's implementation of heinz, available on Bioconductor cite:beisser_bionet:_2010,heinz.

- HotNet2 :: HotNet2 was developed to find connected subgraphs of genes frequently mutated in cancer cite:leiserson_pan-cancer_2015. To that end, it considers both the local topology of the network and the scores of the nodes. The former is captured by an insulated heat diffusion process: at the beginning, the score of the node determines its initial heat; iteratively each node yields heat to its "colder" neighbors, and receives heat from its "hotter" neighbors, while retaining part of its own (hence, /insulated/). This process continues until equilibrium is reached, and results in a similarity matrix F. F is used to compute the similarity matrix E that accounts also for similarities in node scores as 

    \begin{equation*} 
    E = F \operatorname{diag}(w(V)), 
    \end{equation*}

    where $\operatorname{diag}(w(V))$ is a diagonal matrix with the node scores in its diagonal. We scored the nodes as in cite:nakka_gene_2016, assigning a score of 0 for the genes with low probability of being associated to the disease, and -log_{10}(P-value) to those likely to be. Both are separated using a local FDR approach cite:scheid_twilight;_2005 to decide on the threshold, which for this dataset was set to 0.125. To obtain densely connected subnetworks, HotNet2 prunes E, only preserving edges such that w(E) \gt \delta. Lastly, HotNet2 evaluates the statistical significance of the subnetworks by comparing their size to the size of networks obtained by permuting the node scores. HotNet2 has two parameters: the restart probability \beta, and the threshold heat \delta. Both parameters are set automatically by the algorithm, and are robust cite:leiserson_pan-cancer_2015. HotNet2 is implemented in Python cite:hotnet2. 

- LEAN :: LEAN searches disregulated "star" gene subnetworks that is to say subnetworks composed by one central node and all its interactors cite:gwinner_network-based_2016. By imposing this restriction, LEAN is able to exhaustively test all such subnetworks (one per node). For a particular subnetwork of size /m/, the P-values corresponding to the involved nodes are ranked as p_1 \le \dots \le p_m. Then, /k/ binomial tests are conducted, to compute the probability of having /k/ out of /m/ P-values lower or equal to p_k under the null hypothesis. The minimum of these /k/ P-values is the score of the subnetwork. This score is transformed into a P-value through an empirical distribution obtained via a subsampling scheme, where sets of /m/ genes are selected randomly, and their score computed. Lastly, P-values are corrected for multiple testing through a Benjamini-Hochberg correction. We used the implementation of LEAN from the LEANR R package cite:leanr.
- SConES :: SConES searches the minimal, maximally interconnected, maximally associated subnetwork in a SNP graph cite:azencott_efficient_2013. Specifically, it solves the problem

    #+NAME: eq:scones
    \begin{equation}
    \underset{S \subseteq G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} + \underbrace{\lambda \sum_{v \in V_S} \sum_{u \not\in V_S} L_{vu} }_{\text { connectivity }}-\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}
    \end{equation}

    where \lambda and \eta are hyperparameters that control the sparsity and the connectivity of the model. For two hyperparameters, the aforementioned problem has a unique solution, that SConES finds using a graph min-cut procedure. We used the version on SConES implemented in the R package martini cite:martini. As in cite:azencott_efficient_2013, we selected \lambda and \eta by cross-validation, choosing the values that produce the most stable solution across folds. Note that the solution to the above problem can consist of several connected subnetworks which are disconnected from each other. In this case, the selected hyperparameters were \eta = 3.51, \lambda = 210.29 for SConES GS; \eta = 3.51, \lambda = 97.61 for SConES GM; and \eta = 3.51, \lambda = 45.31 for SConES GI.

# TODO Comment similarity with heinz

- SigMod :: SigMod aims at identifying the most densely connected gene subnetwork that is most strongly associated to the phenotype cite:liu_sigmod:_2017. It addresses an optimization problem similar to that of SConES (Equation [[eq:scones]]), but using the Laplacian matrix rather than the adjacency matrix (Section [[#methods:notation]]), to quantify solutions containing many edges.  

    \begin{equation*}
    \underset{S \in G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} + \underbrace{\lambda \sum_{v \in V_S} \sum_{u \in V_S} W_{vu} }_{\text { connectivity }} -\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}.
    \end{equation*}
  
    As SConES, this optimization problem can also be solved by a graph min-cut approach. 

    SigMod presents three important differences with SConES. First it is designed for gene-gene networks. Second, by replacing the Laplacian by the adjacency matrix, it favors subnetworks containing many edges. SConES, instead, penalizes connections between the selected selected and unselected nodes. Third, it returns a single connected subnetwork, which it achieves by exploring a grid of hyperparameters and processing their respective solutions. Specifically, for the range of \lambda = \lambda_{min}, \dots, \lambda_{max} for the same \eta, it prioritizes the solution with the largest change in size from \lambda_n to \lambda_{n+1}. Such a large change implies that the network is strongly interconnected. This results in one candidate solution for each \eta, which are processed by removing any node not connected to any other. A score is assigned to each candidate solution by summing their node scores and normalizing by size. The candidate solution with the highest standardized score is the chosen solution. SigMod is implemented in an R package cite:sigmod. 

*** Consensus network
    :PROPERTIES:
    :CUSTOM_ID: methods:consensus
    :END:
The different high-weight subnetwork discovery algorithms make different assumptions on the properties of the solutions, and employ different strategies to find them. Hence, combining the outcome of the different approaches might provide a more complete outlook on the specific alterations on the GENESIS dataset. We built such consensus network by retaining the nodes that were selected by at least two of the methods. We combined the results of 6 methods: heinz, HotNet2, dmGWAS, LEAN, SConES GI, and SigMod. 

** Validation of selected biomarkers
*** Classification accuracy of selected biomarkers
    :PROPERTIES:
    :CUSTOM_ID: methods:comparison
    :END:
A desirable solution is one that is sparse, while offering a good predictive power on unseen samples. We evaluated the predicting power of the SNPs selected by the different methods through the performance of an L1-penalized logistic regression trained exclusively on those SNPs to predict the outcome (case/control). The L1 penalty helps to account for LD to reduce the number of SNPs included in the model (size of the active set), while improving the generalization of the classifier. The value of the regularization parameter, which controls both the magnitude and the sparsity of the coefficients, was set by cross-validation. To that end, we used the different network-methods on a random subset of 80% of the samples. On this same subset we trained our classifier exclusively on the SNPs selected by a particular method. When the method retrieved a list of genes (all of them except SConES), all the SNPs mapped to any of those genes were used. Then we evaluated performance of the classifier on the remaining 20% of the dataset. We repeated this procedure 5 times to estimate the average and the deviation of the different performance measures. The different performance measures we used where: size of the solution, size of the active set, specificity, sensitivity, and stability. The size of the active set provides an estimate of a plausible, more sparse solution with a comparable predictive power to the original solution. The stability of an algorithm is its sensitivity to small changes of the input, and is measured using the Pearson's correlation between different runs as suggested in cite:nogueira_measuring_2016. To obtain a baseline, we also performed the procedure using all the SNPs. Another desirable property is that the method retrieves a good candidate causal subnetwork. In consequence, we compared the outcome of each of the methods to the consensus subnetwork of all the solutions (Section [[#methods:consensus]]). 

*** Biological relevance of the genes
    :PROPERTIES:
    :CUSTOM_ID: methods:bcac
    :END:
An alternative way to validate the results is comparing our results to an external dataset. For that purpose, we recovered a list of 153 genes associated to familial breast cancer from DisGeNET cite:pinero_disgenet:_2017. Across this article we refer to these genes as /familial breast cancer genes/.
 
Additionally, we used the summary statistics from the Breast Cancer Association Consortium (BCAC) cite:michailidou_genome-wide_2015. BCAC is one of the largest efforts in GWAS, with over 120000 women from European ancestry, albeit from different countries. As opposed to GENESIS, samples were not selected based on family history, and hence is enriched in sporadic breast cancers. Another difference is that BCAC is a relatively heterogeneous study on a pan-European sample, while GENESIS is a homogeneous dataset focused on the French population. Despite these differences, there should be shared genetic architecture. On top of that, that overlap should become larger when the results are aggregated at the gene level. For that purpose, we computed the gene association as in Section [[#methods:node_score]]. iCOGS array was used for genotyping in BCAC cite:sakoda_turning_2013, the same array as for GENESIS cite:sinilnikova_genesis:_2016. Although imputed data is available, we used exclusively the SNPs available on GENESIS after quality control to make the results comparable.

** Code availability

This work required developing computational pipelines for several GWAS analyses, such as physical mapping of SNPs, computing gene scores, and perform six different network-based analyses. For each of those processes, a streamlined, project-agnostic pipeline with a clear interface was created. They are compiled in the following GitHub repository: https://github.com/hclimente/gwas-tools. The code that applies these pipelines to the GENESIS project, as well as the code that reproduces all the analyses in this article are available at https://github.com/hclimente/genewa. Although the GENESIS data is not publicly available, the published code should work on any other GWAS dataset. All the produced gene subnetworks were deposited on NDEx (http://www.ndexbio.org), under the UUID e9b0e22a-e9b0-11e9-bb65-0ac135e8bacf.

* Results
** FGFR2 is strongly associated with familial breast cancer 
    :PROPERTIES:
    :CUSTOM_ID: results:conventional
    :END:

#+CAPTION:Association in GENESIS. The red line represents the Bonferroni threshold. *(A)* SNP association, measured from the outcome of a 1df \chi^2 allelic test. Significant SNPs that are within a coding gene, or within 50 kilobases of its boundaries, are annotated. The Bonferroni threshold is 2.54 \times 10^{-7}. *(B)* Gene association, measured by P-value of VEGAS2v2 cite:mishra_vegas2:_2015 using the 10% of SNPs with the lowest P-values. The Bonferroni threshold is 1.53 \times 10^{-6}.
#+NAME: fig:snp_gene_manhattan
[[./figures/figure_1.pdf]]

We conducted association analyses in the GENESIS dataset at both the SNP and the gene levels (Section [[#methods:node_score]]). Two genomic regions have a P-value lower than the Bonferroni threshold in chromosomes 10 and 16 (Figure [[fig:snp_gene_manhattan]]A). The former overlaps with gene FGFR2; the latter with CASC16, and its located near the protein-coding gene TOX3. Variants in both FGFR2 and TOX3 were related to breast cancer susceptibility in other cohorts negative for BRCA1/2 cite:rinella_genetic_2013. Only the peak in chromosome 10 replicated in the gene-level analysis, with FGFR2 just above the threshold of significance (Figure [[fig:snp_gene_manhattan]]B). 

These results show the overlap in the genetic architecture of the disease between the French population and other cohorts, especially at the gene level. In addition, there are other SNPs whose P-values, although higher than the conventional threshold of significance, show a strong association with familial breast cancer. The most prominent of such regions are 3p24 and 8q24, both of which  which have been associated to breast cancer susceptibility in the past cite:brisbin_meta-analysis_2011,search_newly_2009. This motivates exploring network methods, which trade statistical significance for biological relevance.

** Network methods successfully identify genes linked to breast cancer
    :PROPERTIES:
    :CUSTOM_ID: results:separate_networks
    :END:

#+CAPTION: Summary statistics on the results of multiple network methods on the gene-gene interaction network. The first row contains the summary statistics on the whole network.
#+NAME: tab:gene_solutions
| Network   | # genes | # edges | \mean{betweenness} | \median{P}_{gene} | \rho_{consensus} |
|-----------+---------+---------+--------------------+-------------------+------------------|
| HINT HT   |   13619 |  142541 |              16706 |              0.46 |            0.066 |
|-----------+---------+---------+--------------------+-------------------+------------------|
| Consensus |      55 |     117 |              74062 |            0.0051 |                1 |
| dmGWAS    |     194 |     450 |              49115 |              0.19 |             0.41 |
| heinz     |       4 |       3 |             113633 |            0.0012 |             0.21 |
| HotNet2   |     440 |     374 |               7739 |             0.048 |             0.31 |
| LEAN      |       0 |       0 |                  - |                 - |                - |
| SConES GI |   0 (1) |       0 |                  - |                 - |                - |
| SigMod    |     142 |     249 |              92603 |            0.0083 |             0.73 |
# Add as footnote: /# genes/: number of genes selected out of those that are part of the HINT network; in parentheses, the total number of genes. \mean{Betweenness}: mean betweenness of the selected genes in the full network. \median{P}_{gene}: median P-value of the selected genes. \rho_{consensus}: Pearson's correlation between the subnetwork and the consensus network.

We applied six network methods to the GENESIS dataset (Section [[#methods:methods]]), obtaining six solutions (Supplementary Figure [[sfig:overview_solutions]], Supplementary Files 1 and 2): one for each of the five gene-based methods (Section [[#methods:gene_network]]), and one for SConES GI (Section [[#methods:snp_network]]). The solutions are very heterogeneous (Table [[tab:gene_solutions]] and Supplementary table [[tab:snp_solutions]]): none of the subnetworks examined by LEAN is significant (adjusted P-value < 0.05), while HotNet2 produced the largest solution subnetwork with 440 genes. SConES GI failed to recover genes in the PPIN, but it recovered one genomic region mapped to RNA gene RNU6-420P. All solution subnetworks except LEAN's are, on average, more strongly associated to familial breast cancer than the whole PPIN (median P-values <<< 0.46), despite containing genes with much higher P-values (Supplementary Figure [[sfig:manhattan_solutions]]), exemplifying the trade-off between statistical significance and biological relevance. However, in this regard there are nuances: heinz strongly favored highly associated genes, while dmGWAS is less conservative (median gene P-values 0.0012 and 0.19, respectively); SConES tended to select whole LD-blocks; and HotNet2 and SigMod were less likely to select lowly associated genes. 

The solution subnetworks present other desirable properties. First, four of the methods succeeded at recovering genes involved in the disease (Supplementary Figure [[sfig:disgenet_comparison]]), as their subnetworks were enriched in familial breast cancer genes (dmGWAS, heinz, HotNet2, and SigMod, Fisher's exact test one-sided P-value < 0.03). We also compared the outcome of the network methods to the association tests conducted on the European cohort of the Breast Cancer Association Consortium (BCAC) cite:michailidou_genome-wide_2015 (Supplementary Figure [[sfig:bcac_comparison]]). Encouragingly, every solution subnetwork is enriched in genes or SNPs that are Bonferroni-significant in BCAC. This confirms the capability of network methods to find the same signal as in more powered studies by leveraging on prior knowledge. Second, the genes in three solution subnetworks display on average a higher betweenness centrality than the rest of the genes, a difference that is significant in three solutions (dmGWAS, and SigMod, Wilcoxon rank-sum test P-value < 1.4 \times 10^{-21}). This agrees with the notion that disease genes are more central than other, non-essential genes cite:pinero_uncovering_2016. We observe that this conclusion holds in this disease, as familial breast cancer genes have higher betweenness centrality than others (one-tailed Wilcoxon rank-sum test P-value = 2.64 \times 10^{-5}, Supplementary Figure [[sfig:consensus_stats]]C). Interestingly, SConES' selected SNPs are also more central than the average SNP (Supplementary table [[tab:snp_solutions]]), suggesting that causal SNPs are also more central than unrelated SNPs. However, very central nodes are also more likely to be connecting a random pair of nodes, making then more likely to be selected by the examined methods. Hence, further work is needed draw conclusions.

As the solutions were quite different from each other it is hard to draw joint conclusions. The 4-gene solution selected by heinz includes the familial breast cancer gene TOX3, in region 16q12. By dealing with SNP networks, SConES studies the association of non-coding regions, as well as SNPs in any gene, coding or else. In fact, SConES GI, which adds to GM the interactions between genes, retrieves 4 subnetworks in intergenic regions, and 1 overlapping an RNA gene (RNU6-420P). SigMod, despite being related to SConES, produces a vastly different, large solution. On top of recovering three familial breast cancer genes, a keratin-based region of its subnetwork affects the cytoskeleton (/structural constituent of cytoskeleton/, GO enrichment's adjusted P-value = 9.10 \times 10^{-4}), a potentially novel susceptibility mechanisms to cancer. Interestingly, dmGWAS solution is also related to cytoskeleton (/tubulin binding/, GO enrichment's adjusted P-value = 0.031). But, additionally, it includes a submodule of proteins related to /unfolded protein binding/ (GO enrichment's adjusted P-value = 0.045), which was related to cancer susceptibility cite:calderwood_heat_2016. Lastly, HotNet2 produced 135 subnetworks, 115 of which have less than five genes. The second largest subnetwork (13 nodes), contains two familial breast cancer genes: CASP8 and BLM.

** heinz retrieves a small, highly informative set of biomarkers in a fast and stable fashion

#+CAPTION:Comparison of network-based GWAS methods on GENESIS. Each method was run 5 times of a random subset of the samples, and tested on the remaining samples (Section [[#methods:comparison]]). *(A)* Number of SNPs selected by each method and number of SNPs on the active set used by the Lasso classifier. Points are the average over the 5 runs; lines represent the standard error of the mean. A grey diagonal line with slope 1 is added for comparison. For reference, the active set of Lasso using all the SNPs included, on average, 154117.4 SNPs. *(B)* Sensitivity and specificity on test set of the L1-penalized logistic regression trained on the features selected by each of the methods. In addition, the performance of the classifier trained on all SNPs is displayed. Points are the average over the 5 runs; lines represent the standard error of the mean. *(C)* Pairwise Pearson's correlations of the solutions used by different methods. A Pearson's correlation of 1 means the two solutions are the same. A Pearson's correlation of 0 means that there is no SNP in common between the two solutions. *(D)* Runtime of the evaluated methods, by type of network used (gene or SNP). For gene network-based methods, inverted triangles represent the runtime of the algorithm itself, and circles the total time, which includes the algorithm themselves and the additional 119980 seconds (1 day and 9.33 hours) which took VEGAS2 on average to compute the gene scores from SNP summary statistics.
#+NAME: fig:benchmark
[[./figures/figure_4.pdf]]

As the methods produced such different results, we compared their solutions in a 5-fold subsampling setting (Section [[#methods:comparison]]). Specifically, we measured the following properties (Figure [[fig:benchmark]]): (i) size of the solution subnetwork; (ii) stability; (iii) sensitivity and specificity of an L1-penalized logistic regression on the selected SNPs; and (iv) computational runtime.

Both solution size and active set of SNPs selected by Lasso varies greatly between the different methods (Figure [[fig:benchmark]]A). heinz has the smallest solutions, with an average of 182 selected our of which 5.6% (10.2) are selected by Lasso. The largest solutions come from SConES GM (4548.6 SNPs), and dmGWAS (4307.4 SNPs). Interestingly, SigMod and SConES GI have the highest proportion of the selected SNPs that go into the active set (11.47 and 10.3% respectively). This suggests those methods are selecting more informative SNPs on average. 

The sensitivity and specificity of the classifier on the testing data informs us about the usefulness of the selected SNPs as patient classification (Figure [[fig:benchmark]]B). All classifiers' sensitivities were in the 0.38 -- 0.69 range; the specificities, between 0.40 and 0.70. On average, SConES GS had the highest sensitivity (0.57); heinz, the highest specificity (0.56). Both SConES GS and SConES GM had on average better sensitivity than the classifier trained on all the SNPs, and dmGWAS and heinz superior specificities. However, the differences them were negligible, well within the 95% confidence interval.   

Another desirable quality of an algorithm is stability (Section [[#methods:comparison]]). Both heinz and LEAN displayed a high stability in our benchmark, consistently selecting the same genes and no genes over the 5 subsamples, respectively (Figure [[fig:benchmark]]C). The other methods displayed similarly low stabilities. 

In terms of computational runtime, the fastest method was heinz (Figure [[fig:benchmark]]D), which leverages on its ability to find efficiently the solution in a few seconds. The slowest method was SConES using the GI network, with approximately 1 day and 2.38 hours on average. Including the time required to compute the gene scores, however, slows down considerably gene-based methods; on this benchmark, that step took on average 1 day and 9.33 hours. Considering that time, dmGWAS is the slowest method, taking 1 day and 21.81 hours on average. 

** No solution is perfect 

#+CAPTION: Drawbacks confronted when using network guided methods. *(A)* dmGWAS solution subnetwork. Genes with a P-value < 0.1 are highlighted in red. *(B)* Centrality degree and -log_10 of the VEGAS P-value for the nodes in SigMod solution subnetwork. *(C)* Genomic regions where either SConES GS, GM or GI select SNPs. 
#+NAME: fig:issues
[[./figures/figure_2.pdf]]

In practice, and despite their similarities and their involvement in cancer mechanisms, the solutions are remarkably different (Supplementary Figure [[sfig:pearson_methods]]A). That is due to the particulars of the methods, and directly or indirectly, they provide information about the dataset. For instance, the fact that LEAN did not provide any biomarkers implies that there is no gene such that both itself and its environment are on average strongly associated with the disease. 

In this dataset, heinz's solution is very conservative, providing a small solution with the lowest median P-value for the subnetwork (Table [[tab:gene_solutions]]). Due to this parsimonious and highly associated solution, it was the best method to select a set of good biomarkers for classification. (Figure [[fig:benchmark]]B). Its conservativeness stems from its preprocessing step, which models the gene p-values as a mixture model of a beta and a uniform distribution, controlled by an FDR parameter. Due to the limited signal at the gene level in this dataset (Figure [[fig:snp_gene_manhattan]]B), only 36 of them are retain a positive score after applying the BUM model (Section [[#methods:methods]]). Hence, heinz's solution subnetwork consists only of 4 genes, which does not provide much insight of the biology of cancer. Importantly, it ignores genes that are strongly associated to cancer in this dataset like FGFR2. 

On the other end of the spectrum, we have large solutions provided by dmGWAS, HotNet2, and SigMod. dmGWAS' subnetwork is the least associated subnetwork on average. This is due to the greedy framework it uses, which considers all nodes at distance 2 of the examined, and accepts weakly associated genes if they are linked to another, strongly associated one. This is exacerbated when the results of successive greedy searches are aggregated, leading to a large, tightly connected cluster of unassociated genes (Figure [[fig:issues]]A). SigMod displays the same tendency, as the most central genes are the least associated to the disease (Figure [[fig:issues]]B). This relatively low signal-to-noise ratio combined with the large solution requires additional analyses to draw conclusions, such as enrichment analyses. In the same line, HotNet2's subnetwork is even harder to interpret, being composed of 440 genes divided into 135 subnetworks. Lastly, SigMod misses some of the most strongly associated, familial breast cancer genes in the dataset, like FGFR2 and TOX3.

By virtue of using a SNP subnetwork, SConES analyzes each SNP in their context. It therefore selects SNPs in genes none of whose interactors are associated to the disease, as well as SNPs in non-coding regions or in non-interacting genes. In fact, due to linkage disequilibrium, such genes are favored by SConES, as selecting an SNPs in an LD block which overlaps with a gene favors selecting the rest of the gene. This might explain why the GS and GM networks, heavily affected by linkage disequilibrium, produce similar results (Supplementary Figure [[sfig:pearson_methods]]B). On the other hand, SConES penalizes selecting SNPs and not their neighbors. This makes it conservative regarding SNPs with many interactions, for instance those mapped to hubs in the PPIN. For this reason, SConES GI did not select any protein coding gene, despite selecting similar regions as SConES GS (Figure [[fig:issues]]C). In fact SConES GS and SConES GM select regions related to breast cancer, like 16q12 (TOX3, Section [[#results:conventional]]), 3p24 (SLC4A7/NEK10 cite:ahmed_newly_2009), 5p12 (FGF10, MRPS30 cite:quigley_5p12_2014), and 10q26 (FGFR2, Section [[#results:conventional]]). On top of that only SConES GS selects region 8q24 (POU5F1B cite:breyer_expressed_2014). We hypothesize that the lack of results on the PPIN network of SConES GI and LEAN due to the same cause: the absence of joint association of a module. Although in the case of SConES other hyperparameters could lead to a more informative solution (lower \lambda), finding the appropriate configuration is hard. In addition, the iCOGS platform is not a real GWAS experiment: the genome is not unbiasedly surveyed, some regions are fine-mapped --- which might distort gene structure in GM and GI networks --- while others are under studied - hurting the accuracy with which the GS network captures the genome structure. 

** Aggregating solutions provides insights into the biology of cancer 
    :PROPERTIES:
    :CUSTOM_ID: results:consensus
    :END:

#+CAPTION:Consensus subnetwork on GENESIS (Section [[#methods:consensus]]). Each node is represented by a pie chart, which accounts the methods that selected it. The labeled genes have a VEGAS2 P-value < 0.001 and/or are known familial breast cancer genes (colored in pink).
#+NAME: fig:consensus
[[./figures/figure_3.pdf]]

To leverage on the strengths of each of the methods and compensate their respective weaknesses, we built a consensus subnetwork that captures the mechanisms most shared among the solution subnetworks (Section [[#methods:consensus]]). The consensus subnetwork (Figure [[fig:consensus]]) contains 93 genes and is enriched in familial breast cancer genes (Fisher's exact test P-value = 7.8 \times 10^{-5}). Due to the limited overlap between methods, only 20 genes were common to more than two of them (Supplementary Figure [[sfig:consensus_stats]]A). Encouragingly, the more methods selected a gene, the higher its association was (Supplementary Figure [[sfig:consensus_stats]]B). Globally, a GO enrichment shows the involvement of two cellular processes: unfolded protein binding, and structural constituent of cytoskeleton (adjusted P-values of 0.001, 0.001, respectively), which were already observed in different solutions (Section [[#results:separate_networks]]). Remarkably, many of the selected genes are related to mitochondrial translation. For instance, MRPS30 (VEGAS P-value = 0.001), encodes a mitochondrial ribosomal protein and was also linked to breast cancer susceptibility cite:quigley_5p12_2014. Albeit disconnected from MRPS30, the consensus network includes a 2-node subnetwork composed of two mitochondrial ribosomal protein (MRPS31 - VEGAS P-value = 7.67 \times 10^{-3} - and MRPS18B - VEGAS P-value = 7.92 \times 10^{-3}), which suggests an involvement of mitochondrial ribosomes in carcinogenesis cite:required.
 
We also examined the topological properties of the nodes. The genes in the consensus network have higher betweenness centrality than the rest of the genes (Wilcoxon rank-sum test P-value = 4.29 \times 10^{-18}). Interestingly, within genes in the consensus network, cancer genes are as central as non-cancer genes (Wilcoxon rank-sum test P-value = 0.57). Centrality, however, is weakly anti-correlated with association to the disease (Pearson correlation coefficient = -0.26, Supplementary Figure [[sfig:consensus_stats]]D), which suggests that some highly central genes were selected because they were on the shortest path between two highly associated genes. In view of this, we hypothesize that highly central genes might contribute to the heritability through consistent alterations of their neighborhood, consistent with the omnigenic model of disease cite:boyle_expanded_2017. For instance, the most central node in the consensus network is COPS5 (Supplementary Figure [[sfig:consensus_names]]), a gene related to multiple hallmarks of cancer and which is overexpressed in multiple tumors, including breast and ovarian cancer cite:liu_jab1_cops5_2018. Despite its lack of association in GENESIS (VEGAS P-value = 0.22), its neighbors in the consensus subnetwork have consistently low P-values (median VEGAS P-value = 0.006).
 
The consensus subnetwork is not completely connected: out of the 93 genes, the largest connected subnetwork includes only 49. A GO enrichment analysis showed that this component is related to three major cellular processes: unfolded protein binding, structural constituent of cytoskeleton, and poly(U) RNA binding (adjusted P-values of 0.01, 0.04, and 0.04, respectively). We found support in the literature of the involvement of each of these functions in the development of cancer, as discussed next. The consensus network also contains a protein directly involved in caspase-mediated apoptosis, CASP8 (VEGAS P-value = 1.95 \times 10^{-4}). This is related to the enriched activity, /unfolded protein binding/, which inhibits caspase-dependent apoptosis, raising the chances of developing cancer cite:calderwood_heat_2016. It involves three Hsp70 chaperones of the consensus subnetwork: HSPA1A, HSPA1B, and HSPA1L. They all are closely encoded in 6p21. In fact out of the 22 SNPs that map to any of these three genes, 9 map to all of them, and 4 to two, making hard to disentangle their association. HSPA1A was the most strongly associated one (VEGAS P-value = 8.37 \times 10^{-4}). Remarkably, 14 of the 93 genes are in subnetworks of size 1 (isolated) or two, as they do not have a consistently altered neighborhood. One of them is the familial breast cancer FGFR2 (Section [[#results:conventional]]). Another one is SLC4A7 (VEGAS P-value = 2.70 \times 10^{5}), a gene encoding a sodium bicarbonate cotransporter. In the past the genomic region containing both SLC4A7 and nearby gene NEK10 (VEGAS P-value = 1.56 \times 10^{-5}) were associated with familial breast cancer cite:ahmed_newly_2009. NEK10 is a gene that might be involved in cell-cycle control, but it is absent from the PPIN and hence it could not be studied by gene methods. Despite that, the fact that both dmGWAS, HotNet2 and SigMod link SLC4A7 in their different subnetwork supports the notion that this gene is the responsible for cancer susceptibility. 
# TODO Say interactors of the subnetworks related to BRCA

** Limitations of network analyses

The strength of network-based analyses comes from leveraging prior knowledge to boost discovery. In consequence, they falter in front of understudied genes, especially those not in the network. Out of the 32767 genes that we can map the genotyped SNPs, 60.7% (19887) are not in the protein-protein interaction network. The majority of those (14660) are non-coding genes, mainly lncRNA, miRNA, and snRNA (Supplementary Figure [[sfig:biotypes_excluded]]). The importance of these genes, like CASC16, is highlighted in Section [[#results:conventional]]. Among the excluded protein-coding genes we find genes like NEK10 (P-value 1.6 \times 10^{-5}) or POU5F1B, both linked to breast cancer susceptibility cite:ahmed_newly_2009. However, on average protein coding genes absent from the PPIN are less associated with this phenotype (Wilcoxon rank-sum P-value = 2.79 \times 10^{-8}, median P-values of 0.43 and 0.47). As we are using interactions from high-throughput experiments, such difference cannot be due to well-known genes having more known interactions. As disease genes tend to be more central cite:pinero_uncovering_2016, we hypothesize that it is due to interactions between central genes being more likely. It is worth noting that network approaches that do not use PPIs, like SConES GS and GM, did recover SNPs in NEK10 and CASC16. Lastly, all the methods rely heavily on how SNPs are mapped to genes. In Section [[#results:conventional]] we highlight ambiguities that appear when genes overlap or are in linkage disequilibrium. 

As not all databases compile the same interactions, the choice of the PPIN determines the final output. In this work we used exclusively interactions from HINT from high-throughput experiments. This responds to concerns of some authors about biases introduced by adding interactions coming from targeted studies in the literature cite:cai_broker_2010,das_hint:_2012. It is a "rich getting richer" phenomenon, where popular genes have a higher proportion of their interactions described. On the other hand, one study found that the best predictor of the performance of a network for disease gene discovery is the size of the network cite:huang_systematic_2018. This would support using the largest amount of interactions. To clarify their impact on this study, we compared the impact of using only physical interactions from high-throughput experiment versus interactions from both high-throughput and the literature (Section [[#methods:gene_network]]). For most of the methods using a larger network did not greatly impact the size or the stability of the solution, the classification accuracy, or the runtime (Supplementary Figure [[sfig:lc_ht_comparison]]). 

* Discussion

In this article we evaluate the viability of systems biology approaches to GWAS, and examine a GWAS dataset on familial breast cancer focused on BRCA1/2 negative French women. Systems biology addresses two of the largest GWAS issues: interpretability and an overly conservative statistical framework that hinders discovery. This is achieved by considering the biological context of each of the genes and SNPs. Based on divergent considerations of what the desired set of biomarkers is, several methods for network-guided biomarker discovery have been proposed. In this article we reviewed the performance of six of them on GWAS. Despite their differences, most of them produced a relevant subset of biomarkers, recovering known familial breast cancer genes. We also discuss the limitations of such analyses, related to the lack of known interactions around some genes. A crucial step for the gene based methods is the computation of the gene score. In the manuscript we used VEGAS2 cite:mishra_vegas2:_2015 due to the flexibility it offers to use user-specified gene annotations. However, it presents known problems (selection of an appropriate percentage of top SNPs, long runtimes and P-value precision limited to the number of permutations cite:nakka_gene_2016), other algorithms might have more statistical power. 

The network methods we studied differ in what the optimal solution subnetwork looks like. On the one hand, SConES and heinz prefer small highly associated solutions. On the other hand, SigMod and dmGWAS gravitate towards larger, less associated solutions which provide a wide overview of the biological context. While the former provide a reduced set of biomarkers, the latter deepen our understanding of the disease and provide biological hypotheses. They are not exempt of limitations. dmGWAS and SigMod's solution's size require further analyses, which risk oversimplifying their richness. Also, incautious practitioners might be misled by some genes, which are very central in the solution subnetworks, while being weakly associated. Nonetheless, they are pushed into the solution by their privileged topological properties. On the other end, conservative solutions, like SConES GI and heinz might not shed much light on the etiology of the disease. 

To overcome the problems posed by the individual methods while exploiting their strengths, we propose combining them into a consensus subnetwork. We use a straightforward aggregation to generate it, including any node that was recovered by at least two methods. The resulting network is a synthesis of the altered mechanism: is smaller than the largest solutions (SigMod and dmGWAS), which makes it more manageable, and includes the majority of the strongly associated smaller solutions (SConES and heinz). The consensus subnetwork captures mechanisms and genes known to be related to cancer, recovering familial breast cancer genes as well as genome regions associated to cancer susceptibility. However, thanks to its small size and its network structure, it provides compelling hypotheses of non-canonical mechanisms involved in carcinogenesis, like mitochondrial translation and chaperone activity. 

In order to produce the consensus network, we had to face the different interfaces, preprocessing steps, and unexpected behaviors of the various methods. To facilitate that other authors apply them to new datasets and aggregate their solutions, we built six nextflow pipelines cite:di_tommaso_nextflow_2017 with a consistent interface and, whenever possible, parallelized computation. They are available on GitHub: https://github.com/hclimente/gwas-tools. Importantly, those methods that had a permissive license were compiled into a Docker image for easier use, which is available on Docker Hub [[https://hub.docker.com/r/hclimente/gwas-tools][hclimente/gwas-tools]].

* Acknowledgments

H.C-G. was funded by the European Union’s Horizon 2020 research and innovation program under the Marie Skłodowska-Curie [666003].

bibliographystyle:ieeetr
bibliography:bibliography.bib

#+LATEX: \clearpage
#+LATEX: \setcounter{figure}{0}
#+LATEX: \setcounter{section}{0}
#+LATEX: \setcounter{table}{0}

* Supplementary materials

#+CAPTION: Summary statistics on the results of SConES on the three SNP-SNP interaction networks. The first row within each block contains the summary statistics on the whole network.
#+NAME: tab:snp_solutions
| Network   |   SNPs |    Edges | Subnetworks | \mean{betweenness} | \median{P}_{SNP} |
|-----------+--------+----------+-------------+--------------------+------------------|
| GS        | 197083 |   197060 | -           | 2.03 \times 10^7   |             0.49 |
| SConES GS |   1590 |     1585 | 5           | 2.52 \times 10^7   |            0.023 |
|-----------+--------+----------+-------------+--------------------+------------------|
| GM        | 197083 |  6442446 | -           | 3.99 \times 10^6   |             0.49 |
| SConES GM |   1692 |   177611 | 5           | 4.40 \times 10^6   |            0.055 |
|-----------+--------+----------+-------------+--------------------+------------------|
| GI        | 197083 | 28733720 | -           | 1.46 \times 10^6   |             0.49 |
| SConES GI |    408 |      539 | 5           | 9.33 \times 10^6   |            0.076 |
# Add as footnote: \mean{betweenness}: mean betweenness of the selected SNPs in the corresponding full network;  \median{P}_{SNP}: median P-value of the selected SNPs.

#+CAPTION: Summary statistics on the results of multiple network methods on the gene-gene interaction network. The first row contains the summary statistics on the whole network.
#+NAME: tab:scones_gene_solutions
| Network   | Genes | Edges | Mean betweenness | Median P_{gene}    | Correlation_{consensus} |
|-----------+-------+-------+------------------+--------------------+-------------------------|
| SConES GS |     5 |     0 |             9805 | 2.7 \times 10^{-5} |                    0.19 |
| SConES GM |    28 |     2 |             4267 | 0.067              |                    0.12 |
# Add as footnote: \mean{betweenness}: mean betweenness of the selected genes in the full network;  \median{P}_{gene}: median P-value of the selected genes; \rho_{consensus}: Pearson's correlation with the consensus network.

#+CAPTION: GENESIS shows no differential population structure between cases and controls. *(A,B,C,D)* Eight main principal components computed on the genotypes of GENESIS. Cases are colored in green, controls in orange. 
#+NAME: sfig:pcs
[[./figures/sfigure_1.pdf]]

#+CAPTION: Overview of the subnetworks produced by the different network methods. *(dmGWAS, heinz, HotNet2, LEAN, and SigMod)* contain gene subnetworks; *(SConES GI)*, SNP subnetworks.
#+NAME: sfig:overview_solutions
[[./figures/sfigure_2.pdf]]

#+CAPTION: Manhattan plots showing the biomolecules selected by each method. In *(Consensus, dmGWAS, heinz, HotNet2, and SigMod)* datapoints are genes; in *(SConES GI)*, SNPs. LEAN was excluded, as it did not select any gene.  
#+NAME: sfig:manhattan_solutions
[[./figures/sfigure_3.pdf]]

#+CAPTION:Proportion of the Bonferroni significant biomolecules (in either the GENESIS or the BCAC datasets) selected by each of the methods on the GENESIS data. *(Consensus, dmGWAS, heinz, HotNet2, and SigMod)* involve significant genes, only among those present in the protein-protein interaction network. *(SConES GS, GM and GI)* involve significant SNPs. LEAN was excluded, as it did not select any gene. The presented network methods recover a higher proportion of significant genes than of non-significant genes in both datasets, despite their lack of significance in GENESIS.
#+NAME: sfig:bcac_comparison
[[./figures/sfigure_4.pdf]]

#+CAPTION: :Proportion of the selected genes by each of the methods on the GENESIS data that is a known familial breast cancer gene (Section [[#methods:bcac]]). Only genes present in the protein-protein interaction network were considered. LEAN was excluded, as it did not select any gene. The presented network methods recover a higher proportion of familial breast cancer genes than of other genes, despite their lack of significance in GENESIS.
#+NAME: sfig:disgenet_comparison
[[./figures/sfigure_5.pdf]]

#+CAPTION:Pearson's correlation between the different solution subnetworks. *(A)* Correlation between selected SNPs. *(B)* Correlation between selected genes. In general, the solutions display a very low overlap.
#+NAME: sfig:pearson_methods
[[./figures/sfigure_6.pdf]]

#+CAPTION: Consensus subnetwork on GENESIS (Section [[#methods:consensus]]). *(A)* Each node is represented by a pie chart, which accounts the methods that selected it. The labeled genes have a VEGAS2 P-value < 0.001 and/or are known familial breast cancer genes (colored in pink). This panel is equivalent to Figure [[fig:consensus]]. *(B)* The name of every gene is indicated.
#+NAME: sfig:consensus_names
[[./figures/sfigure_7.pdf]]

#+CAPTION: Genes on the consensus network. Familial breast cancer genes are colored in pink; the rest are colored in grey. *(A)* Number of methods selecting every gene in the subnetwork. *(B)* VEGAS P-values of association of the genes, with regards to the number of methods that selected them. *(C)* Comparison of betweenness centrality of the genes in the consensus network and the other genes in the PPIN and not in the consensus network. To improve visualization, we removed outliers. *(D)* Relationship between the log_10 of the betweenness centrality and the -log_10 of the VEGAS P-value of the genes in the consensus network. The blue line represents a fitted generalized linear model. 
#+NAME: sfig:consensus_stats
[[./figures/sfigure_8.pdf]]

#+CAPTION:Biotypes of genes from the annotation that are not present in the HINT protein-protein interaction network.
#+NAME: sfig:biotypes_excluded
[[./figures/sfigure_9.pdf]]

#+CAPTION:Comparison of benchmark on high-throughput interactions to benchmark on both high-throughput and literature curated interactions. Grey lines represent no change between the benchmarks (1 for ratios, 0 for differences). *(A)* Ratios of the selected features between both benchmarks and of the active set. *(B)* Shifts in sensitivity and specificity. *(C)* Shift in Pearson's correlation between benchmarks. *(D)* Ratio between the runtimes of the benchmarks. For gene network-based methods, inverted triangles represent the ratio of runtimes of the algorithms themselves, and circles the total time, which includes the algorithm themselves and the additional 119980 seconds (1 day and 9.33 hours) which took VEGAS2 on average to compute the gene scores from SNP summary statistics. In general, adding additional interactions slightly improves the stability of the solution, but increases the solution size, has mixed effects on the sensitivity and specificity, and impacts negatively the required runtime of the algorithms.
#+NAME: sfig:lc_ht_comparison 
[[./figures/sfigure_10.pdf]]




