#+TITLE: Aggregation of network-guided GWAS to find the susceptibility network to familial breast cancer
#+AUTHOR: Héctor Climente-González, Christine Lonjou, Chloé-Agathe Azencott
#+OPTIONS: toc:nil num:nil

* TODO Introduction

In human health, the goal of genome-wide association studies (GWAS) is uncovering the genetic variants that predispose to complex diseases, like diabetes or some variants of cancer. To that end, thousands of unrelated samples are genotyped, a fraction of them with the disease of interest, and the rest from the general population. In the traditional setting, a per-SNP statistical test (i.e. \chi^2) is conducted, and only those SNPs with a P-value lower than a conservative threshold are the selected ones.

This experimental design has been successful in identifying the main variants underlying many common diseases, allowing early diagnosis. However, the experimental setting also presents intrinsic challenges. The first one is the high-dimensionality of the problem: all GWAS studies to date genotype more positions in the genome than samples are used. This limits the statistical power of the experiment, as only variants with large and moderate effects can be detected. Additionally, to avoid false positives, a conservative multiple test correction is applied, typically Bonferroni. The second problem is the interpretability of the results, when SNPs in non-coding regions or in large LD-blocks are selected. 

On top of the aforementioned problems, there is evidence suggesting that most of the genome contributes to a degree to any complex trait. This opposes to the traditional polygenic model of complex diseases, which postulated that only key genes in key pathways need to be affected. In the recently postulated omnigenic model, genes are highly interconnected, and influence each others behavior. Hence, a more comprehensive statistical framework is required.

Network-based methods offer a compact and elegant solution to both problems. In essence, they represent the experiment as a network where the nodes contain information from the experiment, and the edges represent functional relatedness between them. A common paradigm consists on aggregating the SNP association into gene association, and relate the genes according to whether their proteins physically interact. The problem involves the selection of one or more subnetworks, densely intra-connected, which contains the highest-associated nodes. 

Hierarchichal HotNet categories:

- Constrain the problem to certain kinds of subnetworks, then explore those subnetworks: dmGWAS, heinz, LEAN.
- Consider jointly the node scores and topology: SConES, HotNet 2, Sigmod.
- Use additional information: Hierarchical Hotnet.

Check classification in Network-guided biomarker discovery.

In this article we focus on GENESIS cite:sinilnikova_genesis:_2016, a homogeneous dataset of familial breast cancer in French population of European ancestry. We apply a representative set of methods, some of them never applied to GWAS. Our goal is two-fold: we compare the methods in this dataset; and we explore new biomarkers of familial breast cancer. 

* Methods

** GENESIS

GENESIS is a GWAS study of familial breast cancer on the French population cite:sinilnikova_genesis:_2016. The index cases are patients with infiltrating mammary or ductal adenocarcinoma, who had sister with breast cancer, and negative for BRCA1 and BRCA2 mutations. The controls are unaffected colleagues and/or friends of the cases, born around the year of birth of the corresponding case (\pm 3 years). In total, the GENESIS dataset consists of 2577 samples, of which 1279 are controls and 1298 are cases. The genotyping platform was iCOGS, a custom Illumina array designed to study genetic susceptibility of hormone-related cancers cite:sakoda_turning_2013. It contains 210918 SNPs previously associated with cancer susceptibility, survival, or other cancer-related traits; and in selected genes and pathways.

** Preprocessing and quality control

We discarded SNPs with a minor allele frequency lower than 0.1%, not in Hardy - Weinberg equilibrium (P-value \textless 0.001), and/or those with missing values on more than 10% of the samples. In addition, we removed the samples with more than 10% missing genotypes. 28 samples with ?? were removed. A subset of 20 duplicated SNPs in FGFR2 were also removed. The final dataset included 1271 controls and 1280 cases, genotyped over 197083 SNPs. 

We looked for population structure that could create confounding associations. A PCA revealed no differential population structure between cases and controls (Supplementary Figure [[sfig:pcs]]). Independently, we did not find evidence of genomic inflation (\lambda = 1.05029) either, thus further dismissing the presence of confounding population structure.

** High-weight subnetwork discovery algorithms
*** Node scores
    :PROPERTIES:
    :CUSTOM_ID: methods:node_score
    :END:
To measure association between a genotype and the phenotype, we performed a per-SNP 1df \chi^2 allelic test using PLINK v1.90 cite:chang_second-generation_2015. Then, we used VEGAS2v2 to compute the gene-level association score cite:mishra_vegas2:_2015 from the SNP P-values. In order to map SNPs to genes we relied used their overlap on the sequence: all SNPs located within the boundaries of a gene, \pm 50 kb, were mapped to that gene. To compute the gene association we used the 10% of SNPs with lowest P-values.

*** Network

We used the physical interactions from BioGRID version 3.5.171 cite:oughtred_biogrid_2019 to get protein-protein interactions when required. Out of the six methods tested, five use a protein-protein interaction network.

*** Mathematical notation

In this article, we refer to undirected, vertex-weighted graphs G = (V,E,w). V = {v_1, \dots, v_n} refers to the vertices, with weights w: V \rightarrow \mathbb{R}. Equivalently, E \subseteq {{x,y} | x,y \in V \wedge x \neq y} refers to the edges. When referring to a subnetwork S, V_S is the set of nodes in S and E_S is the set of edges in S.

In addition, we use several matrices that describe different properties of a graph. The described matrices are square, and have as many rows and columns as nodes are in the network. In fact, the element i,j represent a  selected relationship between v_i and v_j. The adjacency matrix W_G contains a 1 when the corresponding nodes are connected through an edge, and 0 otherwise; the diagonal is zero. The degree matrix D_G is a diagonal matrix which contains the degree of the different nodes. Lastly, the Laplacian matrix L_G is defined as L_G = D_G - W_G.

*** TODO Methods used

#+CAPTION: Summary of the differences between the studied algorithms. 
#+NAME:   tab:method_comparison
| Algorithm           | Purpose | Network     | Exhaustive | Solution properties |
|---------------------+---------+-------------+------------+---------------------|
| heinz               |         | PPIN        | Yes?       | Dense?              |
| Hierarchical HotNet |         | PPIN        |            |                     |
| dmGWAS              | GWAS    | PPIN        | No         | Dense               |
| LEAN                | Omics   | PPIN        | Yes        | Star-shaped         |
| SConES              | GWAS    | SNP network | Yes        | Dense, connected    |
| SigMod              | GWAS    | PPIN        |            |                     |

Finding the highest-scoring, densely interconnected subnetwork on a graph is an open problem in the field. Hence, several solutions have been proposed to the problem. In this paper, we apply six methods designed to explore the protein-protein interaction network, and one method, SConES, which explores SNP-networks. Their main differences are summarized in Table [[tab:method_comparison]].

+ heinz :: The goal of heinz is identifying the highest-scored connected subgraph on the network cite:dittrich_identifying_2008. The problem has a trivial solution when all scores are positive: the whole network; however, it becomes NP-complete when scores are both positive and negative. The authors propose a transformation of the nodes' P-value into a score which takes a negative value when no association with the phenotype is detected, and a positive value when it is. The distinction between both is determined though an FDR approach. Then, the problem is re-casted as the Prize-Collecting Steiner Tree Problem (PCST). This is the problem of selecting the subnetwork S that maximizes the /profit/ p(S):

    \begin{equation}
    p(S) = \sum_{v \in V_S} p(v) - \sum_{e \in E_S} c(e). 
    \end{equation}

    were p(v) is called profit of adding a node, and c(e) is the cost of the edge, both positive values. These quantities are defined from w' = min_{v \in V_G} w(v):

    \begin{multiline}
    p(v) = w(v) - w', \\
    c(e) = w'.
    \end{multiline}

    PCST has a heuristic, efficient solution cite:ljubic_algorithmic_2006. We used the implementation of heinz from BioNet cite:beisser_bionet:_2010, available on Bioconductor.
+ Hierarchichal HotNet :: Hierarchical HotNet was developed in the context of cancer driver identification cite:reyna_hierarchical_2018. Its aim is identifying statistically significant subnetworks, made up by nodes with high-scores that are close in the network. It does so by performing hierarchical clustering on a similarity matrix that captures both the node score similarity and the network similarity. Specifically, it uses the stationary distribution of a random walk with restart procedure to capture a graph-based vertex similarity. By setting up the appropriate restart probability, we tune the locality of this similarity. 

    Interestingly, Hierarchical HotNet addresses the bias of biological graphs, where genes with high scores tend to have higher degrees, as they are better studied. 
    
+ dmGWAS :: dmGWAS aims at identifying the connected subgraph with the largest amount of low P-values cite:jia_dmgwas:_2011. To that end, it first searches several candidate subnetwork solutions using a greedy procedure involving the following steps:

  1. Select a seed node.
  2. Compute Stouffer's Z-score Z_m for the current subgraph as
    
    \begin{equation} 
    Z_m = \frac{\sum z_i}{\sqrt{k}}
    \end{equation}

    where /k/ is the number of genes in the subgraph, z_i = \phi^{-1}(1 - P_i), and \phi^{-1} is the inverse normal distribution function.
  3. Identify neighboring nodes i.e. nodes at shortest path \le /d/. We set d = 2.
  4. Add the neighboring nodes whose inclusion increases the Z_{m+1} more than Z_m \times (1 + r). In our experiments, we set r = 0.1.
  5. Repeat 2-4 until no increment Z_m \times (1 + r) is possible.
  
  Lastly, the module's Z-score is normalized as

  \begin{equation}
  Z_{N}=\frac{Z_{m}-\operatorname{mean}\left(Z_{m}(\pi)\right)}{\operatorname{SD}\left(Z_{m}(\pi)\right)}
  \end{equation} 

  where Z_{m}(\pi) represent a vector with 100000 random subsets of the same number of genes.

  We used the implementation of dmGWAS in the dmGWAS 3.0 R package. We used the function /simpleChoose/ to select the solution subnetwork, which aggregates the top 1% modules into the solution subnetwork.
+ LEAN :: Local enrichment analysis (LEAN) searches disregulated "star" gene subnetworks i.e. subnetworks composed by one central node and all its interactors cite:gwinner_network-based_2016. By imposing this restriction, LEAN is able to exhaustively test all possible solution subnetworks (one per node in the network). For a particular subnetwork of size /m/, the P-values corresponding to the involved nodes are ranked as p_1 \le \dots \le p_m. Then, /k/ binomial tests are conducted, to compute the probability of having /k/ out of /m/ P-values lower or equal to p_k under the null hypothesis. The minimum of these /k/ P-values is the score of the subnetwork. This score is transformed into a P-value through an empirical distribution obtained via a subsampling scheme, where sets of /m/ genes are selected randomly, and their score computed. Lastly, P-values are corrected for multiple testing through a Benjamini-Hochberg correction. We used the implementation of LEAN from the LEANR R package.
+ SConES :: SConES searches the minimal, maximally interconnected, maximally associated subnetwork in a SNP graph cite:azencott_efficient_2013. Specifically, it solves the problem

    \begin{equation}
    \underset{S \in G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }}-\underbrace{\lambda \mathbf{1}^{\top} L_S \mathbf{1}}_{\text { connectivity }}-\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}
    \end{equation}

    where \mathbf{1} is the one-vector of size |V_S|. To build a SNP-graph, the authors suggested three networks: connecting the genotyped SNPs consecutive in the genomic sequence (GS); interconnecting all the SNPs mapped to the same gene, on top of GS (GM); and interconnecting all SNPs mapped to two genes for which a protein-protein interaction exists (GI). 

+ SigMod ::

*** Consensus network
    :PROPERTIES:
    :CUSTOM_ID: methods:consensus
    :END:
The different high-weight subnetwork discovery algorithms make different assumptions on the nature of the solutions, and employ different strategies to find them. Hence, combining the outcome of the different approaches might provide a more complete outlook on the specific alterations on the GENESIS dataset. We built such consensus network by retaining the nodes that were selected by at least two of the methods. We combined the results of 6 methods: heinz, Hierarchical HotNet, dmGWAS, LEAN, SConES on the GM network, and SigMod.

** Validation of selected biomarkers
*** Classification accuracy of selected biomarkers

To evaluate the quality of the solutions offered by the different algorithms, we used their predictor power. We reasoned that a desirable solution is one that is sparse, while offering a good predictor power. To evaluate the predicting power of the SNPs selected by the different methods, we used the performance of an L1-penalized logistic regression trained exclusively on those SNPs to predict the outcome (case/control). The L1 penalty helps to account for LD to reduce the size of the active set, while improving the generalization of the classifier. To that end, we trained our predictor on a random 80% of the samples and all the SNPs selected by a particular method. When the method retrieved a list of genes (all of them except SConES), all the SNPs mapped to any of those genes were used. Then evaluated its performance on the remaining 20% of the dataset. We repeated this procedure 5 times to estimate the average and the deviation of the different performance measures. The different performance measures we used where: size of the solution, size of the active set, specificity and sensitivity. 

*** Biological relevance of the genes

Another desirable property is that the method retrieves a good candidate causal subnetwork. In consequence, we compared the outcome of each of the methods to the consensus subnetwork of all the solutions (Section [[#methods:consensus]]). 

An alternative way to validate the results involves looking at an external dataset. For that purpose, we used the summary statistics from the Breast Cancer Association Consortium (BCAC) cite:michailidou_genome-wide_2015. BCAC is one of the largest efforts in GWAS, with over 120000 women from European ancestry. Despite the studies not being performed on exactly the same genetic population, there should be shared heritability between a pan-European cohort and a French one. On top of that, that overlap should become more notorious when the results are aggregated at the gene level. For that purpose, we computed the gene association as in Section [[#methods:node_score]]. iCOGS array was used for genotyping in BCAC cite:sakoda_turning_2013, the same array as for GENESIS cite:sinilnikova_genesis:_2016. Although imputed data is available, we used exclusively the SNPs on iCOGS on both cases to make the results comparable.

** Code availability

The adaptation of the different network-based methods to GWAS datasets is available at https://github.com/hclimente/gwas-tools. The code necessary to reproduce all the analyses in this article is available at https://github.com/hclimente/genewa.

* Results
** Regions near long non-coding RNAs are associated with breast cancer 

#+CAPTION: Association in GENESIS. The red line represents the Bonferroni threshold. *(A)* SNP association, measured from the outcome of a 1df \chi^2 allelic test. SNPs that are within a coding gene, or within 50 kilobases of its boundaries are annotated. The Bonferroni threshold is 2.54 \times 10^{-7}. *(B)* Gene association, measured by P-value of VEGAS2v2 cite:mishra_vegas2:_2015 using the 10% of SNPs with the lowest P-values. The Bonferroni threshold is 1.51 \times 10^{-6}.
#+NAME: fig:snp_gene_manhattan
[[./figures/figure_1.pdf]]

We conducted association analyses both at the SNP level and at the gene level in the GENESIS dataset (Section [[#methods:node_score]]). Two genomic regions have a P-value lower than the Bonferroni threshold in chromosomes 10 and 16 (Figure [[fig:snp_gene_manhattan]]A), both of which overlap with long non-coding RNAs. The former with LINC02641 and CASC16. These peaks are replicated in the gene-level analysis, with both genes as well below the threshold of significance (Figure [[fig:snp_gene_manhattan]]B). The protein-coding gene TOX3, located near CASC16, is also below the significance threshold. 

Variants in TOX3 were related to breast cancer susceptibility in other cohorts cite:the_search_collaborators_genome-wide_2007. We did not find previous reports that directly involve LINC02641 in cancer susceptibility. These results show the overlap between the genetic architecture of the disease between the French population and other cohorts, especially at the gene level. In addition, there are other regions highly associated with familial breast cancer, albeit well above the conventional threshold of significance. The most prominent regions, which have been associated to breast cancer susceptibility in the past, are 3p24 cite:brisbin_meta-analysis_2011, and 8q24 cite:search_newly_2009. This motivates exploring network methods, which trade statistical association for biological significance.

** TODO Consensus network
** TODO New biomarkers

    - SNP level analyses for SconES
        - Chromosomic region
        - Non-protein SNPs
    - Gene level analyses
        - Pathway enrichment
        - Biomarkers LD clumping/LASSO
          
** TODO Comparison
   
    - Size: how many SNPs and how many genes. Do a Elastic net (more stable, more robust, fix alpha) and report that number as well.
    - Stability.
    - Sensitivity & specificity on Lasso'd SNPs
    - Consensus network & Jaccard index of each of them with it.
    - Runtime?

* TODO Discussion

bibliographystyle:ieeetr
bibliography:bibliography.bib

#+LATEX: \clearpage
#+LATEX: \setcounter{figure}{0}
#+LATEX: \setcounter{section}{0}
#+LATEX: \setcounter{table}{0}

* Supplementary materials

#+CAPTION: *(A,B,C,D)* Eight main principal components computed on the genotypes of GENESIS. Cases are colored in green, controls in orange. 
#+NAME: sfig:pcs
[[./figures/sfigure_1.pdf]]
