#+TITLE: Network-based GWAS on familial breast cancer: method comparison and new biomarkers
#+AUTHOR: Héctor Climente-González, Christine Lonjou, Chloé-Agathe Azencott
#+OPTIONS: toc:nil num:nil

* TODO Introduction

In human health, the goal of genome-wide association studies (GWAS) is uncovering the genetic variants that predispose to complex diseases, like diabetes or some variants of cancer. To that end, thousands of unrelated samples are genotyped, a fraction of them with the disease of interest, and the rest from the general population. In the traditional setting, a per-SNP statistical test (i.e. \chi^2) is conducted, and only those SNPs with a P-value lower than a conservative threshold are the selected ones.

This experimental design has been successful in identifying the main variants underlying many common diseases, allowing early diagnosis. However, the experimental setting also presents intrinsic challenges. The first one is the high-dimensionality of the problem: all GWAS studies to date genotype more positions in the genome than samples are used. This limits the statistical power of the experiment, as only variants with large and moderate effects can be detected. Additionally, to avoid false positives, a conservative multiple test correction is applied, typically Bonferroni. The second problem is the interpretability of the results, when SNPs in non-coding regions or in large LD-blocks are selected. 

On top of the aforementioned problems, there is evidence suggesting that most of the genome contributes to a degree to any complex trait. This opposes to the traditional polygenic model of complex diseases, which postulated that only key genes in key pathways need to be affected. In the recently postulated omnigenic model, genes are highly interconnected, and influence each others behavior. Hence, a more comprehensive statistical framework is required.

Network-based methods offer a compact and elegant solution to both problems. In essence, they represent the experiment as a network where the nodes contain information from the experiment, and the edges represent functional relatedness between them. A common paradigm consists on aggregating the SNP association into gene association, and relate the genes according to whether their proteins physically interact. The problem involves the selection of one or more subnetworks, densely intra-connected, which contains the highest-associated nodes. 

Hierarchichal HotNet categories:

- Constrain the problem to certain kinds of subnetworks, then explore those subnetworks: jActiveModules, heinz, LEAN.
- Consider jointly the node scores and topology: SConES, HotNet 2, Sigmod.
- Use additional information: Hierarchical Hotnet.

Check classification in Network-guided biomarker discovery.

In this article we focus on GENESIS cite:sinilnikova_genesis:_2016, a homogeneous dataset of familial breast cancer in French population of European ancestry. We apply a representative set of methods, some of them never applied to GWAS. Our goal is two-fold: we compare the methods in this dataset; and we explore new biomarkers of familial breast cancer. 

* Methods

** GENESIS

GENESIS is a GWAS study of familial breast cancer on the French population cite:sinilnikova_genesis:_2016. The index cases are patients with infiltrating mammary or ductal adenocarcinoma, who had sister with breast cancer, and negative for BRCA1 and BRCA2 mutations. The controls are unaffected colleagues and/or friends of the cases, born around the year of birth of the corresponding case (\pm 3 years). In total, the GENESIS dataset consists of 2577 samples, of which 1279 are controls and 1298 are cases. The genotyping platform was iCOGS, a custom Illumina array designed to study genetic susceptibility of hormone-related cancers cite:sakoda_turning_2013. It contains 210918 SNPs previously associated with cancer susceptibility, survival, or other cancer-related traits; and in selected genes and pathways.

** Preprocessing and quality control

We discarded SNPs with a minor allele frequency lower than 0.1%, not in Hardy - Weinberg equilibrium (P-value \textless 0.001), and/or those with missing values on more than 10% of the samples. In addition, we removed the samples with more than 10% missing genotypes. 28 samples with ?? were removed. A subset of 20 duplicated SNPs in FGFR2 were also removed. The final dataset included 1271 controls and 1280 cases, genotyped over 197083 SNPs. 

We examined if GENESIS had evidence of population structure that could create confounding associations. A PCA revealed no differential population structure between cases and controls (Supplementary Figure [[sfig:pcs]]). Independently, we did not find evidence of genomic inflation (\lambda = 1.05029), further dismissing the presence of population structure.

** High-weight subnetwork discovery algorithms

*** Node scores

To measure association between a genotype and the phenotype, we performed a per-SNP 1df \chi^2 allelic test using PLINK v1.90 cite:chang_second-generation_2015. Then, we used VEGAS2v2 to compute the gene-level association score cite:mishra_vegas2:_2015 from the SNP P-values. In order to map SNPs to genes we relied used their overlap on the sequence: when a SNP is located within the boundaries of a gene, \pm 50 kb, it was mapped to that gene. To compute the gene association, we used all the SNPs mapped to it, as opposed to other strategies like taking only the 10% of SNPs with lowest P-values.

*** Network

We used the physical interactions from BioGRID version 3.5.171 cite:oughtred_biogrid_2019 to get protein-protein interactions when required. Out of the six methods tested, five use a protein-protein interaction network.

*** Mathematical notation

In this article, we refer to undirected, vertex-weighted graphs G = (V,E,w). V = {v_1, \dots, v_n} refers to the vertices, with weights w: V \rightarrow \mathbb{R}. Equivalently, E \subseteq {{x,y} | x,y \in V \wedge x \neq y} refers to the edges. When referring to a subnetwork S, V_S is the set of nodes in S and E_S is the set of edges in S.

*** Methods used

Finding the highest-scoring, densely interconnected subnetwork on a graph is an open problem in the field. Hence, several solutions have been proposed to the problem. In this paper, we apply 5 methods designed to explore the protein-protein interaction network, and one method, SConES, which explores SNP-networks.

+ heinz :: The goal of heinz is identifying the highest-scored connected subgraph on the network cite:dittrich_identifying_2008. The problem has a trivial solution when all scores are positive: the whole network; however, it becomes NP-complete when scores are both positive and negative. The authors propose a transformation of the nodes' P-value into a score which takes a negative value when no association with the phenotype is detected, and a positive value when it is. The distinction between both is determined though an FDR approach. Then, the problem is re-casted as the Prize-Collecting Steiner Tree Problem (PCST). This is the problem of selecting the subnetwork S that maximizes the /profit/ p(S):

    \begin{equation}
    p(S) = \sum_{v \in V_S} p(v) - \sum_{e \in E_S} c(e). 
    \end{equation}

    were p(v) is called profit of adding a node, and c(e) is the cost of the edge, both positive values. These quantities are defined from w' = min_{v \in V_G} w(v):

    \begin{multiline}
    p(v) = w(v) - w', \\
    c(e) = w'.
    \end{multiline}

    PCST has a heuristic, efficient solution cite:ljubic_algorithmic_2006. We used the implementation of heinz from BioNet cite:beisser_bionet:_2010, available on Bioconductor.
+ Hierarchichal HotNet :: Hierarchical HotNet was developed in the context of cancer driver identification cite:reyna_hierarchical_2018. Its aim is identifying statistically significant subnetworks, made up by nodes with high-scores that are close in the network. It does so by performing hierarchical clustering on a similarity matrix that captures both the node score similarity and the network similarity. Specifically, it uses the stationary distribution of a random walk with restart procedure to capture a graph-based vertex similarity. By setting up the appropriate restart probability, we tune the locality of this similarity. 

    Interestingly, Hierarchical HotNet addresses the bias of biological graphs, where genes with high scores tend to have higher degrees, as they are better studied. 
    
+ jActiveModules ::
+ LEAN :: cite:gwinner_network-based_2016
+ SConES :: SConES searches the minimal, maximally interconnected, maximally associated subnetwork in a SNP graph cite:azencott_efficient_2013. Specifically, it solves the following problem:

    \begin{equation}
    \underset{f \in\{0,1\}^{n}}{\arg \max } \underbrace{c^{\top} f}_{\text { association }}-\underbrace{\lambda f^{\top} L f}_{\text { connectivity }}-\underbrace{\eta\|f\|_{0}}_{\text { sparsity }}
    \end{equation}

    Building a SNP graph is non-trivial. The authors suggested three networks: connecting the genotyped SNPs consecutive in the genomic sequence (GS); interconnecting all the SNPs mapped to the same gene, on top of GS (GM); and interconnecting all SNPs mapped to two genes for which a protein-protein interaction exists (GI). 

+ SigMod ::

** Validation of selected biomarkers

*** Classification accuracy of selected biomarkers

To evaluate the quality of the SNPs selected by the different methods, we used
the performance of a classifier trained exclusively on those SNPs. The
underlying assumption is that SNPs that are good biomarkers will be good
predictiors of the phenotype of unseen samples. To that end, we followed a
5-fold cross-validation schema, where over 5 iterations 80% of the data is used
to select SNPs and train the classifier, and 20% is used to compute and
evaluate the predictions.

Specifically we fitted an L1-penalized logistic regression over the age and the
selected SNPs, using the outcome (case/control) as
target variable. When the method retrieved a list of genes (all of them except
SConES), all the SNPs mapped to any of those genes were used.

*** Comparison to external dataset

An alternative way to validate the results involves looking at an external
dataset. For that purpose, we used the Breast Cancer
Association Consortium (BCAC). BCAC is one of the largest efforts in
GWAS, with over XX samples genotyped over YY SNPs. Specifically, we used the summary
statistics of the European cohort. Despite the studies not being performed on
exactly the shared population, there is a lot of overlap between a pan-European
cohort and a French cohort. On top of that, that overlap should increase when
the results are observed at the gene-level.

** Code availability

The implementation of the different network-based methods to GWAS data is available at https://github.com/hclimente/gwas-tools. The code necessary to reproduce all the analyses in this article is available at https://github.com/hclimente/genewa.

* Results
** TODO SNP and gene level analyses
#+CAPTION: Association in GENESIS. The red line represents the Bonferroni threshold. *(A).* SNP association, measured from the outcome of a 1df \chi^2 allelic test. SNPs that are within a coding gene, or within 50 kilobases of its boundaries are annotated. *(B).* Gene association, measured by P-value of VEGAS2v2 cite:mishra_vegas2:_2015.
#+NAME: fig_snp_gene_manhattan
[[./figures/figure_1.pdf]]
** TODO Comparison

    - Size: how many SNPs and how many genes. Do a Elastic net (more stable, more robust, fix alpha) and report that number as well.
    - Stability.
    - Sensitivity & specificity on Lasso'd SNPs
    - Consensus network & Jaccard index of each of them with it.
    - Runtime?

** TODO New biomarkers

    - SNP level analyses for SconES
        - Chromosomic region
        - Non-protein SNPs
    - Gene level analyses
        - Pathway enrichment
        - Biomarkers LD clumping/LASSO

** TODO Consensus network

* TODO Discussion

bibliographystyle:ieeetr
bibliography:bibliography.bib

#+LATEX: \clearpage
#+LATEX: \setcounter{figure}{0}
#+LATEX: \setcounter{section}{0}
#+LATEX: \setcounter{table}{0}

* Supplementary materials

#+CAPTION: *(A,B,C,D).* Eight main principal components computed on the genotypes of GENESIS. Cases are colored in green, controls in orange. 
#+NAME: sfig:pcs
[[./figures/sfigure_1.pdf]]
