#+TITLE: Combining network-guided GWAS to discover susceptibility mechanisms to breast cancer
#+AUTHOR: Héctor Climente-González, Christine Lonjou, Chloé-Agathe Azencott
#+OPTIONS: toc:nil num:nil

\begin{abstract}

# TODO
This is the abstract.

\end{abstract}

* Introduction

In human health, genome-wide association studies (GWAS) aim at quantifying how genetic variants predispose to complex diseases like diabetes or some forms of cancer cite:bush_chapter_2012. The most commonly studied variants are single-nucleotide polymorphisms (SNPs). Once the relevant SNPs have been discovered, they can be used for diagnosis, computing the risk, and deepen our understanding of the disease. To that end, in a typical retrospective study thousands of unrelated samples are genotyped: the cases, suffering the disease of interest, and the controls from the general population. Then, a per-SNP statistical association test is conducted (e.g. \chi^2). Those SNPs with a P-value lower than a conservative FWER threshold are candidates to further studies in an independent cohort.

GWAS has been successful at identifying the main variants underlying many common diseases. However, the experimental setting also presents intrinsic challenges. Some of them stem from the high-dimensionality of the problem, as every GWAS to date study more variants than samples are genotyped. This limits the statistical power of the experiment, as only variants with large and moderate effects can be detected. However, most of the heritability can be explained by variants with small effects. Additionally, to avoid false positives, a conservative multiple test correction is applied, typically Bonferroni. However, Bonferroni is known to be overly conservative when the statistical tests are correlated. Another open issue is the interpretability of the results, as the functional consequences of most common variants are not well understood. The fact that SNPs in non-coding regions often top the results of GWAS portrays well this challenge. On top of that, recent large-sampled studies suggest that most of the genome contributes to a degree to any complex trait. This opposes to the traditional polygenic model of complex diseases, which postulated that only key genes in key pathways need to be affected. In the recently postulated omnigenic model cite:boyle_expanded_2017, genes are highly interconnected, and influence each others behavior. Hence, a more comprehensive statistical framework is required.

In this context, many authors turn to network biology to handle the complex interplay of biomolecules that lead to disease cite:furlong_human_2013. As its name suggests, network biology models biology as a network, where the biomolecules under study, often genes, are nodes, and selected functional relationships between them are the edges that link them. These functional relationships come from evidence that the genes jointly contribute to a biological function; for instance, their expression is correlated, or a protein-protein interaction between their gene products have been discovered. Under this view, complex diseases are not the consequence of a single altered gene, but of the interaction of multiple interdependent biomolecules cite:barabasi_network_2011. In fact, an examination of biological networks shows that disease genes have differential properties cite:barabasi_network_2011 cite:pinero_uncovering_2016. This was particularly true for cancer driver genes, which tended to be key players in connecting different densely-connected communities of genes. 

# TODO read Barrenas 2012, Furlong 2013 and Cowen 2017.

Crucially, the genes that contribute to a disease tend to participate in similar biological functions. Thus, guilt-by-association strategies have proved effective at identifying disease genes cite:huang_systematic_2018. Network-based, biomarker discovery methods exploit this differential characteristic to identify disease genes on GWAS data. In essence, each SNP has a measure of association with the disease, given by the experiment, and a biological context, given by the network. Then, the problem becomes finding a functionally-related set of genes that is highly associated with the disease. Different solutions have been proposed to this problem, often stemming from divergent considerations of what the optimal solution set look like. Some methods strongly constrain the problem to certain kinds of subnetworks. Such is the extreme case of LEAN cite:gwinner_network-based_2016, which focuses on 'star subnetworks' i.e. instances were a gene and its direct interactors are associated with the disease. Other algorithms, like dmGWAS cite:jia_dmgwas:_2011 and heinz cite:dittrich_identifying_2008, focus on interconnected genes with high association with the disease. However, they differ in their tolerance to include lowly associated nodes. Lastly, other methods explicitly consider the topology of the solution, favoring densely interconnected subnetworks; such is the case of HotNet2 cite:leiserson_pan-cancer_2015, SConES cite:azencott_efficient_2013, and SigMod cite:liu_sigmod:_2017.

TODO Check classification in Network-guided biomarker discovery.

In this work, we analyze the effectiveness of these six methods to discover new biomarkers on GWAS data. We focus on the GENESIS dataset cite:sinilnikova_genesis:_2016, a homogeneous dataset of familial breast cancer in French population of European ancestry. After following a classical GWAS approach, we use these network-based methods to recover additional familial breast cancer biomarkers. Some of them are known, while others are specific to this dataset. Lastly, we carry out a comparison of the solutions obtained by the different methods, and aggregate them to obtain a consensus network of predisposition to familial breast cancer. 

* Methods
** GENESIS

GENESIS is a GWAS study of familial breast cancer on the French population cite:sinilnikova_genesis:_2016. The index cases are patients with infiltrating mammary or ductal adenocarcinoma, who had sister with breast cancer, and negative for BRCA1 and BRCA2 mutations. The controls are unaffected colleagues and/or friends of the cases, born around the year of birth of the corresponding case (\pm 3 years). In total, the GENESIS dataset consists of 2577 samples, of which 1279 are controls and 1298 are cases. The genotyping platform was iCOGS, a custom Illumina array designed to study genetic susceptibility of hormone-related cancers cite:sakoda_turning_2013. It contains 210918 SNPs previously associated with cancer susceptibility, survival, or other cancer-related traits; and in selected genes and pathways.

# Molecular subtypes? TNBC, HER2+...

** Preprocessing and quality control

We discarded SNPs with a minor allele frequency lower than 0.1%, not in Hardy - Weinberg equilibrium (P-value \textless 0.001), and/or those with missing values on more than 10% of the samples. In addition, we removed the samples with more than 10% missing genotypes. 28 samples with TODO were removed. A subset of 20 duplicated SNPs in FGFR2 were also removed. The final dataset included 1271 controls and 1280 cases, genotyped over 197083 SNPs. 

We looked for population structure that could create confounding associations. A PCA revealed no differential population structure between cases and controls (Supplementary Figure [[sfig:pcs]]). Independently, we did not find evidence of genomic inflation (\lambda = 1.05029) either, thus further dismissing the presence of confounding population structure.

** High-weight subnetwork discovery algorithms
*** SNP and gene association 
    :PROPERTIES:
    :CUSTOM_ID: methods:node_score
    :END:
To measure association between a genotype and the phenotype, we performed a per-SNP 1df \chi^2 allelic test using PLINK v1.90 cite:chang_second-generation_2015. Then, we used VEGAS2v2 to compute the gene-level association score cite:mishra_vegas2:_2015 from the SNP P-values. In order to map SNPs to genes we relied used their overlap on the sequence: all SNPs located within the boundaries of a gene, \pm 50 kb, were mapped to that gene. To compute the gene association we used the 10% of SNPs with lowest P-values. We computed the association 62193 genes described in GENCODE 31 cite:frankish_gencode_2019; only 54612 had a SNP mapped to them. Then, we focused exclusively on the 32767 that could be mapped to an HGNC symbol. Out of the SNPs 197083 in iCOGS after quality control, 164037 mapped to at least one of these genes. 

*** Gene-gene network
    :PROPERTIES:
    :CUSTOM_ID: methods:gene_network
    :END:

Out of the six methods tested, five use a gene-gene interaction network (Section [[#methods:methods]]), and their respective statistical frameworks are compatible with any type of network (protein interactions, gene coexpression, regulatory, etc.). However, in order to make the results comparable, we needed to apply all the methods to the same network. Hence, for practical reasons, we focused on a protein-protein interaction network (PPIN), as most of the methods were designed to scale appropriately to them. We built our PPIN from both binary and co-complex interactions stored in the HINT database (release April 2019) cite:das_hint:_2012. Unless specified otherwise, we used only interactions coming from high-throughput experiments to avoid biasing the topology of the network by well-studied genes with more known interactions on average. Out of the 146722 interactions from high-throughput experiments that HINT stores, we were able to map 142541 to a pair of HGNC symbols, which we used as node identifier.

Additionally, we compared the results of the aforementioned network with those obtained on a network built using interactions coming from both high-throughput and targeted studies. In that case, out of the 179332 interactions in HINT, we mapped 173797 to a pair of HGNC symbols. 

The scoring function for the nodes changed from method to method (Section [[#methods:methods]]). 

*** SNP networks
    :PROPERTIES:
    :CUSTOM_ID: methods:snp_network
    :END:

SConES cite:azencott_efficient_2013 is the only of the studied methods designed to handle SNP networks. As in gene networks, two SNPs are linked in a SNP network when there is evidence of shared functionality between two SNPs. The authors suggested three ways of building these networks: connecting the genotyped SNPs consecutive in the genomic sequence ("GS network"); interconnecting all the SNPs mapped to the same gene, on top of GS ("GM network"); and interconnecting all SNPs mapped to two genes for which a protein-protein interaction exists ("GI network"). We used all three. For the GM network, we used the mapping described in Section [[#methods:node_score]]. For the GI network, we used the PPI as described in Section [[#methods:gene_network]]. 

For all three networks the node score used is the association of the individual SNPs with the phenotype; specifically, we used the 1 d.f. \chi^2.

*** Mathematical notation
    :PROPERTIES:
    :CUSTOM_ID: methods:notation
    :END:

In this article, we refer to undirected, vertex-weighted networks, or graphs, G = (V,E,w). V = {v_1, \dots, v_n} refers to the vertices, with weights w: V \rightarrow \mathbb{R}. Equivalently, E \subseteq {{x,y} | x,y \in V \wedge x \neq y} refers to the edges. When referring to a subnetwork S, V_S is the set of nodes in S and E_S is the set of edges in S. A special case of subgraphs are /connected/ subgraphs, which occur when every node in the subgraph can be reached from any other node.

In addition, we use several matrices that describe different properties of a graph. The described matrices are square, and have as many rows and columns as nodes are in the network. In fact, the element i,j represent a  selected relationship between v_i and v_j. The adjacency matrix W_G contains a 1 when the corresponding nodes are connected through an edge, and 0 otherwise; the diagonal is zero. The degree matrix D_G is a diagonal matrix which contains the degree of the different nodes. Lastly, the Laplacian matrix L_G is defined as L_G = D_G - W_G.

*** TODO Methods used
    :PROPERTIES:
    :CUSTOM_ID: methods:methods
    :END:

#+CAPTION:Summary of the differences between the studied algorithms. The colums are: Field, the field in which the algorithm was developed; Node type, the type of network, either gene (protein-protein interaction network usually) or a SNP network; Exhaustive, if all the possible solutions given the selected hyperparameters are explored, or not; Solution, referring properties that are enforced on the solution, other than being dense in high scores and connected; and Input, referring to whether the methods require genotype data or GWAS summary statistics. 
#+NAME:   tab:method_comparison
| Algorithm | Field | Node type | Exhaustive | Solution    | Input     |
|-----------+-------+-----------+------------+-------------+-----------|
| heinz     | Omics | Gene      | Yes?       | -           | Summary   |
| HotNet2   | Omics | Gene      | Yes?       | Modular     | Summary   |
| dmGWAS    | GWAS  | Gene      | No         | -           | Summary   |
| LEAN      | Omics | Gene      | Yes        | Star-shaped | Summary   |
| SConES    | GWAS  | SNP       | Yes        | Modular     | Genotypes |
| SigMod    | GWAS  | Gene      | Yes        | Modular     | Summary   |

TODO explain why it's an open problem i.e. which score should be used (SNP, gene?), what the solution looks like, the problem is NP-hard.
TODO specify how nodes are scored.

Finding the highest-scoring, densely interconnected subnetwork on a graph is an open problem in the field. Hence, several solutions have been proposed to the problem. In this paper, we apply six methods designed to explore the protein-protein interaction network, and one method, SConES, which explores SNP-networks. We selected methods that had a readily available, programmatically accessible implementation. Their main differences are summarized in Table [[tab:method_comparison]].

TODO Re-read heinz paper. It's the solution heuristic? If so, how good is it? Efficient enough to be used in SNP network?
TODO Reformulate heinz to show similarities to SConES.

- heinz :: The goal of heinz is identifying the highest-scored connected subgraph on the network cite:dittrich_identifying_2008. The problem has a trivial solution when all scores are positive: the whole network; however, it becomes NP-complete when scores are both positive and negative. The authors propose a transformation of the nodes' P-value into a score which takes a negative value when no association with the phenotype is detected, and a positive value when it is. The distinction between both is determined though an FDR approach. Then, the problem is re-casted as the Prize-Collecting Steiner Tree Problem (PCST). This is the problem of selecting the connected subnetwork S that maximizes the /profit/ p(S):

    \begin{equation*}
    p(S) = \sum_{v \in V_S} p(v) - \sum_{e \in E_S} c(e). 
    \end{equation*}

    were p(v) is called profit of adding a node, and c(e) is the cost of the edge, both positive values. These quantities are defined from w' = min_{v \in V_G} w(v):

    \begin{multiline}
    p(v) = w(v) - w', \\
    c(e) = w'.
    \end{multiline}

    PCST has a heuristic, efficient solution cite:ljubic_algorithmic_2006. We used the implementation of heinz from BioNet cite:beisser_bionet:_2010, available on Bioconductor cite:heinz.

- HotNet2 :: HotNet2 was developed in the context of tumor driver identification, as a tool to find connected subgraphs of genes mutated more often than expected by chance cite:leiserson_pan-cancer_2015. To that end, it considers both the local topology of the network and the scores of the nodes. The former is captured by an insulated heat diffusion process, modeled by a random walk with restart. At the beginning, the score of the node determines its initial heat. In an iterative procedure, each node gives heat to its "colder" neighbors, and receives heat from its "hotter" neighbors, while retaining part of its heat (hence, /insulated/). This process continues until equilibrium is reached, and results in a similarity matrix F. This matrix is used to compute the similarity matrix E that accounts also for similarities in node scores as 

    \begin{equation*} 
    E = F \operatorname{diag}(w(V)), 
    \end{equation*}

    where $\operatorname{diag}(w(V))$ is a diagonal matrix with the node scores in its diagonal. HotNet2 explores the similarity network built from E to find densely connected subnetworks. Specifically, it only connects a pair of nodes i and j when E(i,j) \gt \delta. Lastly, HotNet2 evaluates the statistical significance of the subnetworks by comparing their size to the size of networks obtained by permuting the node scores. 

    HotNet2 has two parameters: the restart probability \beta, and the threshold heat \delta. Both parameters are set automatically by the algorithm, and are robust cite:leiserson_pan-cancer_2015.

    HotNet2 is implemented in Python cite:hotnet2. 

TODO Read Gene and Network Analysis of Common Variants Reveals Novel Associations in Multiple Complex Diseases

- dmGWAS :: dmGWAS aims at identifying the connected subgraph with the largest amount of low P-values cite:jia_dmgwas:_2011. To that end, it first searches several candidate subnetwork solutions using a greedy procedure involving the following steps:

  1. Select a seed node.
  2. Compute Stouffer's Z-score Z_m for the current subgraph as
    
    \begin{equation*} 
    Z_m = \frac{\sum z_i}{\sqrt{k}}
    \end{equation*}

    where /k/ is the number of genes in the subgraph, z_i = \phi^{-1}(1 - P_i), and \phi^{-1} is the inverse normal distribution function.
  3. Identify neighboring nodes i.e. nodes at shortest path \le /d/. We set d = 2.
  4. Add the neighboring nodes whose inclusion increases the Z_{m+1} more than Z_m \times (1 + r). In our experiments, we set r = 0.1.
  5. Repeat 2-4 until no increment Z_m \times (1 + r) is possible.
  
  Lastly, the module's Z-score is normalized as

  \begin{equation*}
  Z_{N}=\frac{Z_{m}-\operatorname{mean}\left(Z_{m}(\pi)\right)}{\operatorname{SD}\left(Z_{m}(\pi)\right)}
  \end{equation*} 

  where Z_{m}(\pi) represent a vector with 100000 random subsets of the same number of genes.

  We used the implementation of dmGWAS in the dmGWAS 3.0 R package cite:dmgwas. We used the function /simpleChoose/ to select the solution subnetwork, which aggregates the top 1% modules into the solution subnetwork.
- LEAN :: Local enrichment analysis (LEAN) searches disregulated "star" gene subnetworks i.e. subnetworks composed by one central node and all its interactors cite:gwinner_network-based_2016. By imposing this restriction, LEAN is able to exhaustively test all possible solution subnetworks (one per node in the network). For a particular subnetwork of size /m/, the P-values corresponding to the involved nodes are ranked as p_1 \le \dots \le p_m. Then, /k/ binomial tests are conducted, to compute the probability of having /k/ out of /m/ P-values lower or equal to p_k under the null hypothesis. The minimum of these /k/ P-values is the score of the subnetwork. This score is transformed into a P-value through an empirical distribution obtained via a subsampling scheme, where sets of /m/ genes are selected randomly, and their score computed. Lastly, P-values are corrected for multiple testing through a Benjamini-Hochberg correction. We used the implementation of LEAN from the LEANR R package cite:leanr.
- SConES :: SConES searches the minimal, maximally interconnected, maximally associated subnetwork in a SNP graph cite:azencott_efficient_2013. Specifically, it solves the problem

    #+NAME: eq:scones
    \begin{equation}
    \underset{S \subseteq G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }}-\underbrace{\lambda \sum_{v \in V_S} \sum_{u \not\in V_S} W_{vu} }_{\text { connectivity }}-\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}
    \end{equation}

    where \lambda and \eta are hyperparameters that control the sparsity and the connectivity of the model. For two hyperparameters, the aforementioned problem has a unique solution, that SConES finds using a graph min-cut procedure. We used the version on SConES implemented in R package martini cite:martini. We selected \lambda and \eta by cross-validation, choosing the values that produce the most stable solution across folds. Note that the solution to the above problem can consist of several connected subnetworks which are disconnected from each other. In this case, the selected hyperparameters were \eta = 3.51, \lambda = 210.29 for SConES GS; \eta = 3.51, \lambda = 97.61 for SConES GM; and \eta = 3.51, \lambda = 45.31 for SConES GI.

TODO Comment similarity with heinz

- SigMod :: SigMod aims at identifying the most densely connected gene subnetwork that is most strongly associated to the phenotype cite:liu_sigmod:_2017. It addresses an optimization problem similar to that of SConES (Equation [[eq:scones]]), but replacing the Laplacian matrix my the adjacency matrix (Section [[#methods:notation]]). 

    \begin{equation*}
    \underset{S \in G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} + \underbrace{\lambda \sum_{v \in V_S} \sum_{u \in V_S} W_{vu} }_{\text { connectivity }} -\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}.
    \end{equation*}
  
    As SConES, this optimization problem can also be solved by a graph min-cut approach. 

    SigMod presents two important additional differences with SConES. First it is designed for gene-gene networks. Second, it returns a single connected subnetwork, which it achieves by exploring a grid of hyperparameters and processing their respective solutions. Specifically, for the range of \lambda = \lambda_{min}, \dots, \lambda_{max} for the same \eta, it prioritizes the solution with the largest change in size from \lambda_n to \lambda_{n+1}. Such a large change implies that the network is strongly interconnected. This results in one candidate solution for each \eta, which are processed by removing any node not connected to any other. A score is assigned to each candidate solution by summing their node scores and normalizing by size. The candidate solution with the highest standardized score is the chosen solution. SigMod is implemented in an R package cite:sigmod. 

*** Mapping back and forth between gene methods and SConES

In this work dealt with multiple methods, which use GWAS data at different levels. VEGAS2 compute gene statistics from SNP statistics, which are then used by five gene-based network methods to find a subnetwork associated with familial breast cancer. In order to obtain a list of SNP biomarkers from these gene subnetworks, we consider all the genes that can be mapped to that gene as selected by the method. SConES is in the opposite case: it performs selection on a network of SNPs. In this case, when analyzing the genes selected by SConES, we consider any gene that can be mapped to any of the selected SNPs as selected as well. 

*** Consensus network
    :PROPERTIES:
    :CUSTOM_ID: methods:consensus
    :END:
The different high-weight subnetwork discovery algorithms make different assumptions on the nature of the solutions, and employ different strategies to find them. Hence, combining the outcome of the different approaches might provide a more complete outlook on the specific alterations on the GENESIS dataset. We built such consensus network by retaining the nodes that were selected by at least two of the methods. We combined the results of 6 methods: heinz, Hierarchical HotNet, dmGWAS, LEAN, SConES on the GM network, and SigMod.

** Validation of selected biomarkers
*** Classification accuracy of selected biomarkers
    :PROPERTIES:
    :CUSTOM_ID: methods:comparison
    :END:
To evaluate the quality of the solutions offered by the different algorithms, we used their predictor power. We reasoned that a desirable solution is one that is sparse, while offering a good predictor power. To evaluate the predicting power of the SNPs selected by the different methods, we used the performance of an L1-penalized logistic regression trained exclusively on those SNPs to predict the outcome (case/control). The L1 penalty helps to account for LD to reduce the size of the active set, while improving the generalization of the classifier. The value of the \lambda, which controls the size of the coefficients, was set by cross-validation. To that end, we used the different network-methods on a random 80% of the samples and trained our predictor exclusively on the SNPs selected by a particular method, on these samples. When the method retrieved a list of genes (all of them except SConES), all the SNPs mapped to any of those genes were used. Then we evaluated performance of the classifier on the remaining 20% of the dataset. We repeated this procedure 5 times to estimate the average and the deviation of the different performance measures. The different performance measures we used where: size of the solution, size of the active set, specificity, sensitivity and average Jaccard similarity between different runs. In addition, we repeated the procedure without applying a network-based feature selection method.

Another desirable property is that the method retrieves a good candidate causal subnetwork. In consequence, we compared the outcome of each of the methods to the consensus subnetwork of all the solutions (Section [[#methods:consensus]]). 

TODO Machine learning & SNP paper.

*** Biological relevance of the genes
    :PROPERTIES:
    :CUSTOM_ID: methods:bcac
    :END:
An alternative way to validate the results is comparing our results to an external dataset. For that purpose, we recovered a list of 153 genes known to be associated to familial breast cancer from DisGeNET cite:pinero_disgenet:_2017. Across this article, we refer to these genes as /familial breast cancer genes/.
 
Additionally, we used the summary statistics from the Breast Cancer Association Consortium (BCAC) cite:michailidou_genome-wide_2015. BCAC is one of the largest efforts in GWAS, with over 120000 women from European ancestry, albeit from different countries. As opposed to GENESIS, samples were not selected based on family history, and hence is enriched in sporadic breast cancers. Another difference is that BCAC is a relatively heterogeneous study on a pan-European sample, while GENESIS is a homogeneous dataset focused on the French population. Despite these differences, there should be shared genetic architecture. On top of that, that overlap should become more notorious when the results are aggregated at the gene level. For that purpose, we computed the gene association as in Section [[#methods:node_score]]. iCOGS array was used for genotyping in BCAC cite:sakoda_turning_2013, the same array as for GENESIS cite:sinilnikova_genesis:_2016. Although imputed data is available, we used exclusively the SNPs available on GENESIS after quality control to make the results comparable.

** Code availability

This work here presented required developing computational pipelines for several GWAS analyses, such as physical mapping of SNPs, computing gene scores, and perform six different network-based analyses. For each of those processes, a streamlined, project-agnostic pipeline with a clear interface was created. They are compiled in the following GitHub repository: https://github.com/hclimente/gwas-tools. The code that applies these pipelines to the GENESIS project, as well as the code that reproduces all the analyses in this article are available at https://github.com/hclimente/genewa.

* Results
** FGFR2 is strongly associated with familial breast cancer 
    :PROPERTIES:
    :CUSTOM_ID: results:conventional
    :END:

#+CAPTION:Association in GENESIS. The red line represents the Bonferroni threshold. *(A)* SNP association, measured from the outcome of a 1df \chi^2 allelic test. SNPs that are within a coding gene, or within 50 kilobases of its boundaries are annotated. The Bonferroni threshold is 2.54 \times 10^{-7}. *(B)* Gene association, measured by P-value of VEGAS2v2 cite:mishra_vegas2:_2015 using the 10% of SNPs with the lowest P-values. The Bonferroni threshold is 1.53 \times 10^{-6}.
#+NAME: fig:snp_gene_manhattan
[[./figures/figure_1.pdf]]

We conducted association analyses both at the SNP level and at the gene level in the GENESIS dataset (Section [[#methods:node_score]]). Two genomic regions have a P-value lower than the Bonferroni threshold in chromosomes 10 and 16 (Figure [[fig:snp_gene_manhattan]]A). The former overlaps with gene FGFR2; the latter with CASC16, and its located near the protein-coding gene TOX3. Variants in both FGFR2 and TOX3 were related to breast cancer susceptibility in other cohorts negative for BRCA1/2 cite:rinella_genetic_2013. Only the peak in chromosome 10 replicated in the gene-level analysis, with FGFR2 just below threshold of significance (Figure [[fig:snp_gene_manhattan]]B). 

These results show the overlap between the genetic architecture of the disease between the French population and other cohorts, especially at the gene level. In addition, there are other regions highly associated with familial breast cancer, albeit well above the conventional threshold of significance. The most prominent regions, which have been associated to breast cancer susceptibility in the past, are 3p24 cite:brisbin_meta-analysis_2011, and 8q24 cite:search_newly_2009. This motivates exploring network methods, which trade statistical association for biological significance.

** TODO Network methods successfully identify genes linked to breast cancer 

#+CAPTION: Summary statistics on the results of multiple network methods on the gene-gene interaction network. The first row contains the summary statistics on the whole network.
#+NAME: tab:gene_solutions
| Network   | Genes |  Edges | Mean betweenness |    Median P_{gene} | Jaccard_{consensus} |
|-----------+-------+--------+------------------+--------------------+---------------------|
| HT HINT   | 13619 | 142541 |            16706 |               0.46 |               0.004 |
|-----------+-------+--------+------------------+--------------------+---------------------|
| Consensus |    55 |    117 |            74062 |             0.0051 |                   1 |
| dmGWAS    |   194 |    450 |            49115 |               0.19 |                0.26 |
| heinz     |     4 |      3 |           113633 |             0.0012 |               0.073 |
| HotNet2   |       |        |                  |                    |                     |
| LEAN      |     0 |      0 |                - |                  - |                   0 |
| SConES GS |     5 |      0 |             9805 | 2.7 \times 10^{-5} |               0.071 |
| SConES GM |    28 |      2 |             4267 |              0.067 |               0.078 |
| SConES GI |     0 |      0 |                - |                  - |                   0 |
| SigMod    |   142 |    249 |            92603 |             0.0083 |                0.33 |

#+CAPTION: Summary statistics on the results of SConES on the three SNP-SNP interaction networks. The first row within each block contains the summary statistics on the whole network.
#+NAME: tab:snp_solutions
| Network   |   SNPs |    Edges | Subnetworks | Mean betweenness | Median P_{SNP} |
|-----------+--------+----------+-------------+------------------+----------------|
| GS        | 197083 |   197060 | -           | 2.03 \times 10^7 |           0.49 |
| SConES GS |   1590 |     1585 | 5           | 2.52 \times 10^7 |          0.023 |
|-----------+--------+----------+-------------+------------------+----------------|
| GM        | 197083 |  6442446 | -           | 3.99 \times 10^6 |           0.49 |
| SConES GM |   1692 |   177611 | 5           | 4.40 \times 10^6 |          0.055 |
|-----------+--------+----------+-------------+------------------+----------------|
| GI        | 197083 | 28733720 | -           |                  |           0.49 |
| SConES GI |    408 |      539 | 5           |                  |          0.076 |

We applied six network methods to the GENESIS dataset (Section [[#methods:methods]]). We obtained eight solutions (Supplementary files 1 and 2): one for each of the gene-based methods (Section [[#methods:gene_network]]), and one for each of the SNP networks of SConES (Section [[#methods:snp_network]]). The solutions were very heterogeneous (Tables [[tab:gene_solutions]] and [[tab:snp_solutions]]): none of the subnetworks examined by LEAN was significant (adjusted P-value < 0.05), while dmGWAS produced the largest solution subnetwork with 194 genes. However, they succeed at recovering genes involved in the disease: five solution subnetworks are significantly enriched in familial breast cancer genes (dmGWAS, heinz, SConES GS, SConES GM, and SigMod, Fisher's exact test one-sided P-value < 0.03). We also compared the outcome of the network methods to the association tests conducted on the European cohort of the Breast Cancer Association Consortium (BCAC) cite:michailidou_genome-wide_2015 (Supplementary Figure [[sfig:bcac_comparison]]). Encouragingly, every solution subnetwork was enriched in genes or SNPs that were Bonferroni-significant in BCAC. This shows that in practice network methods can find the same signal than more conservative analyses, by leveraging on the association of the biological context as a whole.

In fact, the solution subnetworks present other desirable properties. First, all solution subnetworks except LEAN's are, on average, more strongly associated to familial breast cancer than the whole HINT protein-protein interaction network. In our experiments, we observed that SConES GS strongly favor highly associated genes (median gene P-value = 2.7 \times 10^{-5}), while dmGWAS is less conservative (median gene P-value = 0.19). This exemplifies the differences between the methods: dmGWAS performs a greedy search that examines all neighbors at distance 2, and hence considering adding a weakly associated gene if it has a strongly associated neighbor. Also, the genes in five solution subnetworks display on average a higher betweenness centrality than the rest of the genes, a difference that is significant in two solutions (dmGWAS and SigMod, Wilcoxon rank-sum test one-sided P-value < 6.9 \times 10^{-22}). This agrees with the notion that disease genes are more central than other, non-essential genes cite:pinero_uncovering_2016.

Due to the differences between solutions, it is hard to draw joint conclusions. The 4-gene solution selected by heinz includes TOX3, in region 16q12, a gene that was linked to breast cancer susceptibility cite:required. This region is also picked by SConES GS - which captures the structure of the genome -, and GM - which, on top of it, captures gene membership. These two also share other breast cancer related regions and genes: 3p24 (NEK10 cite:ahmed_newly_2009), 5p12 (FGF10, MRPS30 cite:quigley_5p12_2014), and 10q26 (FGFR2, Section [[#results:conventional]]). On the other hand, they select different regions: only SConES GS selects region 8q24 (PCAT1 cite:required), while only SConES GM selects 10q24 (cite:required). By dealing with SNP networks, SConES studies the association of non-coding regions, as well as SNPs in any gene, coding or else. In fact, SConES GI, which adds to GM the interactions between genes, retrieves 4 subnetworks in intergenic regions, and 1 overlapping an RNA gene (RNU6-420P). SigMod, despite being related to SConES, produces a vastly different, large solution. On top of recovering familial breast cancer genes, a part of its subnetwork composed of keratins is focused on cytoskeleton (/structural constituent of cytoskeleton/, GO enrichment's adjusted P-value = 9.10 \times 10^{-4}), a potentially novel susceptibility mechanisms to cancer.  

# TODO Overview of dmGWAS: check most central genes 
# TODO Write to LEAN about the method
# TODO Check SConES CASC18 and CASC8

** heinz retrieves a small, highly informative set of biomarkers in a fast and stable fashion

#+CAPTION:Comparison of network-based GWAS methods on GENESIS. Each method was run 5 times of a random subset of the samples, and tested on the remaining samples (Section [[#methods:comparison]]). *(A)* Number of SNPs selected by each method and number of SNPs on the active set used by the Lasso classifier. Points are the average over the 5 runs; lines represent the standard error of the mean. The horizontal grey line represents the average active set of Lasso using all the SNPs. *(B)* Sensitivity and specificity on testing set of the L1-penalized logistic regression trained on the features selected by each of the methods. In addition, the performance of the classifier trained on all SNPs is displayed. Points are the average over the 5 runs; lines represent the standard error of the mean. *(C)* Pairwise Jaccard similarities of the solutions used by different methods. A Jaccard similarity of 1 means the two solutions are the same. A Jaccard similarity of 0 means that there is no SNP in common between the two solutions. *(D)* Runtime of the evaluated methods, by type of network used (gene or SNP). The gene network-based methods required an additional 119980 seconds (1 day and 9.33 hours) on average to compute the gene scores from SNP summary statistics (not included in the displayed Time).
#+NAME: fig:benchmark
[[./figures/figure_4.pdf]]

As the methods produced such different results, we compared their solutions in a 5-fold subsampling setting (Section [[#methods:comparison]]). Specifically, we measured the following properties (Figure [[fig:benchmark]]): (i) size of the solution subnetwork; (ii) stability; (iii) sensitivity and specificity of an L1-penalized logistic regression on the selected SNPs; and (iv) computational runtime.

Both solution size and active set of SNPs selected by Lasso varies greatly between the different methods (Figure [[fig:benchmark]]A). heinz has the smallest solutions, with an average of 182 selected our of which 5.6% (10.2) are selected by Lasso. The largest solutions come from SConES GM (4548.6 SNPs), and dmGWAS (4307.4 SNPs). Interestingly, SigMod and SConES GI have the highest proportion of the selected SNPs that go into the active set (11.47 and 10.3% respectively). This suggests those methods are selecting more informative SNPs on average. 

The sensitivity and specificity of the classifier on the testing data informs us about the usefulness of the selected SNPs as patient classification (Figure [[fig:benchmark]]B). All classifiers' sensitivities were in the 0.38 - 0.69 range; the specificities, between 0.40 and 0.70. On average, SConES GS had the highest sensitivity (0.57); heinz, the highest specificity (0.56). Both SConES GS and SConES GM had on average better sensitivity than the classifier trained on all the SNPs, and dmGWAS and heinz superior specificities. However, the differences them were negligible, well within the 95% confidence interval.   

The stability of a method measures its ability to select the same SNPs in face of perturbations on the data. We measured it by computing the pairwise Jaccard similarities between all pairs of solutions (Figure [[fig:benchmark]]C). Heinz's displayed a high stability in our benchmark, consistently selecting the same SNPs over the 5 subsamples. LEAN also showed a high stability consistently selecting no SNP.

In terms of computational runtime, the fastest method was heinz (Figure [[fig:benchmark]]D), which leverages on its ability to find efficiently the solution in a few seconds. The slowest method was SConES using the GI network, with approximately 1 day and 2.38 hours on average. Including the time required to compute the gene scores, however, slows down considerably gene-based methods; on this benchmark, that step took on average 1 day and 9.33 hours. Considering that time, dmGWAS is the slowest method, taking 1 day and 21.81 hours on average. 

** No such thing as perfect 

#+CAPTION: Drawbacks confronted when using network guided methods. *(A)* dmGWAS solution subnetwork. Genes with a P-value < 0.1 are highlighted in red. *(B)* Centrality degree and -log_10 of the VEGAS P-value for the nodes in SigMod solution subnetwork. *(C)* Genomic regions where either SConES GS, GM or GI select SNPs. 
#+NAME: fig:issues
[[./figures/figure_2.pdf]]

In practice, and despite their similarities and their involvement in cancer mechanisms, the solutions are remarkably different (Supplementary figure [[sfig:jaccard_methods]]A). That is due to the particulars of the methods, and directly or indirectly, they provide information about the dataset. For instance, the fact that LEAN did not provide any biomarkers implies that there is no gene such that both itself and its environment are on average strongly associated with the disease. 

In this dataset, heinz's solution is very conservative, providing a small solution with the lowest median P-value for the subnetwork (Table [[tab:gene_solutions]]). Due to this parsimonious and highly associated solution, it was the best method to select a set of good biomarkers for classification. (Figure [[fig:benchmark]]B). Its conservativeness stems from its preprocessing step, which models the gene p-values as a mixture model of a beta and a uniform distribution, controlled by an FDR parameter. Due to the limited signal at the gene level in this dataset (Figure [[fig:snp_gene_manhattan]]B), only 36 of them are considered to be associated to the disease. Hence, heinz's solution subnetwork consists only of 4 genes, which does not provide much insight of the biology of cancer. Importantly, it ignores genes that are strongly associated to cancer in this dataset like FGFR2. 

On the other end of the spectrum, we have large, less conservative solutions provided by dmGWAS and SigMod. In fact both solutions present a relatively large overlap (Jaccard similarity = 0.16). However, they are also among the least associated subnetworks on average. In the case of dmGWAS, this is due to the greedy framework used to solve the problem: considering neighbors at distance 2, it accepts genes weakly associated to cancer if they are linked to another, strongly associated gene. This compounds when aggregating the results of successive greedy searches, and in fact we observe a large cluster of unassociated genes (Figure [[fig:issues]]A). We observe the same tendency in SigMod's network, where the most central genes are the least associated to the disease (Figure [[fig:issues]]B). Additionally, the relatively low signal-to-noise ratio combined with the large solution requires additional analyses to draw conclusions, such as enrichment analyses. Lastly, SigMod misses some of the most strongly associated, familial breast cancer genes in the dataset, like FGFR2 and TOX3.

By virtue of using a SNP subnetwork, SConES analyzes each SNP in their context. Thanks to that, it selects SNPs in genes none of whose interactors are associated to the disease, as well as SNPs in non-coding regions or in non-interacting genes. In fact, due to linkage disequilibrium, such genes are favored by SConES, as selecting one favors selecting another one. This might explain why the GS and GM networks, heavily affected by linkage disequilibrium, produce similar results (Supplementary figure [[sfig:jaccard_methods]]B). On the other hand, SConES penalizes selecting SNPs and not their neighbors. This makes it conservative regarding SNPs with many interactions, for instance those mapped to hubs in the PPIN. Influenced by this, SConES GI did not select any protein coding gene, despite selecting similar regions as SConES GS (Figure [[fig:issues]]C). Also, the iCOGS platform is not a real GWAS experiment: the genome is not unbiasedly surveyed, some regions are fine-mapped - which might distort gene structure in GM and GI networks- while others are under studied - hurting the accuracy with which the GS network captures the genome structure. 

** TODO Aggregating solutions provides insights into the biology of cancer 

#+CAPTION:Consensus subnetwork on GENESIS (Section [[#methods:consensus]]). Each node is represented by a pie chart, which accounts the methods that selected it. The labeled genes have a VEGAS2 P-value < 0.001 and/or are known familial breast cancer genes (colored in pink).
#+NAME: fig:consensus
[[./figures/figure_3.pdf]]

To leverage on the strengths of each of the methods and compensate their respective weaknesses, we built a consensus subnetwork that captures the mechanisms most shared among the solution subnetworks (Section [[#methods:consensus]]). The consensus subnetwork (Figure [[fig:consensus]]) contains 53 genes and is enriched in familial breast cancer genes (Fisher's exact test P-value = 1.63 \times 10^{-10}). Due to the limited overlap between methods, only 7 genes were common to more than two of them (Supplementary figure [[sfig:consensus_stats]]A). Interestingly, the more methods selected a gene, the higher its association was (Supplementary figure [[sfig:consensus_stats]]B). Disease genes have a higher betweenness centrality than non-disease genes cite:pinero_uncovering_2016. We observe that to be the case in the disease under study (one-tailed Wilcoxon rank-sum test P-value = 2.64 \times 10^{-5}, Supplementary figure [[sfig:consensus_stats]]C). Accordingly, the genes on the consensus network also had a higher betweenness centrality than the rest of the genes (one-tailed Wilcoxon rank-sum test P-value = 4.77 \times 10^{-14}). Interestingly, cancer genes in the consensus network are less central than non-cancer genes in the consensus network (Wilcoxon rank-sum test P-value = 0.05). We studied if highly central genes were selected not because they were associated themselves, but because they were involved in the shortest path between two highly associated genes. As we found a weak correlation supporting this (Pearson correlation coefficient = -0.30, Supplementary figure [[sfig:consensus_stats]]D), we hypothesize that genes highly associated with the disease in this dataset and highly central might contribute to the heritability in the French population.

The consensus subnetwork is not completely connected: out of the 53 genes, the largest connected subnetwork includes only 40 genes. A GO enrichment analysis showed that this component is related to three major cellular processes: DNA helicase activity (adjusted P-value = 0.005), unfolded protein binding (adjusted P-value = 0.01), and poly(U) RNA binding (adjusted P-value = 0.01). We found support in the literature of the involvement of each of these functions in the development of cancer. /DNA helicase activity/, for instance, is crucial for DNA repair cite:required. Disruption of the DNA repair machinery is long-known to increase the likelihood of cancer, since mutations of BRCA1/2 were discovered cite:required. Unsurprisingly, it involves three familial breast cancer genes (BLM, NBN, MRE11). Another enriched activity, /unfolded protein binding/, inhibits caspase-dependent apoptosis, improving the chances of developing cancer cite:required. Three heat-shock proteins (HSPA1A, HSPA1B, HSPA1L) participate in this biological function.
 
Remarkably, 4 of the 53 genes are not linked to any other gene. Two of the latter are known familial breast cancer genes: FGFR2 (Section [[#results:conventional]]), and BABAM1 (VEGAS P-value = 3.22 \times 10^{-3}). The other two are SLC4A7 and MRPS30. SLC4A7 (VEGAS P-value = 2.70 \times 10^{5}) is a gene encoding a sodium bicarbonate cotransporter, which is selected by dmGWAS, SigMod, and SConES GM. It has linked to BRCA in the past cite:required. MRPS30 (VEGAS P-value = 0.001) is a nuclear gene encoding a mitochondrial ribosomal protein.

# TODO Out of the 7 genes selected by more than two methods, four familial breast cancer genes (BLM, CASP8, FGFR2, TOX3), and 3 are not (SLC4A7, RAB6A, OFD1).
# TODO HSPA1A (P-value = 8.37 \times 10^{-4}) - Hsp70 member proteins, including Hsp72, inhibit apoptosis by acting on the caspase-dependent pathway and against apoptosis-inducing agents such as tumor necrosis factor-α (TNFα), staurosporine, and doxorubicin. (https://en.wikipedia.org/wiki/HSPA1A)
# TODO Discuss poly(U) RNA binding?
# TODO Talk about how some gene's neighborhood is not altered (FGFR2).
# TODO Due to the overlap of SConES, we selected GM
# TODO SLC4A7: mention nearby NEK10; also role of ion transporters. 
# TODO overlap with consensus of the different methods

** TODO Hindrances of network analyses

The strength of network-based analyses comes from leveraging prior knowledge to boost discovery. However, they falter in front of genes with no prior knowledge, in other words, genes not in the network. Out of the 32767 genes that we can map the genotyped SNPs, 60.7% (19887) are not in the protein-protein interaction network. Out of those 5227 are protein coding (Supplementary figure [[sfig:biotypes_excluded]]). Among them, we find NEK10 (P-value 1.6 \times 10^{-5}), linked to breast cancer susceptibility in the past cite:ahmed_newly_2009, and POU5F1B, linked to prostate cancer cite:breyer_expressed_2014. However SNPs in NEK10 are selected by both SConES GS and GM, which do not use PPIs. Broadly speaking, protein coding genes absent from the PPIN are less associated with the phenotype on average (Wilcoxon rank-sum P-value = 2.79 \times 10^{-8}). As we are dealing with high-throughput interactions, such difference cannot be due to the focus on well-known genes. Likely, it speaks to the fact that interactions involving genes with more interactions are more likely, and disease genes tend to be more central than average cite:pinero_uncovering_2016. However, the difference is rather small: protein-coding genes in the network have a median P-value of 0.43, versus the 0.47 of those absent from it. 

As not all databases compile the same interactions, the choice of the PPIN determines the final output. Specifically, in this work we used exclusively interactions from HINT from high-throughput experiments. This responds to concerns of some authors about biases introduced by adding interactions coming from targeted studies in the literature. It can be summarized as "the rich get richer", where popular genes have a higher proportion of their interactions described. It has been reported that using such interactions might introduce biases in topological analyses cite:das_hint:_2012. On the other hand, one study found that the best predictor of the performance of a network for disease gene discovery is the size of the network cite:huang_systematic_2018. This would support using the largest amount of interactions. To clarify their impact on this study, we compared the impact of using only physical interactions from high-throughput experiment versus interactions from both high-throughput and the literature (Section [[#methods:gene_network]]). However, we found that for most of the methods using this expanded network did not have a great impact in size of the solution, classification accuracy, stability of the solution or runtime (Supplementary figure [[sfig:lc_ht_comparison]]).
 
# TODO Compare the overlap of the two solutions for each of the methods.

Lastly, we cannot forget the genes that are left out of the network due to our choice of focusing on PPIs. In fact, the largest group of unanalyzed genes are non-coding genes, mainly lncRNA, miRNA, and snRNA (Supplementary figure [[sfig:biotypes_excluded]]). The importance of these genes, like CASC16, is highlighted at the SNP-level, gene-level and again in SConES GS and GM analyses. 

* TODO Discussion

# Trash a bit the state of the art in GWAS. Trash a bit the abundance of solutions. Insights into BRCA of the paper.
In this article we evaluate the viability of systems biology approach to GWAS, and examine a GWAS dataset on familial breast cancer focused on BRCA1/2 negative French women. Systems biology addresses two of the largest GWAS issues: interpretability and an overly conservative statistical framework that hinders discovery. This is achieved by considering the biological context of each of the genes and SNPs, and selecting a threshold of association based on it. However, the method of choice is unclear. Based on divergent considerations of what the desired set of biomarkers is, several methods for network-guided biomarker discovery have been proposed. In this article we reviewed the performance of six network-guided of them on GWAS. Despite their differences, most of them produced a relevant subset of biomarkers, recovering known familial breast cancer genes. 

# Say main strenghts and weaknessess observed. Networks still need posprocessing.

# Consensus network good. However this is only one dataset.
To overcome the problems posed by the individual methods, we propose combining them into a consensus subnetwork.

Each method had a different interface, required different preprocessing steps, and some exhibited unexpected behaviors. To facilitate their other authors applying them to new datasets and aggregating their solutions, we built nextflow pipelines cite:di_tommaso_nextflow_2017 with a consistent interface and, whenever possible, parallelize the computation. They are available on GitHub: https://github.com/hclimente/gwas-tools. Importantly, those methods that had a permissive license were compiled into a Docker image for easier use [[https://hub.docker.com/r/hclimente/gwas-tools][hclimente/gwas-tools]].

bibliographystyle:ieeetr
bibliography:bibliography.bib

#+LATEX: \clearpage
#+LATEX: \setcounter{figure}{0}
#+LATEX: \setcounter{section}{0}
#+LATEX: \setcounter{table}{0}

* Supplementary materials

#+CAPTION:*(A,B,C,D)* Eight main principal components computed on the genotypes of GENESIS. Cases are colored in green, controls in orange. 
#+NAME: sfig:pcs
[[./figures/sfigure_1.pdf]]

#+CAPTION:Jaccard similarity between the different solution gene subnetworks.
#+NAME: sfig:jaccard_methods
[[./figures/sfigure_2.pdf]]

#+CAPTION:Biotypes of genes from the annotation that are not present in the HINT protein-protein interaction network.
#+NAME: sfig:biotypes_excluded
[[./figures/sfigure_3.pdf]]

#+CAPTION:Bonferroni significance, in either the GENESIS or the BCAC datasets, of the genes (and SNPs in the case of SConES) detected by the network methods, and in the consensus subnetwork. LEAN was excluded, as it did not select any gene.
#+NAME: sfig:bcac_comparison
[[./figures/sfigure_4.pdf]]

#+CAPTION:Comparison of benchmark on high-throughput interactions to benchmark on both high-throughput and literature curated interactions. Grey lines represent no change between the benchmarks (1 for ratios, 0 for differences). *(A)* Ratios of the selected features between both benchmarks and of the active set. *(B)* Shifts in sensitivity and specificity. *(C)* Shift in Jaccard similarity between benchmarks. *(D)* Ratio between the runtimes of the benchmarks.
#+NAME: sfig:lc_ht_comparison 
[[./figures/sfigure_5.pdf]]

#+CAPTION: Genes on the consensus network. Familial breast cancer genes are colored in pink; the rest are colored in grey. *(A)* Number of methods selecting every gene in the subnetwork. *(B)* VEGAS P-values of association of the genes, with regards to the number of methods that selected them. *(C)* Comparison of betweenness centrality of the genes in the consensus network and the other genes in the PPIN and not in the consensus network. To improve visualization, we removed outliers. *(D)* Relationship between the log_10 of the betweenness centrality and the -log_10 of the VEGAS P-value of the genes in the consensus network. The blue line represents a fitted generalized linear model. 
#+NAME: sfig:consensus_stats
[[./figures/sfigure_6.pdf]]
